{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import sys\n",
    "import pandas as pd\n",
    "from ngly1_gpt import utils, llm\n",
    "import logging\n",
    "logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28825"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('data/extract/PMC4243708.txt', 'r') as f:\n",
    "    text = f.read()\n",
    "len(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Title\n",
      "Mutations in NGLY1 Cause an Inherited Disorder of the Endoplasmic Reticulum-Associated Degradation (ERAD) Pathway\n",
      "\n",
      "Abstract\n",
      "Purpose: The endoplasmic reticulum-associated degradation (ERAD) pathway is responsible for the translocation of misfolded proteins across the ER membrane into the cytosol for subsequent degradation by the proteasome. In order to understand the spectrum of clinical and molecular findings in a complex neurological syndrome, we studied a series of eight patients with inherited deficiency of N-glycanase 1 (NGLY1), a novel disorder of cytosolic ERAD dysfunction.\n",
      "Methods: Whole-genome, whole-exome or standard Sanger sequencing techniques were employed. Retrospective chart reviews were performed in order to obtain clinical data.\n",
      "Results: All patients had global developmental delay, a movement disorder, and hypotonia. Other common findings included hypo- or alacrima (7/8), elevated liver transaminases (6/7), microcephaly (6/8), diminished reflexes (6/8), hepatocyte cytoplasmic storage material or vacuolization (5/6), and seizures (4/8). The nonsense mutation c.1201A>T (p.R401X) was the most common deleterious allele.\n",
      "Conclusions: NGLY1 deficiency is a novel autosomal recessive disorder of the ERAD pathway associated with neurological dysfunction, abnormal tear production, and liver disease. The majority of patients detected to date carry a specific nonsense mutation that appears to be associated with severe disease. The phenotypic spectrum is likely to enlarge as cases with a more broad range of mutations are detected.\n",
      "\n",
      "Keywords\n",
      "NGLY1; alacrima; choreoathetosis; seizures; liver disease\n",
      "\n",
      "INTRODUCTION\n",
      "The enzyme N-glycanase 1 (NGLY1), also known as peptide:N-glycanase (PNGase, EC 3.5.1.52), catalyzes protein deglycosylation by cleaving the -aspartyl glycosylamine bond of N-linked glycoproteins with the subsequent release of intact N-glycan species [1]. NGLY1 activity was first described in almond seeds [1], but is ubiquitously present across a wid\n"
     ]
    }
   ],
   "source": [
    "print(text[:2000])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "FIGURES\n",
      "Figure 1: Distribution of NGLY1 mutations.\n",
      "Shaded bars are exons. Previously unreported mutations are in bold. Mutations previously\n",
      "published and not in this paper include c.1624C>T (exon 11), c.1205_1207delGAA (exon 8) and c.1533_1536delTCAA (exon 10). Four mutations lie within the catalytic domain (amino acid residues 292–353);, i.e., c.730T>C, c.953T>C, c.931G>A, and c.930C>T. The accession numbers for the NM_018297.3:c.931G>A, NM_018297.3:c.1604G>A, NM_018297.3:c.622C>T, NM_018297.3:c.347C>G, NM_018297.3:c.1169G>C, NM_018297.3:c.730T>C, NM_018297.3:c.1910delT, NM_018297.3:c.930C>T, NM_018297.3:c.881+5G>T\n",
      "sequences reported in this paper are ClinVar: SCV000259179, SCV000259180, SCV000259181, SCV000259182, SCV000259183, SCV000259184, SCV000259185, SCV000259186, and SCV000259187 respectively.\n",
      "   \n",
      "Figure 2: Facial features of NGLY1-CDDG.\n",
      "Individuals with NGLY1-CDDG are arranged from youngest (top left) to oldest (bottom right). Facial features include upturned nasal tip, hypotonic facies, ptosis, brachycephaly, thinned facies, hollowed cheeks, and visible zygomatic arches. Interpupillary distance ranged from the 13–95 centiles with the mean at the 66th centile, and SEM of 8. Horizontal fissure length ranged from 3–70 centiles with mean at the 20th centile, and SEM of 1. Inner canthal distance ranged from 3–100 centiles with the mean at the 59th centile and SEM of 10. Outer canthal distance ranged from 6–81 centile with mean at the 43 centile and SEM of 7. Canthal index ranged from −2 to +2 SD from the normal mean.\n",
      "\n",
      "Figure 3: Brain MRI findings.\n",
      "White matter lesions were seen in two of eleven individuals. A shows multiple lesions in the periventricular white matter, some of which were confluent. B shows a single lesion in the periventricular white matter. Both A and B illustrate cerebral atrophy and were performed using T2-weighting. Sulci are slightly prominent in both A and B, and prominent ventricles are visible in A. C illustrates high position of the cerebellar tonsils, and large foramen of Magendie and cisterna magna.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# chunk = \"\"\"\n",
    "# Title\n",
    "# Mutations in NGLY1 Cause an Inherited Disorder of the Endoplasmic Reticulum-Associated Degradation (ERAD) Pathway\n",
    "\n",
    "# Abstract\n",
    "# Purpose: The endoplasmic reticulum-associated degradation (ERAD) pathway is responsible for the translocation of misfolded proteins across the ER membrane into the cytosol for subsequent degradation by the proteasome. In order to understand the spectrum of clinical and molecular findings in a complex neurological syndrome, we studied a series of eight patients with inherited deficiency of N-glycanase 1 (NGLY1), a novel disorder of cytosolic ERAD dysfunction.\n",
    "# Methods: Whole-genome, whole-exome or standard Sanger sequencing techniques were employed. Retrospective chart reviews were performed in order to obtain clinical data.\n",
    "# Results: All patients had global developmental delay, a movement disorder, and hypotonia. Other common findings included hypo- or alacrima (7/8), elevated liver transaminases (6/7), microcephaly (6/8), diminished reflexes (6/8), hepatocyte cytoplasmic storage material or vacuolization (5/6), and seizures (4/8). The nonsense mutation c.1201A>T (p.R401X) was the most common deleterious allele.\n",
    "# Conclusions: NGLY1 deficiency is a novel autosomal recessive disorder of the ERAD pathway associated with neurological dysfunction, abnormal tear production, and liver disease. The majority of patients detected to date carry a specific nonsense mutation that appears to be associated with severe disease. The phenotypic spectrum is likely to enlarge as cases with a more broad range of mutations are detected.\n",
    "# \"\"\"\n",
    "# chunk = \"\"\"\n",
    "# DISCUSSION\n",
    "# Summary of Novel Findings\n",
    "# Our prospective investigations into NGLY1-CDDG revealed several new discoveries associated with this disorder. These included low CSF total protein and albumin, optic atrophy and retinal pigmentary changes/cone dystrophy, poor weight gain starting in mid- childhood, lower than predicted resting energy expenditure, central and obstructive sleep apnea, delayed bone age, joint hypermobility, progressive brain atrophy, abnormalities in brain MRS-measured metabolites, various abnormalities on skeletal survey, high antibody titers after rubella and rubeola vaccination, and low levels of several pro- and anti- coagulation factors. Echocardiograms, electrocardiograms, and CSF lactate levels were normal.\n",
    "# We also confirmed and further detailed the global developmental delay, movement disorder, frequent seizure disorder, hypotonia, hypolacrima or alacrima, corneal disease, ptosis, lagophthalmous, strabismus, peripheral neuropathy and occasional diminished reflexes, hypohidrosis, auditory brainstem response abnormalities, abnormal brain imaging, scoliosis, acquired microcephaly, small hands and feet, dysmorphic features, constipation, elevated liver enzymes present only early in childhood, osteopenia, and hypocholesterolemia associated with NGLY1-CDDG. Compared with previously reported biopsy findings, our transient elastography results showed minimal if any fibrosis. In addition, we could not confirm abnormal storage material in the three liver biopsy specimens we analyzed, nor did we observe ocular apraxia in our individuals, despite performing detailed ophthalmologic evaluations that involved some subjects previously reported to have ocular apraxia11.\n",
    "# Based on our findings, we recommend considering NGLY1-CDDG in the differential diagnosis of any patient with the tetrad of developmental delay/intellectual disability, hyperkinetic movement disorder, hypolacrima, and a history of elevated transaminases during early childhood.\n",
    "# \"\"\"\n",
    "chunk = \"\"\"\n",
    "\n",
    "FIGURES\n",
    "Figure 1: Distribution of NGLY1 mutations.\n",
    "Shaded bars are exons. Previously unreported mutations are in bold. Mutations previously\n",
    "published and not in this paper include c.1624C>T (exon 11), c.1205_1207delGAA (exon 8) and c.1533_1536delTCAA (exon 10). Four mutations lie within the catalytic domain (amino acid residues 292–353);, i.e., c.730T>C, c.953T>C, c.931G>A, and c.930C>T. The accession numbers for the NM_018297.3:c.931G>A, NM_018297.3:c.1604G>A, NM_018297.3:c.622C>T, NM_018297.3:c.347C>G, NM_018297.3:c.1169G>C, NM_018297.3:c.730T>C, NM_018297.3:c.1910delT, NM_018297.3:c.930C>T, NM_018297.3:c.881+5G>T\n",
    "sequences reported in this paper are ClinVar: SCV000259179, SCV000259180, SCV000259181, SCV000259182, SCV000259183, SCV000259184, SCV000259185, SCV000259186, and SCV000259187 respectively.\n",
    "   \n",
    "Figure 2: Facial features of NGLY1-CDDG.\n",
    "Individuals with NGLY1-CDDG are arranged from youngest (top left) to oldest (bottom right). Facial features include upturned nasal tip, hypotonic facies, ptosis, brachycephaly, thinned facies, hollowed cheeks, and visible zygomatic arches. Interpupillary distance ranged from the 13–95 centiles with the mean at the 66th centile, and SEM of 8. Horizontal fissure length ranged from 3–70 centiles with mean at the 20th centile, and SEM of 1. Inner canthal distance ranged from 3–100 centiles with the mean at the 59th centile and SEM of 10. Outer canthal distance ranged from 6–81 centile with mean at the 43 centile and SEM of 7. Canthal index ranged from −2 to +2 SD from the normal mean.\n",
    "\n",
    "Figure 3: Brain MRI findings.\n",
    "White matter lesions were seen in two of eleven individuals. A shows multiple lesions in the periventricular white matter, some of which were confluent. B shows a single lesion in the periventricular white matter. Both A and B illustrate cerebral atrophy and were performed using T2-weighting. Sulci are slightly prominent in both A and B, and prominent ventricles are visible in A. C illustrates high position of the cerebellar tonsils, and large foramen of Magendie and cisterna magna.\n",
    "\"\"\"\n",
    "print(chunk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:ngly1_gpt.llm:Prompt:\n",
      "Text will be provided that contains information from a published, biomedical research article about \"NGLY1 deficiency\". Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Extract only subjects and objects that are specific, named concepts pertaining to the following biomedical entities:\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- tissue\n",
      "- chemical substance\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- protein\n",
      "- gene family \n",
      "- genotype\n",
      "- genetic variant\n",
      "- transcript variant\n",
      "- protein variant\n",
      "- macromolecular complex\n",
      "- molecular activity\n",
      "- organism\n",
      "- pathway\n",
      "- phenotype\n",
      "- symptom\n",
      "\n",
      "Extract only predicates relating the concepts above that are semantically equivalent, or nearly so, to any of the following:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "\n",
      "Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "\n",
      "\n",
      "FIGURES\n",
      "Figure 1: Distribution of NGLY1 mutations.\n",
      "Shaded bars are exons. Previously unreported mutations are in bold. Mutations previously\n",
      "published and not in this paper include c.1624C>T (exon 11), c.1205_1207delGAA (exon 8) and c.1533_1536delTCAA (exon 10). Four mutations lie within the catalytic domain (amino acid residues 292–353);, i.e., c.730T>C, c.953T>C, c.931G>A, and c.930C>T. The accession numbers for the NM_018297.3:c.931G>A, NM_018297.3:c.1604G>A, NM_018297.3:c.622C>T, NM_018297.3:c.347C>G, NM_018297.3:c.1169G>C, NM_018297.3:c.730T>C, NM_018297.3:c.1910delT, NM_018297.3:c.930C>T, NM_018297.3:c.881+5G>T\n",
      "sequences reported in this paper are ClinVar: SCV000259179, SCV000259180, SCV000259181, SCV000259182, SCV000259183, SCV000259184, SCV000259185, SCV000259186, and SCV000259187 respectively.\n",
      "   \n",
      "Figure 2: Facial features of NGLY1-CDDG.\n",
      "Individuals with NGLY1-CDDG are arranged from youngest (top left) to oldest (bottom right). Facial features include upturned nasal tip, hypotonic facies, ptosis, brachycephaly, thinned facies, hollowed cheeks, and visible zygomatic arches. Interpupillary distance ranged from the 13–95 centiles with the mean at the 66th centile, and SEM of 8. Horizontal fissure length ranged from 3–70 centiles with mean at the 20th centile, and SEM of 1. Inner canthal distance ranged from 3–100 centiles with the mean at the 59th centile and SEM of 10. Outer canthal distance ranged from 6–81 centile with mean at the 43 centile and SEM of 7. Canthal index ranged from −2 to +2 SD from the normal mean.\n",
      "\n",
      "Figure 3: Brain MRI findings.\n",
      "White matter lesions were seen in two of eleven individuals. A shows multiple lesions in the periventricular white matter, some of which were confluent. B shows a single lesion in the periventricular white matter. Both A and B illustrate cerebral atrophy and were performed using T2-weighting. Sulci are slightly prominent in both A and B, and prominent ventricles are visible in A. C illustrates high position of the cerebellar tonsils, and large foramen of Magendie and cisterna magna.\n",
      "\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`. The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"). \n",
      "\n",
      "Here is an example response with a single record:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report the CSV headers and the associated records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "DEBUG:openai:message='Request to OpenAI API' method=post path=https://api.openai.com/v1/chat/completions\n",
      "DEBUG:openai:api_version=None data='{\"model\": \"gpt-4\", \"messages\": [{\"role\": \"user\", \"content\": \"Text will be provided that contains information from a published, biomedical research article about \\\\\"NGLY1 deficiency\\\\\". Extract subject-predicate-object relations from this text.\\\\n\\\\nExtract only subjects and objects that are specific, named concepts pertaining to the following biomedical entities:\\\\n- biological process\\\\n- cell type\\\\n- cellular component\\\\n- tissue\\\\n- chemical substance\\\\n- disease\\\\n- drug\\\\n- gene\\\\n- protein\\\\n- gene family \\\\n- genotype\\\\n- genetic variant\\\\n- transcript variant\\\\n- protein variant\\\\n- macromolecular complex\\\\n- molecular activity\\\\n- organism\\\\n- pathway\\\\n- phenotype\\\\n- symptom\\\\n\\\\nExtract only predicates relating the concepts above that are semantically equivalent, or nearly so, to any of the following:\\\\n- affects risk for\\\\n- associated with\\\\n- capable of\\\\n- caused by\\\\n- causes\\\\n- colocalizes with\\\\n- contributes to\\\\n- correlated with\\\\n- decreases abundance of\\\\n- decreases activity of\\\\n- derives from\\\\n- disrupts\\\\n- enables\\\\n- exact match\\\\n- expressed in\\\\n- expresses\\\\n- genetically interacts with\\\\n- has affected feature\\\\n- has attribute\\\\n- has gene product\\\\n- has genotype\\\\n- has metabolite\\\\n- has output\\\\n- has participant\\\\n- has phenotype\\\\n- has role\\\\n- in 1 to 1 orthology relationship with\\\\n- in orthology relationship with\\\\n- in paralogy relationship with\\\\n- in taxon\\\\n- in xenology relationship with\\\\n- increases abundance of\\\\n- increases activity of\\\\n- instance of\\\\n- interacts with\\\\n- involved in\\\\n- is allele of\\\\n- is marker for\\\\n- is model of\\\\n- is part of\\\\n- located in\\\\n- location of\\\\n- manifestation of\\\\n- molecularly interacts with\\\\n- negatively correlated with\\\\n- negatively regulates\\\\n- occurs in\\\\n- part of\\\\n- participates in\\\\n- pathogenic for condition\\\\n- positively correlated with\\\\n- positively regulates\\\\n- precedes\\\\n- prevents\\\\n- produced by\\\\n- produces\\\\n- regulates\\\\n- treats\\\\n\\\\nAssume that \\\\\"NGLY1 deficiency\\\\\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\\\\n\\\\nHere is the text to extract relations from: \\\\n--- BEGIN TEXT ---\\\\n\\\\n\\\\nFIGURES\\\\nFigure 1: Distribution of NGLY1 mutations.\\\\nShaded bars are exons. Previously unreported mutations are in bold. Mutations previously\\\\npublished and not in this paper include c.1624C>T (exon 11), c.1205_1207delGAA (exon 8) and c.1533_1536delTCAA (exon 10). Four mutations lie within the catalytic domain (amino acid residues 292\\\\u2013353);, i.e., c.730T>C, c.953T>C, c.931G>A, and c.930C>T. The accession numbers for the NM_018297.3:c.931G>A, NM_018297.3:c.1604G>A, NM_018297.3:c.622C>T, NM_018297.3:c.347C>G, NM_018297.3:c.1169G>C, NM_018297.3:c.730T>C, NM_018297.3:c.1910delT, NM_018297.3:c.930C>T, NM_018297.3:c.881+5G>T\\\\nsequences reported in this paper are ClinVar: SCV000259179, SCV000259180, SCV000259181, SCV000259182, SCV000259183, SCV000259184, SCV000259185, SCV000259186, and SCV000259187 respectively.\\\\n   \\\\nFigure 2: Facial features of NGLY1-CDDG.\\\\nIndividuals with NGLY1-CDDG are arranged from youngest (top left) to oldest (bottom right). Facial features include upturned nasal tip, hypotonic facies, ptosis, brachycephaly, thinned facies, hollowed cheeks, and visible zygomatic arches. Interpupillary distance ranged from the 13\\\\u201395 centiles with the mean at the 66th centile, and SEM of 8. Horizontal fissure length ranged from 3\\\\u201370 centiles with mean at the 20th centile, and SEM of 1. Inner canthal distance ranged from 3\\\\u2013100 centiles with the mean at the 59th centile and SEM of 10. Outer canthal distance ranged from 6\\\\u201381 centile with mean at the 43 centile and SEM of 7. Canthal index ranged from \\\\u22122 to +2 SD from the normal mean.\\\\n\\\\nFigure 3: Brain MRI findings.\\\\nWhite matter lesions were seen in two of eleven individuals. A shows multiple lesions in the periventricular white matter, some of which were confluent. B shows a single lesion in the periventricular white matter. Both A and B illustrate cerebral atrophy and were performed using T2-weighting. Sulci are slightly prominent in both A and B, and prominent ventricles are visible in A. C illustrates high position of the cerebellar tonsils, and large foramen of Magendie and cisterna magna.\\\\n\\\\n--- END TEXT ---\\\\n\\\\nReport each relation record on a separate line in CSV format using a pipe (i.e. \\\\\"|\\\\\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`. The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \\\\\"pathway\\\\\" or \\\\\"disease\\\\\"). \\\\n\\\\nHere is an example response with a single record:\\\\n--- BEGIN EXAMPLE ---\\\\nsubject|subject_entity|predicate|object|object_entity\\\\nHutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\\\\n--- END EXAMPLE ---\\\\n\\\\nReport the CSV headers and the associated records without explanation or other text of any kind.\\\\n\\\\nExtracted relation records:\"}]}' message='Post details'\n",
      "DEBUG:urllib3.util.retry:Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.openai.com:443\n",
      "DEBUG:urllib3.connectionpool:https://api.openai.com:443 \"POST /v1/chat/completions HTTP/1.1\" 200 None\n",
      "DEBUG:openai:message='OpenAI API response' path=https://api.openai.com/v1/chat/completions processing_ms=40825 request_id=c25c1867a60c9d556210e46237ff8ad3 response_code=200\n",
      "DEBUG:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|Facial features|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|White matter lesions|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|cerebral atrophy|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|prominent ventricles|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|high position of the cerebellar tonsils|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|large foramen of Magendie and cisterna magna|phenotype\n",
      "NGLY1|gene|has genetic variant|c.1624C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.1205_1207delGAA|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.1533_1536delTCAA|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.730T>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.953T>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.931G>A|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.930C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.931G>A|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.1604G>A|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.622C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.347C>G|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.1169G>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.730T>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.1910delT|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.930C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.881+5G>T|genetic variant\n",
      "----------------------------------------------------------------------------------------------------\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|Facial features|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|White matter lesions|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|cerebral atrophy|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|prominent ventricles|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|high position of the cerebellar tonsils|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|large foramen of Magendie and cisterna magna|phenotype\n",
      "NGLY1|gene|has genetic variant|c.1624C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.1205_1207delGAA|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.1533_1536delTCAA|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.730T>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.953T>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.931G>A|genetic variant\n",
      "NGLY1|gene|has genetic variant|c.930C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.931G>A|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.1604G>A|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.622C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.347C>G|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.1169G>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.730T>C|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.1910delT|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.930C>T|genetic variant\n",
      "NGLY1|gene|has genetic variant|NM_018297.3:c.881+5G>T|genetic variant\n"
     ]
    }
   ],
   "source": [
    "result = llm.extract_relations(chunk, disease=\"NGLY1 deficiency\", model=\"gpt-4\")\n",
    "print('-'*100)\n",
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
