{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### [POC 3]  NGLY1 deficiency patient extraction\n",
    "\n",
    "This analysis attempts to extract as many details as possible about NGLY1 patients from 2 full-text papers on the subject.  Outline:\n",
    "\n",
    "1. Define a minimal prompt to per-study patient identifiers and free-text descriptions of associated information\n",
    "2. Summarize the results for each patient, across studies, in a standard schema\n",
    "3. Analyze the patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import io\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from ngly1_gpt import utils, llm, doc\n",
    "import logging\n",
    "logging.basicConfig(level=logging.INFO, stream=sys.stdout)\n",
    "pd.set_option(\"display.max_colwidth\", None, \"display.max_rows\", 400, \"display.max_columns\", None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Text will be provided that contains information from a published, biomedical research article about {disease}.  Extract details about the patients discussed in this text. \n",
      "\n",
      "Requirements:\n",
      "\n",
      "- Exclude any patients where context dictates that they do NOT have {disease}, e.g. when {disease} patients are compared to similar patients with other diseases.\n",
      "- Extract as much information as possible about each patient including associated genotypes, phenotypes, physical or behavioral traits, demographics, lab measurements, treatments, family histories or anything else of clinical and/or biological relevance.\n",
      "- Extract this information in CSV format with the following headers:\n",
      "  - `patient_id`: Identifying information for the patient within the context of the article; typically an integer or anonymized id like \"Patient 1\". If some information applies to ALL patients in a study and the context does not make it possible to enumerate the patient ids, report only the value \"ALL\"\n",
      "  - `external_study`: If context implies that the patient being referenced is from another study, include any identifying information for that study/reference in this field (e.g. citation numbers or author names); if no such distinction is made, assume it is from the current study and report only the value \"NA\"\n",
      "  - `details`: A free-text description of the information discussed for the patient\n",
      "- Ensure that any `details` for a patient can definitively be ascribed to that specific patient\n",
      "- Report results using a pipe (i.e. \"|\") delimiter \n",
      "- Include the CSV headers in the result\n",
      "- If the text contains no patient information, report only the CSV headers\n",
      "- Do NOT include explanation of any kind in the results; report only the CSV content\n",
      "\n",
      "Patient details CSV:\n"
     ]
    }
   ],
   "source": [
    "prompt = (utils.get_paths().prompts / \"patient_extraction_1.txt\").read_text()\n",
    "print(prompt)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Execution"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The prompt was run for all paper chunks via a command like:\n",
    "\n",
    "```bash\n",
    "PYTHONPATH=\"$(pwd)\" python ngly1_gpt/cli.py extract_patients --output-filename=patients_1.tsv 2>&1 | tee data/logs/extract_patients_1.log.txt\n",
    "```\n",
    "\n",
    "The logs for these extractions showing all prompts and results are in [data/logs](data/logs)."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Examples"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Cleanup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 180 entries, 0 to 179\n",
      "Data columns (total 6 columns):\n",
      " #   Column                             Non-Null Count  Dtype \n",
      "---  ------                             --------------  ----- \n",
      " 0   patient_id                         173 non-null    object\n",
      " 1   external_study                     14 non-null     object\n",
      " 2   details                            173 non-null    object\n",
      " 3   doc_id                             180 non-null    object\n",
      " 4   doc_filename                       180 non-null    object\n",
      " 5   patient_id,external_study,details  7 non-null      object\n",
      "dtypes: object(6)\n",
      "memory usage: 8.6+ KB\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>external_study</th>\n",
       "      <th>details</th>\n",
       "      <th>doc_id</th>\n",
       "      <th>doc_filename</th>\n",
       "      <th>patient_id,external_study,details</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>All patients had global developmental delay, a movement disorder, and hypotonia.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7 out of 8 patients had hypo- or alacrima.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6 out of 7 patients had elevated liver transaminases.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6 out of 8 patients had microcephaly.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6 out of 8 patients had diminished reflexes.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 out of 6 patients had hepatocyte cytoplasmic storage material or vacuolization.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4 out of 8 patients had seizures.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The most common deleterious allele was the nonsense mutation c.1201A&gt;T (p.R401X).</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NGLY1 deficiency is a novel autosomal recessive disorder of the ERAD pathway associated with neurological dysfunction, abnormal tear production, and liver disease. The majority of patients detected to date carry a specific nonsense mutation that appears to be associated with severe disease.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>Need et al.</td>\n",
       "      <td>3-year-old boy with compound heterozygous inactivating mutations in NGLY1. Clinical phenotype suggestive of a congenital disorder of glycosylation, although repeated transferrin isoelectric focusing and N-glycan analyses were normal. Liver biopsy showed accumulation of an amorphous unidentified substance throughout the cytoplasm, a finding likely consistent with NGLY1 dysfunction, which would be expected to result in abnormal accumulation of misfolded glycoproteins because of impaired cytosolic degradation.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seven newly diagnosed patients with mutations in NGLY1. These observations confirm NGLY1 deficiency as an inherited disorder associated with the ERAD process and document its clinical presentation.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>Duke University</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Patient 5</td>\n",
       "      <td>University of British Columbia</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>University of British Columbia</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>The patient and parents were sequenced using both Illumina HiSeq2000 and Complete Genomics platforms. Variants in Illumina-sequenced reads were called using both the Hugeseq and Real Time Genomics pipelines and Complete Genomics variants were identified by their own variant callers.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient_id                  external_study  \\\n",
       "0         ALL                             NaN   \n",
       "1        SOME                             NaN   \n",
       "2        SOME                             NaN   \n",
       "3        SOME                             NaN   \n",
       "4        SOME                             NaN   \n",
       "5        SOME                             NaN   \n",
       "6        SOME                             NaN   \n",
       "7         ALL                             NaN   \n",
       "8         ALL                             NaN   \n",
       "9   Patient 1                     Need et al.   \n",
       "10        ALL                             NaN   \n",
       "11  Patient 1                 Duke University   \n",
       "12  Patient 5  University of British Columbia   \n",
       "13  Patient 6  University of British Columbia   \n",
       "14  Patient 3             Stanford University   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             details  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                   All patients had global developmental delay, a movement disorder, and hypotonia.   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         7 out of 8 patients had hypo- or alacrima.   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                              6 out of 7 patients had elevated liver transaminases.   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              6 out of 8 patients had microcephaly.   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       6 out of 8 patients had diminished reflexes.   \n",
       "5                                                                                                                                                                                                                                                                                                                                                                                                                                                  5 out of 6 patients had hepatocyte cytoplasmic storage material or vacuolization.   \n",
       "6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  4 out of 8 patients had seizures.   \n",
       "7                                                                                                                                                                                                                                                                                                                                                                                                                                                  The most common deleterious allele was the nonsense mutation c.1201A>T (p.R401X).   \n",
       "8                                                                                                                                                                                                                                NGLY1 deficiency is a novel autosomal recessive disorder of the ERAD pathway associated with neurological dysfunction, abnormal tear production, and liver disease. The majority of patients detected to date carry a specific nonsense mutation that appears to be associated with severe disease.   \n",
       "9   3-year-old boy with compound heterozygous inactivating mutations in NGLY1. Clinical phenotype suggestive of a congenital disorder of glycosylation, although repeated transferrin isoelectric focusing and N-glycan analyses were normal. Liver biopsy showed accumulation of an amorphous unidentified substance throughout the cytoplasm, a finding likely consistent with NGLY1 dysfunction, which would be expected to result in abnormal accumulation of misfolded glycoproteins because of impaired cytosolic degradation.   \n",
       "10                                                                                                                                                                                                                                                                                                                             Seven newly diagnosed patients with mutations in NGLY1. These observations confirm NGLY1 deficiency as an inherited disorder associated with the ERAD process and document its clinical presentation.   \n",
       "11                                                                                                                                                                                                                                                                                                                                                                                                         Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "12                                                                                                                                                                                                                                                                                                                                                                                                         Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "13                                                                                                                                                                                                                                                                                                                                                                                                         Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "14                                                                                                                                                                                                                                       The patient and parents were sequenced using both Illumina HiSeq2000 and Complete Genomics platforms. Variants in Illumina-sequenced reads were called using both the Hugeseq and Real Time Genomics pipelines and Complete Genomics variants were identified by their own variant callers.   \n",
       "\n",
       "        doc_id    doc_filename patient_id,external_study,details  \n",
       "0   PMC4243708  PMC4243708.txt                               NaN  \n",
       "1   PMC4243708  PMC4243708.txt                               NaN  \n",
       "2   PMC4243708  PMC4243708.txt                               NaN  \n",
       "3   PMC4243708  PMC4243708.txt                               NaN  \n",
       "4   PMC4243708  PMC4243708.txt                               NaN  \n",
       "5   PMC4243708  PMC4243708.txt                               NaN  \n",
       "6   PMC4243708  PMC4243708.txt                               NaN  \n",
       "7   PMC4243708  PMC4243708.txt                               NaN  \n",
       "8   PMC4243708  PMC4243708.txt                               NaN  \n",
       "9   PMC4243708  PMC4243708.txt                               NaN  \n",
       "10  PMC4243708  PMC4243708.txt                               NaN  \n",
       "11  PMC4243708  PMC4243708.txt                               NaN  \n",
       "12  PMC4243708  PMC4243708.txt                               NaN  \n",
       "13  PMC4243708  PMC4243708.txt                               NaN  \n",
       "14  PMC4243708  PMC4243708.txt                               NaN  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patients_raw = pd.read_csv(utils.get_paths().output_data / \"patients_1.tsv\", sep=\"\\t\")\n",
    "patients_raw.info()\n",
    "patients_raw.head(15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>external_study</th>\n",
       "      <th>details</th>\n",
       "      <th>doc_id</th>\n",
       "      <th>doc_filename</th>\n",
       "      <th>patient_id,external_study,details</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>ALL,24,Patients with autosomal recessive mutations in ERLIN2 present profound intellectual disability, developmental regression and multiple contractures. Despite the severity of the intellectual disability and neuromuscular findings, the results of brain imaging, electromyography and muscle biopsy appeared normal in the initial erlin2-deficient patients.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "      <td>ALL,26,Another family was found to have a homozygous null mutation in ERLIN2, with affected individuals presenting with a hereditary spastic paraplegia phenotype.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "      <td>ALL,NA,Clinical studies were designed to detail the phenotypic features of NGLY1-CDDG. Blood, urine, cerebral spinal fluid (CSF), lymphoblasts, and primary dermal fibroblasts were collected, analyzed, and stored. Studies included brain magnetic resonance imaging and spectroscopy (MRI and MRS, supplementary methods), routine and overnight electroencephalograms (EEGs) with a limited montage performed during a sleep study, electromyogram (EMG, supplementary methods) and nerve conduction studies (NCS, supplementary methods), indirect calorimetry, awake and sedated eye examination with Schirmer II testing, optical coherence tomography scans and electroretinography, behavioral determination of pure tone thresholds, tympanometry, distortion product otoacoustic emissions, auditory brainstem evoked potentials (ABR), quantitative sweat analysis autonomic testing (QSWEAT, supplementary methods), gastric aspiration, swallow study, skeletal survey, bone age, dual X-ray absorptiometry (DEXA), abdominal ultrasound, vibration controlled transient elastography (Fibroscan)12, echocardiogram, and electrocardiogram. Consultations included clinical neurology, audiology, nutrition, ophthalmology, hepatology, growth, puberty and hormonal studies, allergy and immunology, genetic counseling, physiatry, and speech, occupational, and physical therapy.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "      <td>ALL,NA,Eleven individuals underwent developmental psychological evaluations, consisting of at least the Vineland Adaptive Behavior Scales, 2nd edition. Cognitive function was assessed with testing specific for age and developmental level that provided either an intelligence quotient (IQ) or developmental quotient (DQ) score.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "      <td>ALL,NA,The Nijmegen pediatric CDG rating scale, a measure of clinical disease progression developed for CDG, was applied to all affected individuals younger than 18 years.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "      <td>ALL,NA,Compared to the reference population, N-acetylaspartylglutamate + N-acetylaspartate (NAA) was lower than normal in the left centrum semiovale (LCSO) (p=0.004), the midline parietal grey matter (PGM) (p=0.02), and superior cerebellar vermis (SVERM) (p&lt;0.0001). There was a deficit of glutamine + glutamate + gamma-aminobutyric acid (Glx) in the PGM (p=0.03), LCSO (p=0.01), and pons (p=0.0002). Choline was higher than expected for age only in the LCSO (p=0.0097), and myo-inositol was higher than expected for age in the pons (p=0.002). Multiple correlations between these MRS-measured metabolites and age, functional assessments, brain volume, and neurotransmitters in the CSF were found. The general trend showed that the differences noted above became more pronounced with increasing age, worsening function, and lower brain volume. MRS metabolite measurements did not correlate with total CSF protein, CSF albumin, or CSF/serum albumin ratio. There was a weak correlation (p=0.09) between atrophy and total CSF protein, but not CSF albumin or CSF/serum albumin ratio.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "      <td>ALL,NA,Strong correlation between brain atrophy on MRI and functional assessments suggests that loss of neurons contributes to the functional impairment. The atrophy also correlated with CSF metabolites (BH4, 5-HIAA, HVA), which are known to be lower when there is damage to neurotransmitter producing neurons. This suggests that these biochemical abnormalities may be secondary to brain atrophy.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    patient_id external_study details      doc_id    doc_filename  \\\n",
       "62         NaN            NaN     NaN  PMC4243708  PMC4243708.txt   \n",
       "63         NaN            NaN     NaN  PMC4243708  PMC4243708.txt   \n",
       "83         NaN            NaN     NaN  PMC7477955  PMC7477955.txt   \n",
       "84         NaN            NaN     NaN  PMC7477955  PMC7477955.txt   \n",
       "85         NaN            NaN     NaN  PMC7477955  PMC7477955.txt   \n",
       "131        NaN            NaN     NaN  PMC7477955  PMC7477955.txt   \n",
       "169        NaN            NaN     NaN  PMC7477955  PMC7477955.txt   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      patient_id,external_study,details  \n",
       "62                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ALL,24,Patients with autosomal recessive mutations in ERLIN2 present profound intellectual disability, developmental regression and multiple contractures. Despite the severity of the intellectual disability and neuromuscular findings, the results of brain imaging, electromyography and muscle biopsy appeared normal in the initial erlin2-deficient patients.  \n",
       "63                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   ALL,26,Another family was found to have a homozygous null mutation in ERLIN2, with affected individuals presenting with a hereditary spastic paraplegia phenotype.  \n",
       "83   ALL,NA,Clinical studies were designed to detail the phenotypic features of NGLY1-CDDG. Blood, urine, cerebral spinal fluid (CSF), lymphoblasts, and primary dermal fibroblasts were collected, analyzed, and stored. Studies included brain magnetic resonance imaging and spectroscopy (MRI and MRS, supplementary methods), routine and overnight electroencephalograms (EEGs) with a limited montage performed during a sleep study, electromyogram (EMG, supplementary methods) and nerve conduction studies (NCS, supplementary methods), indirect calorimetry, awake and sedated eye examination with Schirmer II testing, optical coherence tomography scans and electroretinography, behavioral determination of pure tone thresholds, tympanometry, distortion product otoacoustic emissions, auditory brainstem evoked potentials (ABR), quantitative sweat analysis autonomic testing (QSWEAT, supplementary methods), gastric aspiration, swallow study, skeletal survey, bone age, dual X-ray absorptiometry (DEXA), abdominal ultrasound, vibration controlled transient elastography (Fibroscan)12, echocardiogram, and electrocardiogram. Consultations included clinical neurology, audiology, nutrition, ophthalmology, hepatology, growth, puberty and hormonal studies, allergy and immunology, genetic counseling, physiatry, and speech, occupational, and physical therapy.  \n",
       "84                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               ALL,NA,Eleven individuals underwent developmental psychological evaluations, consisting of at least the Vineland Adaptive Behavior Scales, 2nd edition. Cognitive function was assessed with testing specific for age and developmental level that provided either an intelligence quotient (IQ) or developmental quotient (DQ) score.  \n",
       "85                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          ALL,NA,The Nijmegen pediatric CDG rating scale, a measure of clinical disease progression developed for CDG, was applied to all affected individuals younger than 18 years.  \n",
       "131                                                                                                                                                                                                                                                                              ALL,NA,Compared to the reference population, N-acetylaspartylglutamate + N-acetylaspartate (NAA) was lower than normal in the left centrum semiovale (LCSO) (p=0.004), the midline parietal grey matter (PGM) (p=0.02), and superior cerebellar vermis (SVERM) (p<0.0001). There was a deficit of glutamine + glutamate + gamma-aminobutyric acid (Glx) in the PGM (p=0.03), LCSO (p=0.01), and pons (p=0.0002). Choline was higher than expected for age only in the LCSO (p=0.0097), and myo-inositol was higher than expected for age in the pons (p=0.002). Multiple correlations between these MRS-measured metabolites and age, functional assessments, brain volume, and neurotransmitters in the CSF were found. The general trend showed that the differences noted above became more pronounced with increasing age, worsening function, and lower brain volume. MRS metabolite measurements did not correlate with total CSF protein, CSF albumin, or CSF/serum albumin ratio. There was a weak correlation (p=0.09) between atrophy and total CSF protein, but not CSF albumin or CSF/serum albumin ratio.  \n",
       "169                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        ALL,NA,Strong correlation between brain atrophy on MRI and functional assessments suggests that loss of neurons contributes to the functional impairment. The atrophy also correlated with CSF metabolites (BH4, 5-HIAA, HVA), which are known to be lower when there is damage to neurotransmitter producing neurons. This suggests that these biochemical abnormalities may be secondary to brain atrophy.  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patients_raw.dropna(subset='patient_id,external_study,details')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>external_study</th>\n",
       "      <th>details</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>ALL</td>\n",
       "      <td>24</td>\n",
       "      <td>Patients with autosomal recessive mutations in ERLIN2 present profound intellectual disability, developmental regression and multiple contractures. Despite the severity of the intellectual disability and neuromuscular findings, the results of brain imaging, electromyography and muscle biopsy appeared normal in the initial erlin2-deficient patients.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>ALL</td>\n",
       "      <td>26</td>\n",
       "      <td>Another family was found to have a homozygous null mutation in ERLIN2, with affected individuals presenting with a hereditary spastic paraplegia phenotype.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NA</td>\n",
       "      <td>Clinical studies were designed to detail the phenotypic features of NGLY1-CDDG. Blood, urine, cerebral spinal fluid (CSF), lymphoblasts, and primary dermal fibroblasts were collected, analyzed, and stored. Studies included brain magnetic resonance imaging and spectroscopy (MRI and MRS, supplementary methods), routine and overnight electroencephalograms (EEGs) with a limited montage performed during a sleep study, electromyogram (EMG, supplementary methods) and nerve conduction studies (NCS, supplementary methods), indirect calorimetry, awake and sedated eye examination with Schirmer II testing, optical coherence tomography scans and electroretinography, behavioral determination of pure tone thresholds, tympanometry, distortion product otoacoustic emissions, auditory brainstem evoked potentials (ABR), quantitative sweat analysis autonomic testing (QSWEAT, supplementary methods), gastric aspiration, swallow study, skeletal survey, bone age, dual X-ray absorptiometry (DEXA), abdominal ultrasound, vibration controlled transient elastography (Fibroscan)12, echocardiogram, and electrocardiogram. Consultations included clinical neurology, audiology, nutrition, ophthalmology, hepatology, growth, puberty and hormonal studies, allergy and immunology, genetic counseling, physiatry, and speech, occupational, and physical therapy.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NA</td>\n",
       "      <td>Eleven individuals underwent developmental psychological evaluations, consisting of at least the Vineland Adaptive Behavior Scales, 2nd edition. Cognitive function was assessed with testing specific for age and developmental level that provided either an intelligence quotient (IQ) or developmental quotient (DQ) score.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NA</td>\n",
       "      <td>The Nijmegen pediatric CDG rating scale, a measure of clinical disease progression developed for CDG, was applied to all affected individuals younger than 18 years.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NA</td>\n",
       "      <td>Compared to the reference population, N-acetylaspartylglutamate + N-acetylaspartate (NAA) was lower than normal in the left centrum semiovale (LCSO) (p=0.004), the midline parietal grey matter (PGM) (p=0.02), and superior cerebellar vermis (SVERM) (p&lt;0.0001). There was a deficit of glutamine + glutamate + gamma-aminobutyric acid (Glx) in the PGM (p=0.03), LCSO (p=0.01), and pons (p=0.0002). Choline was higher than expected for age only in the LCSO (p=0.0097), and myo-inositol was higher than expected for age in the pons (p=0.002). Multiple correlations between these MRS-measured metabolites and age, functional assessments, brain volume, and neurotransmitters in the CSF were found. The general trend showed that the differences noted above became more pronounced with increasing age, worsening function, and lower brain volume. MRS metabolite measurements did not correlate with total CSF protein, CSF albumin, or CSF/serum albumin ratio. There was a weak correlation (p=0.09) between atrophy and total CSF protein, but not CSF albumin or CSF/serum albumin ratio.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NA</td>\n",
       "      <td>Strong correlation between brain atrophy on MRI and functional assessments suggests that loss of neurons contributes to the functional impairment. The atrophy also correlated with CSF metabolites (BH4, 5-HIAA, HVA), which are known to be lower when there is damage to neurotransmitter producing neurons. This suggests that these biochemical abnormalities may be secondary to brain atrophy.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    patient_id external_study  \\\n",
       "62         ALL             24   \n",
       "63         ALL             26   \n",
       "83         ALL             NA   \n",
       "84         ALL             NA   \n",
       "85         ALL             NA   \n",
       "131        ALL             NA   \n",
       "169        ALL             NA   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         details  \n",
       "62                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Patients with autosomal recessive mutations in ERLIN2 present profound intellectual disability, developmental regression and multiple contractures. Despite the severity of the intellectual disability and neuromuscular findings, the results of brain imaging, electromyography and muscle biopsy appeared normal in the initial erlin2-deficient patients.  \n",
       "63                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Another family was found to have a homozygous null mutation in ERLIN2, with affected individuals presenting with a hereditary spastic paraplegia phenotype.  \n",
       "83   Clinical studies were designed to detail the phenotypic features of NGLY1-CDDG. Blood, urine, cerebral spinal fluid (CSF), lymphoblasts, and primary dermal fibroblasts were collected, analyzed, and stored. Studies included brain magnetic resonance imaging and spectroscopy (MRI and MRS, supplementary methods), routine and overnight electroencephalograms (EEGs) with a limited montage performed during a sleep study, electromyogram (EMG, supplementary methods) and nerve conduction studies (NCS, supplementary methods), indirect calorimetry, awake and sedated eye examination with Schirmer II testing, optical coherence tomography scans and electroretinography, behavioral determination of pure tone thresholds, tympanometry, distortion product otoacoustic emissions, auditory brainstem evoked potentials (ABR), quantitative sweat analysis autonomic testing (QSWEAT, supplementary methods), gastric aspiration, swallow study, skeletal survey, bone age, dual X-ray absorptiometry (DEXA), abdominal ultrasound, vibration controlled transient elastography (Fibroscan)12, echocardiogram, and electrocardiogram. Consultations included clinical neurology, audiology, nutrition, ophthalmology, hepatology, growth, puberty and hormonal studies, allergy and immunology, genetic counseling, physiatry, and speech, occupational, and physical therapy.  \n",
       "84                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Eleven individuals underwent developmental psychological evaluations, consisting of at least the Vineland Adaptive Behavior Scales, 2nd edition. Cognitive function was assessed with testing specific for age and developmental level that provided either an intelligence quotient (IQ) or developmental quotient (DQ) score.  \n",
       "85                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The Nijmegen pediatric CDG rating scale, a measure of clinical disease progression developed for CDG, was applied to all affected individuals younger than 18 years.  \n",
       "131                                                                                                                                                                                                                                                                              Compared to the reference population, N-acetylaspartylglutamate + N-acetylaspartate (NAA) was lower than normal in the left centrum semiovale (LCSO) (p=0.004), the midline parietal grey matter (PGM) (p=0.02), and superior cerebellar vermis (SVERM) (p<0.0001). There was a deficit of glutamine + glutamate + gamma-aminobutyric acid (Glx) in the PGM (p=0.03), LCSO (p=0.01), and pons (p=0.0002). Choline was higher than expected for age only in the LCSO (p=0.0097), and myo-inositol was higher than expected for age in the pons (p=0.002). Multiple correlations between these MRS-measured metabolites and age, functional assessments, brain volume, and neurotransmitters in the CSF were found. The general trend showed that the differences noted above became more pronounced with increasing age, worsening function, and lower brain volume. MRS metabolite measurements did not correlate with total CSF protein, CSF albumin, or CSF/serum albumin ratio. There was a weak correlation (p=0.09) between atrophy and total CSF protein, but not CSF albumin or CSF/serum albumin ratio.  \n",
       "169                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Strong correlation between brain atrophy on MRI and functional assessments suggests that loss of neurons contributes to the functional impairment. The atrophy also correlated with CSF metabolites (BH4, 5-HIAA, HVA), which are known to be lower when there is damage to neurotransmitter producing neurons. This suggests that these biochemical abnormalities may be secondary to brain atrophy.  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patients_patch = (\n",
    "    patients_raw.dropna(subset='patient_id,external_study,details')\n",
    "    ['patient_id,external_study,details'].str.split(',', n=2, expand=True)\n",
    "    .rename(columns={0: 'patient_id', 1: 'external_study', 2: 'details'})\n",
    ")\n",
    "patients_patch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 180 entries, 0 to 179\n",
      "Data columns (total 5 columns):\n",
      " #   Column          Non-Null Count  Dtype \n",
      "---  ------          --------------  ----- \n",
      " 0   patient_id      180 non-null    object\n",
      " 1   external_study  21 non-null     object\n",
      " 2   details         180 non-null    object\n",
      " 3   doc_id          173 non-null    object\n",
      " 4   doc_filename    173 non-null    object\n",
      "dtypes: object(5)\n",
      "memory usage: 7.2+ KB\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>external_study</th>\n",
       "      <th>details</th>\n",
       "      <th>doc_id</th>\n",
       "      <th>doc_filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>All patients had global developmental delay, a movement disorder, and hypotonia.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7 out of 8 patients had hypo- or alacrima.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6 out of 7 patients had elevated liver transaminases.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6 out of 8 patients had microcephaly.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6 out of 8 patients had diminished reflexes.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 out of 6 patients had hepatocyte cytoplasmic storage material or vacuolization.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SOME</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4 out of 8 patients had seizures.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The most common deleterious allele was the nonsense mutation c.1201A&gt;T (p.R401X).</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NGLY1 deficiency is a novel autosomal recessive disorder of the ERAD pathway associated with neurological dysfunction, abnormal tear production, and liver disease. The majority of patients detected to date carry a specific nonsense mutation that appears to be associated with severe disease.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>Need et al.</td>\n",
       "      <td>3-year-old boy with compound heterozygous inactivating mutations in NGLY1. Clinical phenotype suggestive of a congenital disorder of glycosylation, although repeated transferrin isoelectric focusing and N-glycan analyses were normal. Liver biopsy showed accumulation of an amorphous unidentified substance throughout the cytoplasm, a finding likely consistent with NGLY1 dysfunction, which would be expected to result in abnormal accumulation of misfolded glycoproteins because of impaired cytosolic degradation.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Seven newly diagnosed patients with mutations in NGLY1. These observations confirm NGLY1 deficiency as an inherited disorder associated with the ERAD process and document its clinical presentation.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>Duke University</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Patient 5</td>\n",
       "      <td>University of British Columbia</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>University of British Columbia</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>The patient and parents were sequenced using both Illumina HiSeq2000 and Complete Genomics platforms. Variants in Illumina-sequenced reads were called using both the Hugeseq and Real Time Genomics pipelines and Complete Genomics variants were identified by their own variant callers.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient_id                  external_study  \\\n",
       "0         ALL                             NaN   \n",
       "1        SOME                             NaN   \n",
       "2        SOME                             NaN   \n",
       "3        SOME                             NaN   \n",
       "4        SOME                             NaN   \n",
       "5        SOME                             NaN   \n",
       "6        SOME                             NaN   \n",
       "7         ALL                             NaN   \n",
       "8         ALL                             NaN   \n",
       "9   Patient 1                     Need et al.   \n",
       "10        ALL                             NaN   \n",
       "11  Patient 1                 Duke University   \n",
       "12  Patient 5  University of British Columbia   \n",
       "13  Patient 6  University of British Columbia   \n",
       "14  Patient 3             Stanford University   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             details  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                   All patients had global developmental delay, a movement disorder, and hypotonia.   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         7 out of 8 patients had hypo- or alacrima.   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                              6 out of 7 patients had elevated liver transaminases.   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              6 out of 8 patients had microcephaly.   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       6 out of 8 patients had diminished reflexes.   \n",
       "5                                                                                                                                                                                                                                                                                                                                                                                                                                                  5 out of 6 patients had hepatocyte cytoplasmic storage material or vacuolization.   \n",
       "6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  4 out of 8 patients had seizures.   \n",
       "7                                                                                                                                                                                                                                                                                                                                                                                                                                                  The most common deleterious allele was the nonsense mutation c.1201A>T (p.R401X).   \n",
       "8                                                                                                                                                                                                                                NGLY1 deficiency is a novel autosomal recessive disorder of the ERAD pathway associated with neurological dysfunction, abnormal tear production, and liver disease. The majority of patients detected to date carry a specific nonsense mutation that appears to be associated with severe disease.   \n",
       "9   3-year-old boy with compound heterozygous inactivating mutations in NGLY1. Clinical phenotype suggestive of a congenital disorder of glycosylation, although repeated transferrin isoelectric focusing and N-glycan analyses were normal. Liver biopsy showed accumulation of an amorphous unidentified substance throughout the cytoplasm, a finding likely consistent with NGLY1 dysfunction, which would be expected to result in abnormal accumulation of misfolded glycoproteins because of impaired cytosolic degradation.   \n",
       "10                                                                                                                                                                                                                                                                                                                             Seven newly diagnosed patients with mutations in NGLY1. These observations confirm NGLY1 deficiency as an inherited disorder associated with the ERAD process and document its clinical presentation.   \n",
       "11                                                                                                                                                                                                                                                                                                                                                                                                         Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "12                                                                                                                                                                                                                                                                                                                                                                                                         Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "13                                                                                                                                                                                                                                                                                                                                                                                                         Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "14                                                                                                                                                                                                                                       The patient and parents were sequenced using both Illumina HiSeq2000 and Complete Genomics platforms. Variants in Illumina-sequenced reads were called using both the Hugeseq and Real Time Genomics pipelines and Complete Genomics variants were identified by their own variant callers.   \n",
       "\n",
       "        doc_id    doc_filename  \n",
       "0   PMC4243708  PMC4243708.txt  \n",
       "1   PMC4243708  PMC4243708.txt  \n",
       "2   PMC4243708  PMC4243708.txt  \n",
       "3   PMC4243708  PMC4243708.txt  \n",
       "4   PMC4243708  PMC4243708.txt  \n",
       "5   PMC4243708  PMC4243708.txt  \n",
       "6   PMC4243708  PMC4243708.txt  \n",
       "7   PMC4243708  PMC4243708.txt  \n",
       "8   PMC4243708  PMC4243708.txt  \n",
       "9   PMC4243708  PMC4243708.txt  \n",
       "10  PMC4243708  PMC4243708.txt  \n",
       "11  PMC4243708  PMC4243708.txt  \n",
       "12  PMC4243708  PMC4243708.txt  \n",
       "13  PMC4243708  PMC4243708.txt  \n",
       "14  PMC4243708  PMC4243708.txt  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patients = (\n",
    "    pd.concat([\n",
    "        patients_raw.pipe(lambda df: df[df['patient_id,external_study,details'].isnull()]).drop(columns='patient_id,external_study,details'),\n",
    "        (\n",
    "            patients_raw.dropna(subset='patient_id,external_study,details')\n",
    "            ['patient_id,external_study,details'].str.split(',', n=2, expand=True)\n",
    "            .rename(columns={0: 'patient_id', 1: 'external_study', 2: 'details'})\n",
    "        )\n",
    "    ], ignore_index=True)\n",
    ")\n",
    "assert len(patients_raw) == len(patients)\n",
    "path = utils.get_paths().output_data / \"patients_1.1.tsv\"\n",
    "if not path.exists():\n",
    "    patients.to_csv(path, sep=\"\\t\", index=False)\n",
    "patients.info()\n",
    "patients.head(15)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doc_id</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>ALL</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>SOME</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>SOME</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>ALL</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 3</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 4</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 7</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 2</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 1</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 8</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>#6</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>---</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>Index Case</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>#2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>#10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>#1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patients 5 and 6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 3's mother</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>Patient 3's father</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>#11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        doc_id          patient_id  count\n",
       "0   PMC7477955                 ALL     55\n",
       "1   PMC7477955                SOME     13\n",
       "2   PMC4243708                SOME     13\n",
       "3   PMC4243708                 ALL     12\n",
       "4   PMC4243708           Patient 3      7\n",
       "5   PMC4243708           Patient 6      6\n",
       "6   PMC4243708           Patient 4      5\n",
       "7   PMC4243708           Patient 5      5\n",
       "8   PMC4243708           Patient 7      5\n",
       "9   PMC4243708           Patient 2      5\n",
       "10  PMC4243708           Patient 1      5\n",
       "11  PMC4243708           Patient 8      4\n",
       "12  PMC7477955                   8      3\n",
       "13  PMC7477955                   7      3\n",
       "14  PMC7477955                  #6      3\n",
       "15  PMC7477955                   9      2\n",
       "16  PMC7477955                  12      2\n",
       "17  PMC7477955                   6      2\n",
       "18  PMC7477955                 ---      2\n",
       "19  PMC7477955                   3      2\n",
       "20  PMC7477955                   2      2\n",
       "21  PMC7477955                  11      2\n",
       "22  PMC7477955          Index Case      1\n",
       "23  PMC7477955                   5      1\n",
       "24  PMC7477955                   4      1\n",
       "25  PMC4243708                   1      1\n",
       "26  PMC7477955                  10      1\n",
       "27  PMC7477955                   1      1\n",
       "28  PMC7477955                  #2      1\n",
       "29  PMC4243708                   2      1\n",
       "30  PMC7477955                 #10      1\n",
       "31  PMC7477955                  #1      1\n",
       "32  PMC4243708    Patients 5 and 6      1\n",
       "33  PMC4243708  Patient 3's mother      1\n",
       "34  PMC4243708  Patient 3's father      1\n",
       "35  PMC4243708                   3      1\n",
       "36  PMC7477955                 #11      1"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    patients[['doc_id', 'patient_id']]\n",
    "    .value_counts()\n",
    "    .reset_index()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>external_study</th>\n",
       "      <th>details</th>\n",
       "      <th>doc_id</th>\n",
       "      <th>doc_filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>Need et al.</td>\n",
       "      <td>3-year-old boy with compound heterozygous inactivating mutations in NGLY1. Clinical phenotype suggestive of a congenital disorder of glycosylation, although repeated transferrin isoelectric focusing and N-glycan analyses were normal. Liver biopsy showed accumulation of an amorphous unidentified substance throughout the cytoplasm, a finding likely consistent with NGLY1 dysfunction, which would be expected to result in abnormal accumulation of misfolded glycoproteins because of impaired cytosolic degradation.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>Duke University</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Patient 5</td>\n",
       "      <td>University of British Columbia</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>University of British Columbia</td>\n",
       "      <td>Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>Stanford University</td>\n",
       "      <td>The patient and parents were sequenced using both Illumina HiSeq2000 and Complete Genomics platforms. Variants in Illumina-sequenced reads were called using both the Hugeseq and Real Time Genomics pipelines and Complete Genomics variants were identified by their own variant callers.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>Baylor College of Medicine</td>\n",
       "      <td>DNA was capture- sequenced using a commercially developed capture reagent (VCRome2). Sequence data were generated on an Illumina HiSeq2000 producing an average coverage of 80× with &gt;90% of targeted bases at 20× coverage or higher.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Patient 2</td>\n",
       "      <td>Baylor College of Medicine Whole Genome Laboratory, Houston, Texas</td>\n",
       "      <td>WES was performed on a clinical basis.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>Emory Genetics Laboratory, Atlanta, Georgia</td>\n",
       "      <td>WES was performed on a clinical basis.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Patient 7</td>\n",
       "      <td>Emory Genetics Laboratory, Atlanta, Georgia</td>\n",
       "      <td>WES was performed on a clinical basis.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Patient 4</td>\n",
       "      <td>Duke University</td>\n",
       "      <td>Sanger sequencing of NGLY1 was performed and results were confirmed by a clinical laboratory (GeneDx, Gaithersburg, Maryland).</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 5 years, Gender: Male, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.C1891del (p.Q631S)/c.1201A&gt;T(p.R401X), IUGR: No, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: No, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: No, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: Yes, Strabismus: No, ABR abnormalities: No, Lactic acidosis: No, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: Yes, Liver fibrosis: Yes, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: No, Scoliosis: No, Small hands/feet: Yes, Peripheral neuropathy: Yes</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Patient 2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 20 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: Yes, Mutations: c.1370dupG(p.R458fs)/c.1370dupG(p.R458fs), IUGR: Yes, Brain imaging abnormalities: No, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: No, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: No, Strabismus: No, ABR abnormalities: No, Lactic acidosis: Yes, Neonatal jaundice: No, Elevated liver transaminases: Yes, Elevated AFP: No, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: No, Scoliosis: Yes, Small hands/feet: No, Peripheral neuropathy: Yes</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 4 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1205_1207del(p.402_403del)/c.1570C&gt;(p. R524X), IUGR: No, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: No, Seizures: No, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: No, Chalazions: Yes, Strabismus: Yes, ABR abnormalities: Yes, Lactic acidosis: Yes, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: No, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: No, Scoliosis: No, Small hands/feet: Yes, Peripheral neuropathy: Not Determined</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Patient 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 2 years, Gender: Male, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A&gt;T(p. R401X)c.1201A&gt;T(pR401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: No, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: No, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: Yes, Strabismus: Yes, ABR abnormalities: Yes, Lactic acidosis: Yes, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: Yes, Liver fibrosis: Yes, Liver storage or vacuolization: No, Constipation: Yes, Dysmorphic features: No, Scoliosis: Yes, Small hands/feet: Yes, Peripheral neuropathy: Not Determined</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Patient 5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 0.5 years, Gender: Male, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A&gt;T(p. R401X)/c.1201A&gt;T(p.R401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: No, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: No, Chalazions: No, Strabismus: No, ABR abnormalities: No, Lactic acidosis: No, Neonatal jaundice: No, Elevated liver transaminases: Yes, Elevated AFP: Yes, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: Yes, Scoliosis: Yes, Small hands/feet: No, Peripheral neuropathy: Yes</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 0.9 months, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A&gt;T(p. R401X)/c.1201A&gt;T(p.R401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: No, Seizures: No, Ocular apraxia: No, Alacrima/hypolacrima: No, Corneal ulcerations/scarring: No, Chalazions: No, Strabismus: No, ABR abnormalities: Not Determined, Lactic acidosis: Not Determined, Neonatal jaundice: No, Elevated liver transaminases: Not Determined, Elevated AFP: Not Determined, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: No, Dysmorphic features: Yes, Scoliosis: No, Small hands/feet: No, Peripheral neuropathy: Not Determined</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Patient 7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 3 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A&gt;Y(p. R401X)/c.1201A&gt;T(p.R401X), IUGR: No, Brain imaging abnormalities: No, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: No, Decreased DTRs: Yes, Seizures: No, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: No, Chalazions: Yes, Strabismus: Yes, ABR abnormalities: Not Determined, Lactic acidosis: Yes, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: Not Determined, Liver fibrosis: Not Determined, Liver storage or vacuolization: Not Determined, Constipation: Yes, Dysmorphic features: Yes, Scoliosis: No, Small hands/feet: No, Peripheral neuropathy: Not Determined</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Patient 8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Age: 16 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c1201A&gt;T(p. R401X)/c.1201A&gt;T(p.R401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: No, Strabismus: Yes, ABR abnormalities: Not Determined, Lactic acidosis: Not Determined, Neonatal jaundice: No, Elevated liver transaminases: No, Elevated AFP: Not Determined, Liver fibrosis: Not Determined, Liver storage or vacuolization: Not Determined, Constipation: Yes, Dysmorphic features: Yes, Scoliosis: Yes, Small hands/feet: Yes, Peripheral neuropathy: Not Determined</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5-year-old male, presented in the neonatal period with involuntary movements, including athetosis involving the trunk and extremities and constant lip smacking and pursing while awake. Pregnancy and birth history were unremarkable. He had mild neonatal jaundice requiring phototherapy, but otherwise appeared well. Global developmental delay, hypotonia, intractable multifocal epilepsy, consisting of myoclonic seizures, drop attacks, and staring or tonic episodes, and liver disease were present in infancy. He has cortical vision loss and congenital alacrima and corneal ulcerations with scarring were noted at age 4 years. Now, at age 5 years, the movement disorder has not abated and he has central hypotonia and global developmental delay.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Patient 2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20-year-old female, born at 39 weeks of gestation via Cesarean section due to intrauterine growth retardation and an abnormal appearing placenta. At four months of age, hypotonia, developmental delay and elevated liver transaminases were noted. At approximately 4 years of age, a slight intention tremor and frequent involuntary movements of her neck, hands and arm were observed. At 5 years of age, she was noted to have ocular apraxia, distal tapering of hands and feet, and diminished deep tendon reflexes. She has cortical vision impairment, as well as alacrima and dry eyes that require lubrication, but has not developed corneal scarring. Presently, she has marked intellectual disabilities and requires total care. She has very little expressive speech and communicates through an electronic speech-generating device. She continues to ambulate with a walker.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4-year-old girl, born via Cesarean section at term for a non-reassuring fetal heart tracing, monitored in the NICU due to poor feeding and lethargy. Pregnancy was complicated by a positive second trimester screen noting increased risk for Smith-Lemli Opitz syndrome (SLOS) and trisomy 18, but karyotype on amniocentesis was normal. As a neonate, she had hyperbilirubinemia treated with phototherapy, elevated liver transaminases and transient thrombocytopenia. In infancy, she had global developmental delay, acquired microcephaly, bilateral exotropia, hypotonia, constipation, and intermittent mild lactic acidemia. At age 1 year, she did not make tears when crying, but had adequate tear production to keep her eyes moist. She had intermittent chalazions, but no corneal scarring. She developed staring spells, lasting up to 15 seconds, at approximately age 1 year; these episodes occur about once daily and can be interrupted by gentle contact. By age 17 months she had developed an extrapyramidal movement disorder consisting of asynchronous myoclonic jerks of the limbs and shoulders and subtle choreoathetotic movements of the hands and fingers. At 4 years she can ambulate unassisted, although her gait is unsteady, and communicates with vocalizations, gestures and use of a speech-generating device.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Patient 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2-year-old boy, delivered by Cesarean section at 38 weeks of gestation after fetal distress was noted on cardiotocography. Pregnancy history was positive for intrauterine growth restriction (IUGR) and oligohydramnios. Had mild hyperbilirubinemia, but otherwise his neonatal course was unremarkable and he was discharged on day of life three. Intermittent head flexion was noted at 6 months, and an EEG at 8 months showed generalized poly-spike discharges. Mild tonic seizures with head and body flexion started soon thereafter, and evolved to single, symmetric spasms with bilateral arm extension. Involuntary movements of the upper extremities were also noted at this time. In addition, global developmental delay, bilateral ptosis, abnormal tear production, elevated liver transaminases (3 to 4 times upper limit of normal), and constipation were noted in infancy. He has had recurrent episodes of keratoconjunctivitis and poor lid closure during sleep with resultant corneal scarring.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Patient 5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A boy who died at the age of 5 years, was born at term following a pregnancy that was complicated by a positive second trimester serum screening for trisomy 18 and Smith- Lemli-Opitz syndrome (SLOS) (AFP 1.97 MoM, uE3 0.24 MoM and hCG 0.48 MoM). Cytogenetic analysis of cultured amniocytes showed a normal male karyotype and measurement of 7-dehydrocholesterol in amniotic fluid excluded SLOS. He was delivered by Cesarean section at 36 weeks of gestation due to concerns for IUGR and a non-reassuring stress test. He had mild flexion contractures of both knees, but had an uneventful neonatal period. He had global developmental delay and constant movements of his arms and legs since early infancy and developed head bobbing at 7 months. At 8 months, liver transaminase elevations (approximately 1.5 times the upper limit of normal) were noted, and the elevations persisted until age 3 1⁄2 years. His reflexes appeared normal in infancy, but were diminished by age 2 years and at 38 months could no longer be elicited. During the second year of life, he was noted to have dry eyes that were treated with lubricant drops at bedtime, and microcephaly was present by 16 months. At 2 1⁄2 years, he developed myoclonic seizures that became intractable despite numerous therapeutic trials. Between the ages of 10 months and five years, he showed slow developmental progress, but regressed during the last year. He died at age 5 years following a viral illness and a prolonged seizure.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Younger sister of Subject 5. Pregnancy was complicated by a positive second trimester serum screen for trisomy 18 and SLOS (AFP 0.87 MoM, uE3 0.31 MoM, hCG 0.57 MoM). Cytogenetic analysis of cultured amniocytes showed a normal female karyotype. Delivered by Cesarean section at 35 weeks of gestation following an ultrasound that was concerning for IUGR. Had jaundice requiring phototherapy, but her course in the nursery was otherwise uneventful. By age 9 months, developmental delay was apparent and she had developed hypotonia, microcephaly, a mildly myopathic- appearing face and constant involuntary movements with a tendency to hyperextend her arms and close her fists. At 9 1⁄2 months of age, she died unexpectedly in her sleep and the cause of death remains unknown.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Patient 7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3-year-old girl, born at term via scheduled Cesarean section, had mild jaundice at birth, prenatal screen showed an increased risk for trisomy 21. In infancy, she had strabismus, hypotonia, athetoid arm and hand movements, clasped hands, and elevated liver transaminases (approximately 3 times upper limit of normal). Microcephaly was present by age 2 years. Severe developmental delay was present, but she has shown slow progress without regression. She learned to sit alone and started to crawl at 2 years, but had no words.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Patient 8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16-year-old girl, older sister of Subject 7, born at term via Cesarean section for decreased fetal movement and bradycardia. Prenatal history was significant for IUGR. Had cyanosis at birth and required resuscitation. Anal stenosis requiring dilatation was noted in the first week of life. Noted to have hypotonia, a movement disorder consisting of head bobbing and extremity athetosis, and developmental delay in early infancy. At 3 years, she could stand unsupported, ambulate with a walker, and started to babble. Seizures developed at age 11 years and she lost all mobility. Other medical history includes acquired microcephaly, an inability to close her eyes completely during sleep, chronic conjunctivitis, corneal clouding, hypolacrima, strabismus, hearing impairment, gastro-esophageal reflux, chronic constipation, severe scoliosis and talipes equinovarus.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Whole exome sequencing (WES) performed as part of a research protocol detected putative knock out mutations forming a compound heterozygote genotype in the NGLY1 gene (Maternal frameshift: Q631S. at cDNA level: C1891del in transcript ENST00000280700. EXON 12. Paternal nonsense: 3_25750426_A, which causes a nonsense mutation, R401X, in transcript ENST00000280700. At the cDNA level this is A1201T EXON 8)</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Did not carry the c.1201A&gt;T (p.R401X) mutation and their clinical phenotype was relatively mild in comparison</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Did not carry the c.1201A&gt;T (p.R401X) mutation and their clinical phenotype was relatively mild in comparison</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Patient 2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WES (Baylor College of Medicine Whole Genome Laboratory) revealed a homozygous mutation in exon 9 of the NGLY1 gene denoted as c.1370dupG or p.R458fs. Both parents were confirmed to be heterozygous carriers by Sanger sequencing. The mutation causes a frame shift in codon 458, causing insertion of 13 incorrect residues before a stop codon is introduced towards the end of exon 9. The mutation was not seen in any of 3321 other subjects sequenced at Duke, nor was it seen in 6503 subjects on the Exome Variant Server (NHLBI GO Exome Sequencing Project (ESP), Seattle, WA.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WES and whole-genome sequencing were performed using research protocols at Baylor College of Medicine and Stanford University. Mutations in NGLY1 that followed a compound heterozygous inheritance pattern were identified. A stop gain mutation caused by a G&gt;A mutation at position 3:25761670 (hg19) resulting in p.R542X was identified in both the father and daughter. A 3 base pair in-frame deletion TCC&gt; beginning at position 3:25775416 (hg19) was identified in both the mother and daughter. An additional G&gt;T mutation resulting in a heterozygous SMP at position 3:25777564 was identified in the daughter, mother and father. This mutation was not previously observed in 1000 genomes and is a coding region; however, it is present in heterozygous form in all three individuals.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Patient 3's father</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A stop gain mutation caused by a G&gt;A mutation at position 3:25761670 (hg19) resulting in p.R542X was identified in both the father and daughter. An additional G&gt;T mutation resulting in a heterozygous SMP at position 3:25777564 was identified in the daughter, mother and father. This mutation was not previously observed in 1000 genomes and is a coding region; however, it is present in heterozygous form in all three individuals.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Patient 3's mother</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A 3 base pair in-frame deletion TCC&gt; beginning at position 3:25775416 (hg19) was identified in both the mother and daughter. An additional G&gt;T mutation resulting in a heterozygous SMP at position 3:25777564 was identified in the daughter, mother and father. This mutation was not previously observed in 1000 genomes and is a coding region; however, it is present in heterozygous form in all three individuals.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Patient 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sanger sequencing (Duke University) detected a homozygous nonsense mutation, p.R401X, at position 3:25775422 (hg19) in transcript ENST00000280700. At the cDNA level this is c.1201A&gt;T in exon 8 of NGLY1. This finding was confirmed in a CLIA- certified laboratory (GeneDx).</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Patient 5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The patient has a homozygous variant in NGLY1, specifically a single nucleotide variant T&gt; A at position 3:25775422 (hg19). This variant was independently validated by Sanger sequencing. The variant causes a nonsense mutation, R401X. The variant was also present in the mother’s exome as a heterozygous call.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The patient has a homozygous variant in NGLY1, specifically a single nucleotide variant T&gt; A at position 3:25775422 (hg19). This variant was independently validated by Sanger sequencing. The variant causes a nonsense mutation, R401X. The variant was also present in the mother’s exome as a heterozygous call.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Patient 7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WES detected a homozygous nonsense mutation, R401X, at position 3:25775422 (hg19) in transcript ENST00000280700. At the cDNA level this is c.1201A&gt;T in exon 8 of NGLY1.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Patient 8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WES detected a homozygous nonsense mutation, R401X, at position 3:25775422 (hg19) in transcript ENST00000280700. At the cDNA level this is c.1201A&gt;T in exon 8 of NGLY1.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A moderate reduction in mitochondrial DNA content was identified in a liver sample.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Patients 5 and 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Significant brain disease was noted on autopsy. They were found to have pathological changes consistent with hypoxic-ischemic encephalopathy (HIE). Neuronal loss and gliosis may occur in both HIE and mitochondrial disorders, but spongiform degeneration is more typically seen in the latter and was not present in our cases.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Patient 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homozygous for the nonsense mutation c.1201A&gt;T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Patient 5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homozygous for the nonsense mutation c.1201A&gt;T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Patient 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homozygous for the nonsense mutation c.1201A&gt;T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Patient 7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homozygous for the nonsense mutation c.1201A&gt;T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Patient 8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homozygous for the nonsense mutation c.1201A&gt;T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Patient 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Compound heterozygous for the nonsense mutation c.1201A&gt;T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Patient 2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Does not carry the c.1201A&gt;T (p.R401X) mutation and appears to have a relatively mild phenotype.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Does not carry the c.1201A&gt;T (p.R401X) mutation and appears to have a relatively mild phenotype.</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>Index Case</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3-year-old male with NGLY1 deficiency. Inherited a maternal c.1891delC (p.Q631S NM_018297.3) variant and a paternal c.1201A&gt;T (p.R401* NM_018297.3) variant. NGLY1 protein expression in leukocytes was undetectable compared to controls. Transferrin isoelectric focusing and N-glycan profiling were normal.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Subject 3 is a sibling of subject 11. Included in previous clinical publications.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Subject 11 is a sibling of subject 3. Included in previous clinical publications.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Subject 7 is a sibling of subject 8.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Subject 8 is a sibling of subject 7.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Included in previous clinical publications.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Included in previous clinical publications.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Included in previous clinical publications.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Included in previous clinical publications.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3 year old male with NGLY1 deficiency. Allele 1 mutation: c.953T&gt;C (p.L318P). Allele 2 mutation: c.1169G&gt;C (p.R390P). Nijmegen score: 14. IQ or DQ not determined. Vineland score: 62.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4 year old male with NGLY1 deficiency. Allele 1 mutation: c.1201A&gt;T (p.R401*). Allele 2 mutation: c.1201A&gt;T (p.R401*). Nijmegen score: 33. IQ or DQ: 25. Vineland score: 52.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4 year old female with NGLY1 deficiency. Allele 1 mutation: c.1201A&gt;T (p.R401*). Allele 2 mutation: c.1201A&gt;T (p.R401*). Nijmegen score: 34. IQ or DQ: 5. Vineland score: 40.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5 year old female with NGLY1 deficiency. Allele 1 mutation: c.931G&gt;A (p.E311K). Allele 2 mutation: c.730T&gt;C (p.W244R). Nijmegen score: 33. IQ or DQ not determined. Vineland score not determined.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6 year old male with NGLY1 deficiency. Allele 1 mutation: c.1604G&gt;A (p.W535*). Allele 2 mutation: c.1910delT (p.L637*). Nijmegen score: 25. IQ or DQ: 8. Vineland score: 43.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7 year old male with NGLY1 deficiency. Allele 1 mutation: c.1891delC (p.Q631S). Allele 2 mutation: c.1201A&gt;T (p.R401*). Nijmegen score: 36. IQ or DQ: 5. Vineland score: 37.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8 year old female with NGLY1 deficiency. Allele 1 mutation: c.622C&gt;T (p.Q208*). Allele 2 mutation: c.930C&gt;T (p.G310G - splice site). Nijmegen score: 10. IQ or DQ: 74. Vineland score: 98.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10 year old male with NGLY1 deficiency. Allele 1 mutation: c.622C&gt;T (p.Q208*). Allele 2 mutation: c.930C&gt;T (p.G310G - splice site). Nijmegen score: 9. IQ or DQ: 81. Vineland score: 94.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16 year old male with NGLY1 deficiency. Allele 1 mutation: c.347C&gt;G (p.S116*). Allele 2 mutation: c.881+5G&gt;T (IVS5+5G&gt;T). Nijmegen score: 32. IQ or DQ: 2. Vineland score: 28.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17 year old female with NGLY1 deficiency. Allele 1 mutation: c.1201A&gt;T (p.R401*). Allele 2 mutation: c.1201A&gt;T (p.R401*). Nijmegen score: 25. IQ or DQ: 16. Vineland score: 42.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18 year old female with NGLY1 deficiency. Allele 1 mutation: c.1201A&gt;T (p.R401*). Allele 2 mutation: c.1201A&gt;T (p.R401*). Nijmegen score: 52. IQ or DQ: 2. Vineland score: 24.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21 year old female with NGLY1 deficiency. Allele 1 mutation: c.1370dupG (p.R458Kfs*14). Allele 2 mutation: c.1370dupG (p.R458Kfs*14). Nijmegen score not determined. IQ or DQ not determined. Vineland score: 37.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The sibling with a private cryptic splice site mutation (c.930C&gt;T) and a private nonsense mutation (c.622C&gt;T) exhibited relatively mild impairment in all domains.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The sibling with a private cryptic splice site mutation (c.930C&gt;T) and a private nonsense mutation (c.622C&gt;T) exhibited relatively mild impairment in all domains.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>#6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>On overnight EEG, this individual had active seizures recorded.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>#1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Had slight cerebellar atrophy.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>#2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Had slight cerebellar atrophy.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>#6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Had slight cerebellar atrophy.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>#10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Had slight cerebellar atrophy.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>#11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In one teenager, follow-up imaging showed atrophy measurably worse after a 20-month interval (net loss of 34 cm3 relative to expected).</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>#6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A single individual (#6) had a repeat study at one year, showing progression of the neuropathy.</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient_id  \\\n",
       "9             Patient 1   \n",
       "11            Patient 1   \n",
       "12            Patient 5   \n",
       "13            Patient 6   \n",
       "14            Patient 3   \n",
       "15            Patient 3   \n",
       "16            Patient 2   \n",
       "17            Patient 6   \n",
       "18            Patient 7   \n",
       "19            Patient 4   \n",
       "28            Patient 1   \n",
       "29            Patient 2   \n",
       "30            Patient 3   \n",
       "31            Patient 4   \n",
       "32            Patient 5   \n",
       "33            Patient 6   \n",
       "34            Patient 7   \n",
       "35            Patient 8   \n",
       "36            Patient 1   \n",
       "37            Patient 2   \n",
       "38            Patient 3   \n",
       "39            Patient 4   \n",
       "40            Patient 5   \n",
       "41            Patient 6   \n",
       "42            Patient 7   \n",
       "43            Patient 8   \n",
       "44                    1   \n",
       "46                    2   \n",
       "47                    3   \n",
       "48            Patient 2   \n",
       "49            Patient 3   \n",
       "50   Patient 3's father   \n",
       "51   Patient 3's mother   \n",
       "52            Patient 4   \n",
       "53            Patient 5   \n",
       "54            Patient 6   \n",
       "55            Patient 7   \n",
       "56            Patient 8   \n",
       "60            Patient 3   \n",
       "61     Patients 5 and 6   \n",
       "64            Patient 4   \n",
       "65            Patient 5   \n",
       "66            Patient 6   \n",
       "67            Patient 7   \n",
       "68            Patient 8   \n",
       "69            Patient 1   \n",
       "70            Patient 2   \n",
       "71            Patient 3   \n",
       "76           Index Case   \n",
       "82                    3   \n",
       "83                   11   \n",
       "84                    7   \n",
       "85                    8   \n",
       "86                    2   \n",
       "87                    6   \n",
       "88                    9   \n",
       "89                   12   \n",
       "90                    1   \n",
       "91                    2   \n",
       "92                    3   \n",
       "93                    4   \n",
       "94                    5   \n",
       "95                    6   \n",
       "96                    7   \n",
       "97                    8   \n",
       "98                    9   \n",
       "99                   10   \n",
       "100                  11   \n",
       "101                  12   \n",
       "111                   7   \n",
       "112                   8   \n",
       "118                  #6   \n",
       "120                  #1   \n",
       "121                  #2   \n",
       "122                  #6   \n",
       "123                 #10   \n",
       "125                 #11   \n",
       "128                  #6   \n",
       "\n",
       "                                                         external_study  \\\n",
       "9                                                           Need et al.   \n",
       "11                                                      Duke University   \n",
       "12                                       University of British Columbia   \n",
       "13                                       University of British Columbia   \n",
       "14                                                  Stanford University   \n",
       "15                                           Baylor College of Medicine   \n",
       "16   Baylor College of Medicine Whole Genome Laboratory, Houston, Texas   \n",
       "17                          Emory Genetics Laboratory, Atlanta, Georgia   \n",
       "18                          Emory Genetics Laboratory, Atlanta, Georgia   \n",
       "19                                                      Duke University   \n",
       "28                                                                  NaN   \n",
       "29                                                                  NaN   \n",
       "30                                                                  NaN   \n",
       "31                                                                  NaN   \n",
       "32                                                                  NaN   \n",
       "33                                                                  NaN   \n",
       "34                                                                  NaN   \n",
       "35                                                                  NaN   \n",
       "36                                                                  NaN   \n",
       "37                                                                  NaN   \n",
       "38                                                                  NaN   \n",
       "39                                                                  NaN   \n",
       "40                                                                  NaN   \n",
       "41                                                                  NaN   \n",
       "42                                                                  NaN   \n",
       "43                                                                  NaN   \n",
       "44                                                                  NaN   \n",
       "46                                                                  NaN   \n",
       "47                                                                  NaN   \n",
       "48                                                                  NaN   \n",
       "49                                                                  NaN   \n",
       "50                                                                  NaN   \n",
       "51                                                                  NaN   \n",
       "52                                                                  NaN   \n",
       "53                                                                  NaN   \n",
       "54                                                                  NaN   \n",
       "55                                                                  NaN   \n",
       "56                                                                  NaN   \n",
       "60                                                                  NaN   \n",
       "61                                                                  NaN   \n",
       "64                                                                  NaN   \n",
       "65                                                                  NaN   \n",
       "66                                                                  NaN   \n",
       "67                                                                  NaN   \n",
       "68                                                                  NaN   \n",
       "69                                                                  NaN   \n",
       "70                                                                  NaN   \n",
       "71                                                                  NaN   \n",
       "76                                                                  NaN   \n",
       "82                                                                  NaN   \n",
       "83                                                                  NaN   \n",
       "84                                                                  NaN   \n",
       "85                                                                  NaN   \n",
       "86                                                                  NaN   \n",
       "87                                                                  NaN   \n",
       "88                                                                  NaN   \n",
       "89                                                                  NaN   \n",
       "90                                                                  NaN   \n",
       "91                                                                  NaN   \n",
       "92                                                                  NaN   \n",
       "93                                                                  NaN   \n",
       "94                                                                  NaN   \n",
       "95                                                                  NaN   \n",
       "96                                                                  NaN   \n",
       "97                                                                  NaN   \n",
       "98                                                                  NaN   \n",
       "99                                                                  NaN   \n",
       "100                                                                 NaN   \n",
       "101                                                                 NaN   \n",
       "111                                                                 NaN   \n",
       "112                                                                 NaN   \n",
       "118                                                                 NaN   \n",
       "120                                                                 NaN   \n",
       "121                                                                 NaN   \n",
       "122                                                                 NaN   \n",
       "123                                                                 NaN   \n",
       "125                                                                 NaN   \n",
       "128                                                                 NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      details  \\\n",
       "9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            3-year-old boy with compound heterozygous inactivating mutations in NGLY1. Clinical phenotype suggestive of a congenital disorder of glycosylation, although repeated transferrin isoelectric focusing and N-glycan analyses were normal. Liver biopsy showed accumulation of an amorphous unidentified substance throughout the cytoplasm, a finding likely consistent with NGLY1 dysfunction, which would be expected to result in abnormal accumulation of misfolded glycoproteins because of impaired cytosolic degradation.   \n",
       "11                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "12                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "13                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Exome sequencing was performed using the Illumina HiSeq2000 platform and the Agilent SureSelect Human All Exon 50 Mb Kit.   \n",
       "14                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                The patient and parents were sequenced using both Illumina HiSeq2000 and Complete Genomics platforms. Variants in Illumina-sequenced reads were called using both the Hugeseq and Real Time Genomics pipelines and Complete Genomics variants were identified by their own variant callers.   \n",
       "15                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     DNA was capture- sequenced using a commercially developed capture reagent (VCRome2). Sequence data were generated on an Illumina HiSeq2000 producing an average coverage of 80× with >90% of targeted bases at 20× coverage or higher.   \n",
       "16                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     WES was performed on a clinical basis.   \n",
       "17                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     WES was performed on a clinical basis.   \n",
       "18                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     WES was performed on a clinical basis.   \n",
       "19                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Sanger sequencing of NGLY1 was performed and results were confirmed by a clinical laboratory (GeneDx, Gaithersburg, Maryland).   \n",
       "28                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Age: 5 years, Gender: Male, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.C1891del (p.Q631S)/c.1201A>T(p.R401X), IUGR: No, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: No, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: No, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: Yes, Strabismus: No, ABR abnormalities: No, Lactic acidosis: No, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: Yes, Liver fibrosis: Yes, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: No, Scoliosis: No, Small hands/feet: Yes, Peripheral neuropathy: Yes   \n",
       "29                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Age: 20 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: Yes, Mutations: c.1370dupG(p.R458fs)/c.1370dupG(p.R458fs), IUGR: Yes, Brain imaging abnormalities: No, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: No, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: No, Strabismus: No, ABR abnormalities: No, Lactic acidosis: Yes, Neonatal jaundice: No, Elevated liver transaminases: Yes, Elevated AFP: No, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: No, Scoliosis: Yes, Small hands/feet: No, Peripheral neuropathy: Yes   \n",
       "30                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Age: 4 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1205_1207del(p.402_403del)/c.1570C>(p. R524X), IUGR: No, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: No, Seizures: No, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: No, Chalazions: Yes, Strabismus: Yes, ABR abnormalities: Yes, Lactic acidosis: Yes, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: No, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: No, Scoliosis: No, Small hands/feet: Yes, Peripheral neuropathy: Not Determined   \n",
       "31                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Age: 2 years, Gender: Male, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A>T(p. R401X)c.1201A>T(pR401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: No, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: No, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: Yes, Strabismus: Yes, ABR abnormalities: Yes, Lactic acidosis: Yes, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: Yes, Liver fibrosis: Yes, Liver storage or vacuolization: No, Constipation: Yes, Dysmorphic features: No, Scoliosis: Yes, Small hands/feet: Yes, Peripheral neuropathy: Not Determined   \n",
       "32                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Age: 0.5 years, Gender: Male, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A>T(p. R401X)/c.1201A>T(p.R401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: No, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: No, Chalazions: No, Strabismus: No, ABR abnormalities: No, Lactic acidosis: No, Neonatal jaundice: No, Elevated liver transaminases: Yes, Elevated AFP: Yes, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: Yes, Dysmorphic features: Yes, Scoliosis: Yes, Small hands/feet: No, Peripheral neuropathy: Yes   \n",
       "33                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Age: 0.9 months, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A>T(p. R401X)/c.1201A>T(p.R401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: No, Seizures: No, Ocular apraxia: No, Alacrima/hypolacrima: No, Corneal ulcerations/scarring: No, Chalazions: No, Strabismus: No, ABR abnormalities: Not Determined, Lactic acidosis: Not Determined, Neonatal jaundice: No, Elevated liver transaminases: Not Determined, Elevated AFP: Not Determined, Liver fibrosis: No, Liver storage or vacuolization: Yes, Constipation: No, Dysmorphic features: Yes, Scoliosis: No, Small hands/feet: No, Peripheral neuropathy: Not Determined   \n",
       "34                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Age: 3 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c.1201A>Y(p. R401X)/c.1201A>T(p.R401X), IUGR: No, Brain imaging abnormalities: No, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: No, Decreased DTRs: Yes, Seizures: No, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: No, Chalazions: Yes, Strabismus: Yes, ABR abnormalities: Not Determined, Lactic acidosis: Yes, Neonatal jaundice: Yes, Elevated liver transaminases: Yes, Elevated AFP: Not Determined, Liver fibrosis: Not Determined, Liver storage or vacuolization: Not Determined, Constipation: Yes, Dysmorphic features: Yes, Scoliosis: No, Small hands/feet: No, Peripheral neuropathy: Not Determined   \n",
       "35                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Age: 16 years, Gender: Female, Ethnicity: Caucasian, Consanguinity: No, Mutations: c1201A>T(p. R401X)/c.1201A>T(p.R401X), IUGR: Yes, Brain imaging abnormalities: Yes, Global developmental delay: Yes, Microcephaly: Yes, Hypotonia: Yes, Movement disorder: Yes, EEG abnormalities: Yes, Decreased DTRs: Yes, Seizures: Yes, Ocular apraxia: Yes, Alacrima/hypolacrima: Yes, Corneal ulcerations/scarring: Yes, Chalazions: No, Strabismus: Yes, ABR abnormalities: Not Determined, Lactic acidosis: Not Determined, Neonatal jaundice: No, Elevated liver transaminases: No, Elevated AFP: Not Determined, Liver fibrosis: Not Determined, Liver storage or vacuolization: Not Determined, Constipation: Yes, Dysmorphic features: Yes, Scoliosis: Yes, Small hands/feet: Yes, Peripheral neuropathy: Not Determined   \n",
       "36                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   5-year-old male, presented in the neonatal period with involuntary movements, including athetosis involving the trunk and extremities and constant lip smacking and pursing while awake. Pregnancy and birth history were unremarkable. He had mild neonatal jaundice requiring phototherapy, but otherwise appeared well. Global developmental delay, hypotonia, intractable multifocal epilepsy, consisting of myoclonic seizures, drop attacks, and staring or tonic episodes, and liver disease were present in infancy. He has cortical vision loss and congenital alacrima and corneal ulcerations with scarring were noted at age 4 years. Now, at age 5 years, the movement disorder has not abated and he has central hypotonia and global developmental delay.   \n",
       "37                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          20-year-old female, born at 39 weeks of gestation via Cesarean section due to intrauterine growth retardation and an abnormal appearing placenta. At four months of age, hypotonia, developmental delay and elevated liver transaminases were noted. At approximately 4 years of age, a slight intention tremor and frequent involuntary movements of her neck, hands and arm were observed. At 5 years of age, she was noted to have ocular apraxia, distal tapering of hands and feet, and diminished deep tendon reflexes. She has cortical vision impairment, as well as alacrima and dry eyes that require lubrication, but has not developed corneal scarring. Presently, she has marked intellectual disabilities and requires total care. She has very little expressive speech and communicates through an electronic speech-generating device. She continues to ambulate with a walker.   \n",
       "38                                                                                                                                                                                4-year-old girl, born via Cesarean section at term for a non-reassuring fetal heart tracing, monitored in the NICU due to poor feeding and lethargy. Pregnancy was complicated by a positive second trimester screen noting increased risk for Smith-Lemli Opitz syndrome (SLOS) and trisomy 18, but karyotype on amniocentesis was normal. As a neonate, she had hyperbilirubinemia treated with phototherapy, elevated liver transaminases and transient thrombocytopenia. In infancy, she had global developmental delay, acquired microcephaly, bilateral exotropia, hypotonia, constipation, and intermittent mild lactic acidemia. At age 1 year, she did not make tears when crying, but had adequate tear production to keep her eyes moist. She had intermittent chalazions, but no corneal scarring. She developed staring spells, lasting up to 15 seconds, at approximately age 1 year; these episodes occur about once daily and can be interrupted by gentle contact. By age 17 months she had developed an extrapyramidal movement disorder consisting of asynchronous myoclonic jerks of the limbs and shoulders and subtle choreoathetotic movements of the hands and fingers. At 4 years she can ambulate unassisted, although her gait is unsteady, and communicates with vocalizations, gestures and use of a speech-generating device.   \n",
       "39                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                2-year-old boy, delivered by Cesarean section at 38 weeks of gestation after fetal distress was noted on cardiotocography. Pregnancy history was positive for intrauterine growth restriction (IUGR) and oligohydramnios. Had mild hyperbilirubinemia, but otherwise his neonatal course was unremarkable and he was discharged on day of life three. Intermittent head flexion was noted at 6 months, and an EEG at 8 months showed generalized poly-spike discharges. Mild tonic seizures with head and body flexion started soon thereafter, and evolved to single, symmetric spasms with bilateral arm extension. Involuntary movements of the upper extremities were also noted at this time. In addition, global developmental delay, bilateral ptosis, abnormal tear production, elevated liver transaminases (3 to 4 times upper limit of normal), and constipation were noted in infancy. He has had recurrent episodes of keratoconjunctivitis and poor lid closure during sleep with resultant corneal scarring.   \n",
       "40   A boy who died at the age of 5 years, was born at term following a pregnancy that was complicated by a positive second trimester serum screening for trisomy 18 and Smith- Lemli-Opitz syndrome (SLOS) (AFP 1.97 MoM, uE3 0.24 MoM and hCG 0.48 MoM). Cytogenetic analysis of cultured amniocytes showed a normal male karyotype and measurement of 7-dehydrocholesterol in amniotic fluid excluded SLOS. He was delivered by Cesarean section at 36 weeks of gestation due to concerns for IUGR and a non-reassuring stress test. He had mild flexion contractures of both knees, but had an uneventful neonatal period. He had global developmental delay and constant movements of his arms and legs since early infancy and developed head bobbing at 7 months. At 8 months, liver transaminase elevations (approximately 1.5 times the upper limit of normal) were noted, and the elevations persisted until age 3 1⁄2 years. His reflexes appeared normal in infancy, but were diminished by age 2 years and at 38 months could no longer be elicited. During the second year of life, he was noted to have dry eyes that were treated with lubricant drops at bedtime, and microcephaly was present by 16 months. At 2 1⁄2 years, he developed myoclonic seizures that became intractable despite numerous therapeutic trials. Between the ages of 10 months and five years, he showed slow developmental progress, but regressed during the last year. He died at age 5 years following a viral illness and a prolonged seizure.   \n",
       "41                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Younger sister of Subject 5. Pregnancy was complicated by a positive second trimester serum screen for trisomy 18 and SLOS (AFP 0.87 MoM, uE3 0.31 MoM, hCG 0.57 MoM). Cytogenetic analysis of cultured amniocytes showed a normal female karyotype. Delivered by Cesarean section at 35 weeks of gestation following an ultrasound that was concerning for IUGR. Had jaundice requiring phototherapy, but her course in the nursery was otherwise uneventful. By age 9 months, developmental delay was apparent and she had developed hypotonia, microcephaly, a mildly myopathic- appearing face and constant involuntary movements with a tendency to hyperextend her arms and close her fists. At 9 1⁄2 months of age, she died unexpectedly in her sleep and the cause of death remains unknown.   \n",
       "42                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             3-year-old girl, born at term via scheduled Cesarean section, had mild jaundice at birth, prenatal screen showed an increased risk for trisomy 21. In infancy, she had strabismus, hypotonia, athetoid arm and hand movements, clasped hands, and elevated liver transaminases (approximately 3 times upper limit of normal). Microcephaly was present by age 2 years. Severe developmental delay was present, but she has shown slow progress without regression. She learned to sit alone and started to crawl at 2 years, but had no words.   \n",
       "43                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          16-year-old girl, older sister of Subject 7, born at term via Cesarean section for decreased fetal movement and bradycardia. Prenatal history was significant for IUGR. Had cyanosis at birth and required resuscitation. Anal stenosis requiring dilatation was noted in the first week of life. Noted to have hypotonia, a movement disorder consisting of head bobbing and extremity athetosis, and developmental delay in early infancy. At 3 years, she could stand unsupported, ambulate with a walker, and started to babble. Seizures developed at age 11 years and she lost all mobility. Other medical history includes acquired microcephaly, an inability to close her eyes completely during sleep, chronic conjunctivitis, corneal clouding, hypolacrima, strabismus, hearing impairment, gastro-esophageal reflux, chronic constipation, severe scoliosis and talipes equinovarus.   \n",
       "44                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Whole exome sequencing (WES) performed as part of a research protocol detected putative knock out mutations forming a compound heterozygote genotype in the NGLY1 gene (Maternal frameshift: Q631S. at cDNA level: C1891del in transcript ENST00000280700. EXON 12. Paternal nonsense: 3_25750426_A, which causes a nonsense mutation, R401X, in transcript ENST00000280700. At the cDNA level this is A1201T EXON 8)   \n",
       "46                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Did not carry the c.1201A>T (p.R401X) mutation and their clinical phenotype was relatively mild in comparison   \n",
       "47                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Did not carry the c.1201A>T (p.R401X) mutation and their clinical phenotype was relatively mild in comparison   \n",
       "48                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                WES (Baylor College of Medicine Whole Genome Laboratory) revealed a homozygous mutation in exon 9 of the NGLY1 gene denoted as c.1370dupG or p.R458fs. Both parents were confirmed to be heterozygous carriers by Sanger sequencing. The mutation causes a frame shift in codon 458, causing insertion of 13 incorrect residues before a stop codon is introduced towards the end of exon 9. The mutation was not seen in any of 3321 other subjects sequenced at Duke, nor was it seen in 6503 subjects on the Exome Variant Server (NHLBI GO Exome Sequencing Project (ESP), Seattle, WA.   \n",
       "49                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    WES and whole-genome sequencing were performed using research protocols at Baylor College of Medicine and Stanford University. Mutations in NGLY1 that followed a compound heterozygous inheritance pattern were identified. A stop gain mutation caused by a G>A mutation at position 3:25761670 (hg19) resulting in p.R542X was identified in both the father and daughter. A 3 base pair in-frame deletion TCC> beginning at position 3:25775416 (hg19) was identified in both the mother and daughter. An additional G>T mutation resulting in a heterozygous SMP at position 3:25777564 was identified in the daughter, mother and father. This mutation was not previously observed in 1000 genomes and is a coding region; however, it is present in heterozygous form in all three individuals.   \n",
       "50                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              A stop gain mutation caused by a G>A mutation at position 3:25761670 (hg19) resulting in p.R542X was identified in both the father and daughter. An additional G>T mutation resulting in a heterozygous SMP at position 3:25777564 was identified in the daughter, mother and father. This mutation was not previously observed in 1000 genomes and is a coding region; however, it is present in heterozygous form in all three individuals.   \n",
       "51                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  A 3 base pair in-frame deletion TCC> beginning at position 3:25775416 (hg19) was identified in both the mother and daughter. An additional G>T mutation resulting in a heterozygous SMP at position 3:25777564 was identified in the daughter, mother and father. This mutation was not previously observed in 1000 genomes and is a coding region; however, it is present in heterozygous form in all three individuals.   \n",
       "52                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Sanger sequencing (Duke University) detected a homozygous nonsense mutation, p.R401X, at position 3:25775422 (hg19) in transcript ENST00000280700. At the cDNA level this is c.1201A>T in exon 8 of NGLY1. This finding was confirmed in a CLIA- certified laboratory (GeneDx).   \n",
       "53                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The patient has a homozygous variant in NGLY1, specifically a single nucleotide variant T> A at position 3:25775422 (hg19). This variant was independently validated by Sanger sequencing. The variant causes a nonsense mutation, R401X. The variant was also present in the mother’s exome as a heterozygous call.   \n",
       "54                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The patient has a homozygous variant in NGLY1, specifically a single nucleotide variant T> A at position 3:25775422 (hg19). This variant was independently validated by Sanger sequencing. The variant causes a nonsense mutation, R401X. The variant was also present in the mother’s exome as a heterozygous call.   \n",
       "55                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   WES detected a homozygous nonsense mutation, R401X, at position 3:25775422 (hg19) in transcript ENST00000280700. At the cDNA level this is c.1201A>T in exon 8 of NGLY1.   \n",
       "56                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   WES detected a homozygous nonsense mutation, R401X, at position 3:25775422 (hg19) in transcript ENST00000280700. At the cDNA level this is c.1201A>T in exon 8 of NGLY1.   \n",
       "60                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        A moderate reduction in mitochondrial DNA content was identified in a liver sample.   \n",
       "61                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Significant brain disease was noted on autopsy. They were found to have pathological changes consistent with hypoxic-ischemic encephalopathy (HIE). Neuronal loss and gliosis may occur in both HIE and mitochondrial disorders, but spongiform degeneration is more typically seen in the latter and was not present in our cases.   \n",
       "64                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Homozygous for the nonsense mutation c.1201A>T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.   \n",
       "65                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Homozygous for the nonsense mutation c.1201A>T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.   \n",
       "66                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Homozygous for the nonsense mutation c.1201A>T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.   \n",
       "67                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Homozygous for the nonsense mutation c.1201A>T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.   \n",
       "68                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Homozygous for the nonsense mutation c.1201A>T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.   \n",
       "69                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Compound heterozygous for the nonsense mutation c.1201A>T (p.R401X) in NGLY1, associated with a severe phenotype. The range of outcomes is variable, with some patients having an early demise, and others living at least into their teenage years.   \n",
       "70                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Does not carry the c.1201A>T (p.R401X) mutation and appears to have a relatively mild phenotype.   \n",
       "71                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Does not carry the c.1201A>T (p.R401X) mutation and appears to have a relatively mild phenotype.   \n",
       "76                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            3-year-old male with NGLY1 deficiency. Inherited a maternal c.1891delC (p.Q631S NM_018297.3) variant and a paternal c.1201A>T (p.R401* NM_018297.3) variant. NGLY1 protein expression in leukocytes was undetectable compared to controls. Transferrin isoelectric focusing and N-glycan profiling were normal.   \n",
       "82                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Subject 3 is a sibling of subject 11. Included in previous clinical publications.   \n",
       "83                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Subject 11 is a sibling of subject 3. Included in previous clinical publications.   \n",
       "84                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Subject 7 is a sibling of subject 8.   \n",
       "85                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Subject 8 is a sibling of subject 7.   \n",
       "86                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Included in previous clinical publications.   \n",
       "87                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Included in previous clinical publications.   \n",
       "88                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Included in previous clinical publications.   \n",
       "89                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Included in previous clinical publications.   \n",
       "90                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     3 year old male with NGLY1 deficiency. Allele 1 mutation: c.953T>C (p.L318P). Allele 2 mutation: c.1169G>C (p.R390P). Nijmegen score: 14. IQ or DQ not determined. Vineland score: 62.   \n",
       "91                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               4 year old male with NGLY1 deficiency. Allele 1 mutation: c.1201A>T (p.R401*). Allele 2 mutation: c.1201A>T (p.R401*). Nijmegen score: 33. IQ or DQ: 25. Vineland score: 52.   \n",
       "92                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              4 year old female with NGLY1 deficiency. Allele 1 mutation: c.1201A>T (p.R401*). Allele 2 mutation: c.1201A>T (p.R401*). Nijmegen score: 34. IQ or DQ: 5. Vineland score: 40.   \n",
       "93                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         5 year old female with NGLY1 deficiency. Allele 1 mutation: c.931G>A (p.E311K). Allele 2 mutation: c.730T>C (p.W244R). Nijmegen score: 33. IQ or DQ not determined. Vineland score not determined.   \n",
       "94                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               6 year old male with NGLY1 deficiency. Allele 1 mutation: c.1604G>A (p.W535*). Allele 2 mutation: c.1910delT (p.L637*). Nijmegen score: 25. IQ or DQ: 8. Vineland score: 43.   \n",
       "95                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               7 year old male with NGLY1 deficiency. Allele 1 mutation: c.1891delC (p.Q631S). Allele 2 mutation: c.1201A>T (p.R401*). Nijmegen score: 36. IQ or DQ: 5. Vineland score: 37.   \n",
       "96                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 8 year old female with NGLY1 deficiency. Allele 1 mutation: c.622C>T (p.Q208*). Allele 2 mutation: c.930C>T (p.G310G - splice site). Nijmegen score: 10. IQ or DQ: 74. Vineland score: 98.   \n",
       "97                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   10 year old male with NGLY1 deficiency. Allele 1 mutation: c.622C>T (p.Q208*). Allele 2 mutation: c.930C>T (p.G310G - splice site). Nijmegen score: 9. IQ or DQ: 81. Vineland score: 94.   \n",
       "98                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             16 year old male with NGLY1 deficiency. Allele 1 mutation: c.347C>G (p.S116*). Allele 2 mutation: c.881+5G>T (IVS5+5G>T). Nijmegen score: 32. IQ or DQ: 2. Vineland score: 28.   \n",
       "99                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            17 year old female with NGLY1 deficiency. Allele 1 mutation: c.1201A>T (p.R401*). Allele 2 mutation: c.1201A>T (p.R401*). Nijmegen score: 25. IQ or DQ: 16. Vineland score: 42.   \n",
       "100                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            18 year old female with NGLY1 deficiency. Allele 1 mutation: c.1201A>T (p.R401*). Allele 2 mutation: c.1201A>T (p.R401*). Nijmegen score: 52. IQ or DQ: 2. Vineland score: 24.   \n",
       "101                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         21 year old female with NGLY1 deficiency. Allele 1 mutation: c.1370dupG (p.R458Kfs*14). Allele 2 mutation: c.1370dupG (p.R458Kfs*14). Nijmegen score not determined. IQ or DQ not determined. Vineland score: 37.   \n",
       "111                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        The sibling with a private cryptic splice site mutation (c.930C>T) and a private nonsense mutation (c.622C>T) exhibited relatively mild impairment in all domains.   \n",
       "112                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        The sibling with a private cryptic splice site mutation (c.930C>T) and a private nonsense mutation (c.622C>T) exhibited relatively mild impairment in all domains.   \n",
       "118                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           On overnight EEG, this individual had active seizures recorded.   \n",
       "120                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Had slight cerebellar atrophy.   \n",
       "121                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Had slight cerebellar atrophy.   \n",
       "122                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Had slight cerebellar atrophy.   \n",
       "123                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Had slight cerebellar atrophy.   \n",
       "125                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   In one teenager, follow-up imaging showed atrophy measurably worse after a 20-month interval (net loss of 34 cm3 relative to expected).   \n",
       "128                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           A single individual (#6) had a repeat study at one year, showing progression of the neuropathy.   \n",
       "\n",
       "         doc_id    doc_filename  \n",
       "9    PMC4243708  PMC4243708.txt  \n",
       "11   PMC4243708  PMC4243708.txt  \n",
       "12   PMC4243708  PMC4243708.txt  \n",
       "13   PMC4243708  PMC4243708.txt  \n",
       "14   PMC4243708  PMC4243708.txt  \n",
       "15   PMC4243708  PMC4243708.txt  \n",
       "16   PMC4243708  PMC4243708.txt  \n",
       "17   PMC4243708  PMC4243708.txt  \n",
       "18   PMC4243708  PMC4243708.txt  \n",
       "19   PMC4243708  PMC4243708.txt  \n",
       "28   PMC4243708  PMC4243708.txt  \n",
       "29   PMC4243708  PMC4243708.txt  \n",
       "30   PMC4243708  PMC4243708.txt  \n",
       "31   PMC4243708  PMC4243708.txt  \n",
       "32   PMC4243708  PMC4243708.txt  \n",
       "33   PMC4243708  PMC4243708.txt  \n",
       "34   PMC4243708  PMC4243708.txt  \n",
       "35   PMC4243708  PMC4243708.txt  \n",
       "36   PMC4243708  PMC4243708.txt  \n",
       "37   PMC4243708  PMC4243708.txt  \n",
       "38   PMC4243708  PMC4243708.txt  \n",
       "39   PMC4243708  PMC4243708.txt  \n",
       "40   PMC4243708  PMC4243708.txt  \n",
       "41   PMC4243708  PMC4243708.txt  \n",
       "42   PMC4243708  PMC4243708.txt  \n",
       "43   PMC4243708  PMC4243708.txt  \n",
       "44   PMC4243708  PMC4243708.txt  \n",
       "46   PMC4243708  PMC4243708.txt  \n",
       "47   PMC4243708  PMC4243708.txt  \n",
       "48   PMC4243708  PMC4243708.txt  \n",
       "49   PMC4243708  PMC4243708.txt  \n",
       "50   PMC4243708  PMC4243708.txt  \n",
       "51   PMC4243708  PMC4243708.txt  \n",
       "52   PMC4243708  PMC4243708.txt  \n",
       "53   PMC4243708  PMC4243708.txt  \n",
       "54   PMC4243708  PMC4243708.txt  \n",
       "55   PMC4243708  PMC4243708.txt  \n",
       "56   PMC4243708  PMC4243708.txt  \n",
       "60   PMC4243708  PMC4243708.txt  \n",
       "61   PMC4243708  PMC4243708.txt  \n",
       "64   PMC4243708  PMC4243708.txt  \n",
       "65   PMC4243708  PMC4243708.txt  \n",
       "66   PMC4243708  PMC4243708.txt  \n",
       "67   PMC4243708  PMC4243708.txt  \n",
       "68   PMC4243708  PMC4243708.txt  \n",
       "69   PMC4243708  PMC4243708.txt  \n",
       "70   PMC4243708  PMC4243708.txt  \n",
       "71   PMC4243708  PMC4243708.txt  \n",
       "76   PMC7477955  PMC7477955.txt  \n",
       "82   PMC7477955  PMC7477955.txt  \n",
       "83   PMC7477955  PMC7477955.txt  \n",
       "84   PMC7477955  PMC7477955.txt  \n",
       "85   PMC7477955  PMC7477955.txt  \n",
       "86   PMC7477955  PMC7477955.txt  \n",
       "87   PMC7477955  PMC7477955.txt  \n",
       "88   PMC7477955  PMC7477955.txt  \n",
       "89   PMC7477955  PMC7477955.txt  \n",
       "90   PMC7477955  PMC7477955.txt  \n",
       "91   PMC7477955  PMC7477955.txt  \n",
       "92   PMC7477955  PMC7477955.txt  \n",
       "93   PMC7477955  PMC7477955.txt  \n",
       "94   PMC7477955  PMC7477955.txt  \n",
       "95   PMC7477955  PMC7477955.txt  \n",
       "96   PMC7477955  PMC7477955.txt  \n",
       "97   PMC7477955  PMC7477955.txt  \n",
       "98   PMC7477955  PMC7477955.txt  \n",
       "99   PMC7477955  PMC7477955.txt  \n",
       "100  PMC7477955  PMC7477955.txt  \n",
       "101  PMC7477955  PMC7477955.txt  \n",
       "111  PMC7477955  PMC7477955.txt  \n",
       "112  PMC7477955  PMC7477955.txt  \n",
       "118  PMC7477955  PMC7477955.txt  \n",
       "120  PMC7477955  PMC7477955.txt  \n",
       "121  PMC7477955  PMC7477955.txt  \n",
       "122  PMC7477955  PMC7477955.txt  \n",
       "123  PMC7477955  PMC7477955.txt  \n",
       "125  PMC7477955  PMC7477955.txt  \n",
       "128  PMC7477955  PMC7477955.txt  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    patients\n",
    "    .pipe(lambda df: df[~df['patient_id'].isin(['ALL', 'SOME', '---'])])\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "text = \"\"\"\n",
    "Neurologic Phenotype\n",
    "Seven of twelve subjects had clinical seizures, and one had subclinical seizures recognized on previous EEG. Details regarding age of onset, seizure type and frequency, medications, and EEG findings are noted in Supplementary Table S. On overnight EEG, only one individual (#6) had active seizures recorded, but seven had multifocal epileptiform activity. There were no age or genotype differences between individuals having seizures and those without. In fact, in each sibling pair, one had seizures and the other did not.\n",
    "All twelve individuals exhibited hyperkinetic movement disorders that included choreiform, athetoid, dystonic, myoclonic, action tremor, and dysmetric movements and were more severe in the younger individuals (Supplementary Movie S1).\n",
    "\n",
    "Brain MRI and MRS\n",
    "Eleven individuals underwent MRI and MRS of the brain. Clinical assessment of the images was not striking (Figure 3). Delayed myelination was present in three of the four youngest individuals, but all the older individuals had complete myelination. Six of nine individuals had qualitatively-evident cerebral atrophy that ranged from slight to moderate. Four individuals (#1, #2, #6, #10) also had slight cerebellar atrophy. The atrophy tended to be greater in the older individuals (p=0.17, Supplementary Figure S3), and in one teenager (#11) follow-up imaging showed atrophy measurably worse after a 20-month interval (net loss of 34 cm3 relative to expected). Increased atrophy correlated with worsening of all functional measurements (Supplementary Figure S3), including IQ or DQ (p<0.03), Vineland assessments (p<0.03), and Nijmegen scores (p=0.01). Brain volume also directly correlated with CSF levels of 5-HIAA (p=0.03), tetrahydrobiopterin (p=0.02), and 5-HVA (p=0.06) (Supplementary Figure S3).\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "text = \"\"\"\n",
    "Table 1\n",
    "Clinical and molecular findings in NGLY1 deficiency\n",
    "Patient 1\tPatient 2\tPatient 3\tPatient 4\tPatient 5\tPatient 6\tPatient 7\tPatient 8\tTotals\n",
    "Age\t5 y\t20 y\t4 y\t2 y\td.5 y\td.9 m\t3 y\t16 y\t\n",
    "Gender\tM\tF\tF\tM\tM\tF\tF\tF\t\n",
    "Ethnicity\tCaucasian\tCaucasian\tCaucasian\tCaucasian\tCaucasian\tCaucasian\tCaucasian\tCaucasian\t\n",
    "Consanguinity\t−\t+\t−\t−\t−\t−\t−\t−\t1/8\n",
    "Mutations (maternal/paternal allele)\tc.C1891del (p.Q631S)/c.1201A>T(p.R401X)\tc.1370dupG(p.R458fs)/c.1370dupG(p.R458fs)\tc.1205_1207del(p.402_403del)/c.1570C>(p.R524X)\tc.1201A>T(p.R401X)c.1201A>T(pR401X)\tc.1201A>T(p.R401X)/c.1201A>T(p.R401X)\tc.1201A>T(p.R401X)/c.1201A>T(p.R401X)\tc.1201A>Y(p.R401X)/c.1201A>T(p.R401X)\tc1201A>T(p.R401X)/c.1201A>T(p.R401X)\t\n",
    "IUGR\t−\t+\t−\t+\t+\t+\t−\t+\t5/8\n",
    "Brain imaging abnormalities\t+a\t−b\t+c\t+d\t+e\t+f\t−\t+g\t6/8\n",
    "Global developmental delay\t+\t+\t+\t+\t+\t+\t+\t+\t8/8\n",
    "Microcephalyh\t−\t+\t+\t−\t+\t+\t+\t+\t6/8\n",
    "Hypotonia\t+\t+\t+\t+\t+\t+\t+\t+\t8/8\n",
    "Movement disorder\t+\t+\t+\t+\t+\t+\t+\t+\t8/8\n",
    "EEG abnormalities\t+\t+\t+\t+\t+\t+\t−\t+\t7/8\n",
    "↓DTRs\t+\t+\t−\t+\t+\t−\t+\t+\t6/8\n",
    "Seizures\t+\t−\t−\t+\t+\t−\t−\t+\t4/8\n",
    "Ocular apraxia\t−\t+\t+\t−\t−\t−\t+\t+\t4/8\n",
    "Alacrima/hypolacrima\t+\t+\t+\t+\t+\t−\t+\t+\t7/8\n",
    "Corneal ulcerations/scarring\t+\t+\t−\t+\t−\t−\t−\t+\t4/8\n",
    "Chalazions\t+\t−\t+\t+\t−\t−\t+\t−\t4/8\n",
    "Strabismus\t−\t−\t+\t+\t−\t−\t+\t+\t5/8\n",
    "ABR abnormalities\t−\t−\t+\t+\t−\tND\tND\tND\t2/5\n",
    "Lactic acidosis\t−\t+\t+\t+\t−i\tND\t+\tND\t4/6\n",
    "Neonatal jaundice\t+\t−\t+\t+\t−\t−\t+\t−\t4/8\n",
    "Elevated liver transaminases\t+\t+\t+\t+\t+\tND\t+\t−\t6/7\n",
    "Elevated AFP\t+\t−\t−j\t+\t+\tND\tND\tND\t3/5\n",
    "Liver fibrosis\t+\t−\t−\t+\t−\t−\tND\tND\t2/6\n",
    "Liver storage or vacuolization\t+\t+\t+\t−\t+k\t+l\tND\tND\t5/6\n",
    "Constipation\t+\t+\t+\t+\t+\t−\t+\t+\t7/8\n",
    "Dysmorphic features\t−\t−\t−\t−\t+m\t+n\t+o\t+p\t4/8\n",
    "Scoliosis\t−\t+\t−\t+\t+\t−\t−\t+\t4/8\n",
    "Small hands/feet\t+\t−\t+\t+\t−\t−\t−\t+\t4/8\n",
    "Peripheral neuropathyq\t+\t+\tND\tND\t+\tND\tND\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "text = \"\"\"\n",
    "Patient 2, a now 20-year-old female, was born at 39 weeks of gestation via Cesarean section because of intrauterine growth retardation and an abnormal appearing placenta. At four months of age, hypotonia, developmental delay and elevated liver transaminases were noted. At approximately 4 years of age, a slight intention tremor and frequent involuntary movements of her neck, hands and arm were observed. At 5 years of age, she was noted to have ocular apraxia, distal tapering of hands and feet, and diminished deep tendon reflexes. She has cortical vision impairment, as well as alacrima and dry eyes that require lubrication, but has not developed corneal scarring. Presently, she has marked intellectual disabilities and requires total care. She has very little expressive speech and communicates through an electronic speech-generating device. She continues to ambulate with a walker.\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency.  Extract details about the patients discussed in this text: \n",
      "\n",
      "--- BEGIN TEXT ---\n",
      "\n",
      "Patient 2, a now 20-year-old female, was born at 39 weeks of gestation via Cesarean section because of intrauterine growth retardation and an abnormal appearing placenta. At four months of age, hypotonia, developmental delay and elevated liver transaminases were noted. At approximately 4 years of age, a slight intention tremor and frequent involuntary movements of her neck, hands and arm were observed. At 5 years of age, she was noted to have ocular apraxia, distal tapering of hands and feet, and diminished deep tendon reflexes. She has cortical vision impairment, as well as alacrima and dry eyes that require lubrication, but has not developed corneal scarring. Presently, she has marked intellectual disabilities and requires total care. She has very little expressive speech and communicates through an electronic speech-generating device. She continues to ambulate with a walker.\n",
      "\n",
      "--- END TEXT ---\n",
      "\n",
      "Requirements:\n",
      "\n",
      "- Exclude any patients where context dictates that they do NOT have NGLY1 deficiency, e.g. when NGLY1 deficiency patients are compared to similar patients with other diseases.\n",
      "- Extract as much information as possible about each patient including associated genotypes, phenotypes, physical or behavioral traits, demographics, lab measurements, treatments, family histories or anything else of clinical and/or biological relevance.\n",
      "- Extract this information in CSV format with the following headers:\n",
      "  - `patient_id`: Identifying information for the patient within the context of the article\n",
      "    - When a single patient is unambiguously referenced by an identifier, this must be equal to that identifier; typically an integer or anonymized name like \"Patient 1\" or other arbitrary string \n",
      "    - When all patients are referenced without individual attribution, use the value \"ALL\" for this field\n",
      "  - `external_study`: If context implies that the patient being referenced is from another study, include any identifying information for that study/reference in this field (e.g. citation numbers or author names); if no such distinction is made, assume it is from the current study and report only the value \"NA\"\n",
      "  - `details`: A free-text description of the information discussed for the patient that includes as much detail as possible\n",
      "- If multiple patient identifiers are referenced at once, repeat the information for them on separate CSV lines with one of those identifiers used in each line (in `patient_id`)\n",
      "- Ensure that any `details` for a patient can definitively be ascribed to that specific patient\n",
      "- Report results using a pipe (i.e. \"|\") delimiter \n",
      "- Include the CSV headers in the result\n",
      "- If the text contains no patient information, report only the CSV headers\n",
      "- Do NOT include explanation of any kind in the results; report only the CSV content\n",
      "\n",
      "\n",
      "Patient details CSV:\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:ngly1_gpt.llm:Response:\n",
      "patient_id|external_study|details\n",
      "Patient 2|NA|20-year-old female, born at 39 weeks of gestation via Cesarean section due to intrauterine growth retardation and an abnormal appearing placenta. At four months of age, hypotonia, developmental delay and elevated liver transaminases were noted. At approximately 4 years of age, a slight intention tremor and frequent involuntary movements of her neck, hands and arm were observed. At 5 years of age, she was noted to have ocular apraxia, distal tapering of hands and feet, and diminished deep tendon reflexes. She has cortical vision impairment, as well as alacrima and dry eyes that require lubrication, but has not developed corneal scarring. Presently, she has marked intellectual disabilities and requires total care. She has very little expressive speech and communicates through an electronic speech-generating device. She continues to ambulate with a walker.\n",
      "patient_id|external_study|details\n",
      "Patient 2|NA|20-year-old female, born at 39 weeks of gestation via Cesarean section due to intrauterine growth retardation and an abnormal appearing placenta. At four months of age, hypotonia, developmental delay and elevated liver transaminases were noted. At approximately 4 years of age, a slight intention tremor and frequent involuntary movements of her neck, hands and arm were observed. At 5 years of age, she was noted to have ocular apraxia, distal tapering of hands and feet, and diminished deep tendon reflexes. She has cortical vision impairment, as well as alacrima and dry eyes that require lubrication, but has not developed corneal scarring. Presently, she has marked intellectual disabilities and requires total care. She has very little expressive speech and communicates through an electronic speech-generating device. She continues to ambulate with a walker.\n"
     ]
    }
   ],
   "source": [
    "response = llm.chat_completion(\"patient_extraction_1.txt\", model=\"gpt-4\", temperature=0.0, disease=utils.NGLY1_DEFICIENCY, text=text)\n",
    "print(response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "patient_id|external_study|details\n",
      "ALL|NA|Seven of twelve subjects had clinical seizures, and one had subclinical seizures recognized on previous EEG. On overnight EEG, only one individual had active seizures recorded, but seven had multifocal epileptiform activity. There were no age or genotype differences between individuals having seizures and those without. In each sibling pair, one had seizures and the other did not.\n",
      "ALL|NA|All twelve individuals exhibited hyperkinetic movement disorders that included choreiform, athetoid, dystonic, myoclonic, action tremor, and dysmetric movements and were more severe in the younger individuals.\n",
      "ALL|NA|Eleven individuals underwent MRI and MRS of the brain. Delayed myelination was present in three of the four youngest individuals, but all the older individuals had complete myelination. Six of nine individuals had qualitatively-evident cerebral atrophy that ranged from slight to moderate.\n",
      "#1|NA|Had slight cerebellar atrophy.\n",
      "#2|NA|Had slight cerebellar atrophy.\n",
      "#6|NA|Had active seizures recorded on overnight EEG. Had slight cerebellar atrophy.\n",
      "#10|NA|Had slight cerebellar atrophy.\n",
      "#11|NA|In one teenager, follow-up imaging showed atrophy measurably worse after a 20-month interval (net loss of 34 cm3 relative to expected).\n"
     ]
    }
   ],
   "source": [
    "print(response)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
