{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import sys\n",
    "import pandas as pd\n",
    "from ngly1_gpt import utils, llm, doc\n",
    "import logging\n",
    "logging.basicConfig(level=logging.INFO, stream=sys.stdout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#! PYTHONPATH=\"$(pwd)\" python ngly1_gpt/cli.py extract_relations 2>&1 | tee data/logs/extract_relations.log.txt\n",
    "# PYTHONPATH=\"$(pwd)\" python ngly1_gpt/cli.py extract_relations --output-filename=relations_2.tsv 2>&1 | tee data/logs/extract_relations_2.log.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "CSF Laboratory Results\n",
      "Nine subjects underwent lumbar puncture. CSF total protein and albumin concentrations, as well as the CSF/serum albumin ratios, were low in nearly all individuals (Supplementary Table S3). There was no correlation between age and CSF protein or albumin levels. In the two oldest subjects, CSF 5-hydroxyindolacetic acid (5-HIAA) and homovanillic acid (HVA) were decreased, suggesting neuronal loss. Neopterin levels were normal but decreased with age, and CSF tetrahydrobiopterin (BH4) levels were below the lower limit of normal in all but one subject tested; however, there was no correlation with age. CSF 5-HIAA, HVA, and BH4 levels strongly and directly correlated with brain atrophy. CSF lactate and amino acid levels were essentially normal (Supplementary Table S3). CSF leucocyte counts (0–4; normal 0–5), glucose concentrations (54–72 mg/dL; normal 40–70), 3-O-methyldopa concentrations (12–28 nM; normal <150) and 5-methyltetrahydrofolate concentrations (45– 85 nmol/L; normal 40–150) were unremarkable (data not shown).\n",
      "--- END TEXT ---\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|associated with|CSF total protein and albumin concentrations|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|CSF/serum albumin ratios|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|CSF 5-hydroxyindolacetic acid (5-HIAA)|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|homovanillic acid (HVA)|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|neuronal loss|phenotype\n",
      "NGLY1 deficiency|disease|associated with|Neopterin levels|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|CSF tetrahydrobiopterin (BH4) levels|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|brain atrophy|phenotype\n",
      "NGLY1 deficiency|disease|associated with|CSF lactate and amino acid levels|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|CSF leucocyte counts|cell type\n",
      "NGLY1 deficiency|disease|associated with|glucose concentrations|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|3-O-methyldopa concentrations|chemical substance\n",
      "NGLY1 deficiency|disease|associated with|5-methyltetrahydrofolate concentrations|chemical substance\n",
      "\n",
      "\n",
      " 44%|████▍     | 15/34 [05:23<07:03, 22.27s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "NCS, EMG, and QSWEAT Testing\n",
      "Nerve conduction studies were performed in 11 individuals (Supplementary Table S4). The predominant finding was an axonal sensorimotor polyneuropathy (n=8) with additional demyelinative features (n=6). Results of neurophysiological testing demonstrated a length- dependent, progressive loss of sensory and motor axons and sympathetic nerve function, apparently static abnormalities in myelination of peripheral nerves, and possible motor neuron degeneration. A single individual (#6) had a repeat study at one year, showing progression of the neuropathy. On needle EMG, neurogenic findings were noted in 9 subjects with varying degrees of acute and chronic changes consistent with the NCS results.\n",
      " QSWEATs, performed in 11 individuals, were abnormal in the same 8 individuals who had axonal sensorimotor neuropathies. The distal lower extremity QSWEAT was more frequently absent (7/11) compared to the forearm (1/11), consistent with a length dependent neuropathy. There was a trend to greater severity of neuropathy in the older individuals, with an inverse correlation between tibial motor amplitude and age (p = 0.002, r2 = 0.799); the amplitude decreased by −0.3 mV per year (normal, >+2.5mV per year).\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Nerve conduction studies|assay|has participant|11 individuals|organism\n",
      "axonal sensorimotor polyneuropathy|disease|has phenotype|demyelinative features|phenotype\n",
      "neurophysiological testing|assay|demonstrates|loss of sensory and motor axons|phenotype\n",
      "neurophysiological testing|assay|demonstrates|sympathetic nerve function abnormalities|phenotype\n",
      "neurophysiological testing|assay|demonstrates|abnormalities in myelination of peripheral nerves|phenotype\n",
      "neurophysiological testing|assay|demonstrates|possible motor neuron degeneration|phenotype\n",
      "individual #6|organism|has phenotype|progression of the neuropathy|phenotype\n",
      "needle EMG|assay|has participant|9 subjects|organism\n",
      "QSWEATs|assay|has participant|11 individuals|organism\n",
      "QSWEATs|assay|associated with|axonal sensorimotor neuropathies|disease\n",
      "distal lower extremity QSWEAT|assay|associated with|absence|phenotype\n",
      "forearm QSWEAT|assay|associated with|presence|phenotype\n",
      "severity of neuropathy|phenotype|correlated with|age|<other>\n",
      "tibial motor amplitude|phenotype|negatively correlated with|age|<other>\n",
      "\n",
      "\n",
      " 47%|████▋     | 16/34 [05:44<06:35, 21.95s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "Ophthalmology\n",
      "Awake and sedated ophthalmic examinations were performed on 11 study participants (Supplementary Table S5). Lagophthalmous, ptosis, exotropia and/or esotropia, corneal neovascularization, pannus formation or scarring, op tic nerve pallor or atrophy, retinal\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|Lagophthalmous|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|ptosis|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|exotropia|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|esotropia|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|corneal neovascularization|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|pannus formation|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|scarring|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|optic nerve pallor|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|atrophy|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|retinal|phenotype\n",
      "\n",
      "\n",
      " 50%|█████     | 17/34 [05:55<05:19, 18.80s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "Audiology\n",
      "pigmentary changes including pigment granularity and pigmentary retinopathy or pigmentary changes, blonde or poorly pigmented retinal periphery, and refractive errors were observed (Figure 4). All individuals had evidence of hypo- or complete alacrima on Schirmer II testing (Supplementary Table S5). Tear production did not correlate with age. One participant underwent sedated electroretinography and was diagnosed with cone dystrophy; the photopic responses were reduced to ~50% of the lower limit of normal for age.\n",
      " Audiologic assessments, including otoacoustic emissions and/or cochlear microphonics to evaluate for the functional integrity of the inner ear, and auditory brainstem response (ABRs), were conducted on 11 subjects. Behavioral hearing thresholds could be established reliably in only three subjects and all had normal hearing sensitivity. Tympanometry was unremarkable. Most remarkable was the dyssynchronous and/or absent transmission through the auditory brainstem and/or 8th nerve in most individuals. The severity of the transmission abnormality through the auditory brainstem increased with increasing age, and the eighth nerve was bilaterally involved the 4 oldest individuals (Supplementary Table S6).\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|pigmentary changes|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|alacrima|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|cone dystrophy|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|abnormal auditory brainstem transmission|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|abnormal 8th nerve transmission|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|normal hearing sensitivity|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|unremarkable tympanometry|phenotype\n",
      "age|<other>|correlated with|severity of the transmission abnormality through the auditory brainstem|phenotype\n",
      "age|<other>|correlated with|bilateral involvement of the eighth nerve|phenotype\n",
      "\n",
      "\n",
      " 53%|█████▎    | 18/34 [06:08<04:30, 16.91s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "Clinical Feeding and Swallowing Assessments\n",
      "Clinical feeding and modified barium swallow assessments were performed on 11 subjects. Oral motor deficits, including persistent nutritive suckle swallow and poor oral bolus formation, were present in 10 of 11. Premature spillage and pharyngeal swallow response delays remained the primary deficits of all subjects. No aspiration or laryngeal penetration for any textures were observed in the swallow studies. Developmental delays in m astication were evident and characterized by munching patterns without matured rotary chewing of solid textures. Generalized weakness of the lips and tongue was observed in all individuals during cranial nerve assessments. Dystonic movements of the tongue and persistent oral reflexes of suckling and suck/swallows were seen in the majority of individuals.\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|Oral motor deficits|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|Premature spillage|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|pharyngeal swallow response delays|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|Developmental delays in mastication|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|Generalized weakness of the lips and tongue|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|Dystonic movements of the tongue|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|persistent oral reflexes of suckling and suck/swallows|phenotype\n",
      "\n",
      "\n",
      " 56%|█████▌    | 19/34 [06:22<03:58, 15.92s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "Cardiorespiratory Status\n",
      "Echocardiograms were unremarkable (n=12). Electrocardiograms showed heart rates in the low 100s in all subjects (n=12), with two individuals having QTcB >440 ms, but a normal QTcF. Sleep studies (n=9) identified evidence of obstructive sleep apnea (n=2), central sleep apnea (n=1), and combined obstructive and central sleep apnea (n=2). Five individuals had frequent periodic limb movements. There were no associations between age and either sleep apnea or frequent periodic limb movements.\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|unremarkable Echocardiograms|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|heart rates in the low 100s|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|QTcB >440 ms|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|normal QTcF|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|obstructive sleep apnea|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|central sleep apnea|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|combined obstructive and central sleep apnea|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|frequent periodic limb movements|phenotype\n",
      "\n",
      "\n",
      " 59%|█████▉    | 20/34 [06:36<03:36, 15.45s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "Gastrointestinal and Nutritional Findings\n",
      "Gastric pH was assessed after H2 blockers and osmotic pump inhibitors had been discontinued for 5 days. Gastric pH was appropriately acidic in all individuals tested except one, whose pH was 7.25. Ten of twelve individuals had some degree of constipation. Abdominal ultrasounds were normal in six of 12 individuals. Abnormalities in abdominal ultrasound included splenomegaly, steatosis, coarse or inhomogeneous liver texture, or hepatomegaly. No hydronephrosis, polycystic kidneys, horseshoe kidneys, kidney calculi, nephrocalcinosis, or perinephric fluid were seen. Three of twelve had Fibroscan scores above 7 kPa, the upper limit of normal, indicating possible liver fibrosis. Resting energy expenditure was close to 100% of expected in three individuals, but ranged from 51% to 82% of predicted in the other 9 (data not shown).\n",
      " Laboratory values reflecting gastrointestinal and hepatic function were essentially normal at the time of the NIH evaluation (Supplementary Table S7). However, chart review revealed elevations in aspartate aminotransferase (AST) and/or alanine transaminase (ALT) in all eight subjects that had these measured in their first two years of life. One individual had undergone liver transplantation for presumed hepatocellular carcinoma (further details in Supplementary Materials and Methods). These transaminase levels normalized around age 4 (Supplementary Figure S5). Circulating proteins were normal or borderline low in all individuals (Supplementary Table S7). Total cholesterol, LDL, and triglycerides were also low (Supplementary Table S7). In the two subjects with the lowest cholesterol levels, high density lipoprotein, low density lipoprotein, and very low density lipoprotein particle number and sizes were normal (data not shown).\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|Gastric pH alteration|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|constipation|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|abnormal abdominal ultrasound|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|liver fibrosis|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|altered resting energy expenditure|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|elevated aspartate aminotransferase|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|elevated alanine transaminase|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|hepatocellular carcinoma|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low total cholesterol|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low LDL|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low triglycerides|phenotype\n",
      "\n",
      "\n",
      " 62%|██████▏   | 21/34 [06:57<03:41, 17.05s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "Hematologic, Endocrine, Immunologic, and Biochemical Phenotype\n",
      "Hematologic profiles showed unremarkable complete blood counts (Supplementary Table S8) with the exception of one subject with lymphoblasts, subsequently diagnosed with acute lymphocytic leukemia (supplementary results). Coagulation studies showed normal prothrombin time (PT), partial thromboplastin time (PTT), and international normalized ratio (INR), but low protein C activity (n=6), Factor II activity (n=1), Factor IX (n=2), Factor XI (n=2), and fibrinogen (n=5) (Supplementary Table S8). On average, all factors except for Antithrombin III were lower in individuals with the c.1201A>T mutation; for Factor II, Factor IX, and Factor XI, the differences were significant at p<0.05 using the two sample t- test.\n",
      " Detailed laboratory analyses of endocrine (Supplementary Table S9) and immune (Supplementary Table S10) function were performed. The most striking finding was that seven of 11 individuals tested exhibited out of range elevations in antibody titers towards rubella and/or rubeola following MMR vaccination. Biochemical markers known to be abnormal in mitochondropathies and hypoglycosylation disorders showed inconsistent results (Supplementary Tables S11 and S12).\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "NGLY1 deficiency|disease|has phenotype|unremarkable complete blood counts|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low protein C activity|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low Factor II activity|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low Factor IX|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low Factor XI|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|low fibrinogen|phenotype\n",
      "c.1201A>T mutation|genetic variant|associated with|lower Factor II, Factor IX, and Factor XI|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|elevations in antibody titers towards rubella and/or rubeola|phenotype\n",
      "NGLY1 deficiency|disease|has phenotype|abnormal biochemical markers in mitochondropathies and hypoglycosylation disorders|phenotype\n",
      "one subject|organism|has phenotype|acute lymphocytic leukemia|disease\n",
      "\n",
      "\n",
      " 65%|██████▍   | 22/34 [07:16<03:33, 17.82s/it]\u001b[AINFO:ngly1_gpt.llm:Prompt (temperature=0.0, model=gpt-4):\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n"
     ]
    }
   ],
   "source": [
    "!cat data/logs/extract_relations_1.log.txt | grep 'CSF Laboratory Results' -B 3 -A 3\n",
    "!cat data/logs/extract_relations_1.log.txt | grep 'CSF Laboratory Results' -B 3 -A 1000 | grep 'Extracted relation records:' -A 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here is the text to extract relations from: \n",
    "\n",
    "```\n",
    "--- BEGIN TEXT ---\n",
    "CSF Laboratory Results\n",
    "\n",
    "Nine subjects underwent lumbar puncture. CSF total protein and albumin concentrations, as well as the CSF/serum albumin ratios, were low in nearly all individuals (Supplementary Table S3). There was no correlation between age and CSF protein or albumin levels. In the two oldest subjects, CSF 5-hydroxyindolacetic acid (5-HIAA) and homovanillic acid (HVA) were decreased, suggesting neuronal loss. Neopterin levels were normal but decreased with age, and CSF tetrahydrobiopterin (BH4) levels were below the lower limit of normal in all but one subject tested; however, there was no correlation with age. CSF 5-HIAA, HVA, and BH4 levels strongly and directly correlated with brain atrophy. CSF lactate and amino acid levels were essentially normal (Supplementary Table S3). CSF leucocyte counts (0–4; normal 0–5), glucose concentrations (54–72 mg/dL; normal 40–70), 3-O-methyldopa concentrations (12–28 nM; normal <150) and 5-methyltetrahydrofolate concentrations (45– 85 nmol/L; normal 40–150) were unremarkable (data not shown).\n",
    "--- END TEXT ---\n",
    "```\n",
    "\n",
    "INFO:ngly1_gpt.llm:Response:\n",
    "NGLY1 deficiency|disease|associated with|CSF total protein and albumin concentrations|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|CSF/serum albumin ratios|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|CSF 5-hydroxyindolacetic acid (5-HIAA)|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|homovanillic acid (HVA)|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|neuronal loss|phenotype\n",
    "NGLY1 deficiency|disease|associated with|Neopterin levels|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|CSF tetrahydrobiopterin (BH4) levels|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|brain atrophy|phenotype\n",
    "NGLY1 deficiency|disease|associated with|CSF lactate and amino acid levels|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|CSF leucocyte counts|cell type\n",
    "NGLY1 deficiency|disease|associated with|glucose concentrations|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|3-O-methyldopa concentrations|chemical substance\n",
    "NGLY1 deficiency|disease|associated with|5-methyltetrahydrofolate concentrations|chemical substance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28826"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('data/extract/PMC4243708.txt', 'r') as f:\n",
    "    text = f.read()\n",
    "len(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chunks = doc.split_text(text, 1000, doc.nlp(), doc.encoding(llm.DEFAULT_MODEL))\n",
    "len(chunks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Hypotheses Generated for Pathophysiology of Clinical Findings:\n",
      "- Loss of neurons, indicated by brain atrophy on MRI and functional assessments, contribute to the functional impairment seen in NGLY1-CDDG. \n",
      "- Biochemical abnormalities such as lower CSF metabolites (BH4, 5-HIAA, HVA) may be secondary to brain atrophy.\n",
      "- An imbalance between glial cells and neurons, possibly due to loss of neurons, significantly contributes to the severity of the phenotype. This imbalance is indicated by lower NAA levels and increased choline, myo-inositol and creatine levels. \n",
      "- Impaired sweat response, largely affecting distal responses, may be due to a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. \n",
      "- Decreased CSF protein and albumin concentrations, potentially representing a novel diagnostic marker, could be related to the disorder's etiology. \n",
      "- NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation due to the role of N-glycanase 1 in cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein. \n",
      "- A potential pathophysiologic contributor to NGLY1-CDDG is dysfunction in endoplasmic reticulum-associated degradation pathway (ERAD), as misfolded glycoproteins are processed through ERAD and then retrotranslocated into the cytoplasm, with N-glycanase, implicated in NGLY1-CDDG, being the first step in further degradation of these molecules. However, this hypothesis is tempered by preliminary experiments that found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# chunk = \"\"\"\n",
    "# Title\n",
    "# Mutations in NGLY1 Cause an Inherited Disorder of the Endoplasmic Reticulum-Associated Degradation (ERAD) Pathway\n",
    "\n",
    "# Abstract\n",
    "# Purpose: The endoplasmic reticulum-associated degradation (ERAD) pathway is responsible for the translocation of misfolded proteins across the ER membrane into the cytosol for subsequent degradation by the proteasome. In order to understand the spectrum of clinical and molecular findings in a complex neurological syndrome, we studied a series of eight patients with inherited deficiency of N-glycanase 1 (NGLY1), a novel disorder of cytosolic ERAD dysfunction.\n",
    "# Methods: Whole-genome, whole-exome or standard Sanger sequencing techniques were employed. Retrospective chart reviews were performed in order to obtain clinical data.\n",
    "# Results: All patients had global developmental delay, a movement disorder, and hypotonia. Other common findings included hypo- or alacrima (7/8), elevated liver transaminases (6/7), microcephaly (6/8), diminished reflexes (6/8), hepatocyte cytoplasmic storage material or vacuolization (5/6), and seizures (4/8). The nonsense mutation c.1201A>T (p.R401X) was the most common deleterious allele.\n",
    "# Conclusions: NGLY1 deficiency is a novel autosomal recessive disorder of the ERAD pathway associated with neurological dysfunction, abnormal tear production, and liver disease. The majority of patients detected to date carry a specific nonsense mutation that appears to be associated with severe disease. The phenotypic spectrum is likely to enlarge as cases with a more broad range of mutations are detected.\n",
    "# \"\"\"\n",
    "# chunk = \"\"\"\n",
    "# DISCUSSION\n",
    "# Summary of Novel Findings\n",
    "# Our prospective investigations into NGLY1-CDDG revealed several new discoveries associated with this disorder. These included low CSF total protein and albumin, optic atrophy and retinal pigmentary changes/cone dystrophy, poor weight gain starting in mid- childhood, lower than predicted resting energy expenditure, central and obstructive sleep apnea, delayed bone age, joint hypermobility, progressive brain atrophy, abnormalities in brain MRS-measured metabolites, various abnormalities on skeletal survey, high antibody titers after rubella and rubeola vaccination, and low levels of several pro- and anti- coagulation factors. Echocardiograms, electrocardiograms, and CSF lactate levels were normal.\n",
    "# We also confirmed and further detailed the global developmental delay, movement disorder, frequent seizure disorder, hypotonia, hypolacrima or alacrima, corneal disease, ptosis, lagophthalmous, strabismus, peripheral neuropathy and occasional diminished reflexes, hypohidrosis, auditory brainstem response abnormalities, abnormal brain imaging, scoliosis, acquired microcephaly, small hands and feet, dysmorphic features, constipation, elevated liver enzymes present only early in childhood, osteopenia, and hypocholesterolemia associated with NGLY1-CDDG. Compared with previously reported biopsy findings, our transient elastography results showed minimal if any fibrosis. In addition, we could not confirm abnormal storage material in the three liver biopsy specimens we analyzed, nor did we observe ocular apraxia in our individuals, despite performing detailed ophthalmologic evaluations that involved some subjects previously reported to have ocular apraxia11.\n",
    "# Based on our findings, we recommend considering NGLY1-CDDG in the differential diagnosis of any patient with the tetrad of developmental delay/intellectual disability, hyperkinetic movement disorder, hypolacrima, and a history of elevated transaminases during early childhood.\n",
    "# \"\"\"\n",
    "# chunk = \"\"\"\n",
    "\n",
    "# FIGURES\n",
    "# Figure 1: Distribution of NGLY1 mutations.\n",
    "# Shaded bars are exons. Previously unreported mutations are in bold. Mutations previously\n",
    "# published and not in this paper include c.1624C>T (exon 11), c.1205_1207delGAA (exon 8) and c.1533_1536delTCAA (exon 10). Four mutations lie within the catalytic domain (amino acid residues 292–353);, i.e., c.730T>C, c.953T>C, c.931G>A, and c.930C>T. The accession numbers for the NM_018297.3:c.931G>A, NM_018297.3:c.1604G>A, NM_018297.3:c.622C>T, NM_018297.3:c.347C>G, NM_018297.3:c.1169G>C, NM_018297.3:c.730T>C, NM_018297.3:c.1910delT, NM_018297.3:c.930C>T, NM_018297.3:c.881+5G>T\n",
    "# sequences reported in this paper are ClinVar: SCV000259179, SCV000259180, SCV000259181, SCV000259182, SCV000259183, SCV000259184, SCV000259185, SCV000259186, and SCV000259187 respectively.\n",
    "   \n",
    "# Figure 2: Facial features of NGLY1-CDDG.\n",
    "# Individuals with NGLY1-CDDG are arranged from youngest (top left) to oldest (bottom right). Facial features include upturned nasal tip, hypotonic facies, ptosis, brachycephaly, thinned facies, hollowed cheeks, and visible zygomatic arches. Interpupillary distance ranged from the 13–95 centiles with the mean at the 66th centile, and SEM of 8. Horizontal fissure length ranged from 3–70 centiles with mean at the 20th centile, and SEM of 1. Inner canthal distance ranged from 3–100 centiles with the mean at the 59th centile and SEM of 10. Outer canthal distance ranged from 6–81 centile with mean at the 43 centile and SEM of 7. Canthal index ranged from −2 to +2 SD from the normal mean.\n",
    "\n",
    "# Figure 3: Brain MRI findings.\n",
    "# White matter lesions were seen in two of eleven individuals. A shows multiple lesions in the periventricular white matter, some of which were confluent. B shows a single lesion in the periventricular white matter. Both A and B illustrate cerebral atrophy and were performed using T2-weighting. Sulci are slightly prominent in both A and B, and prominent ventricles are visible in A. C illustrates high position of the cerebellar tonsils, and large foramen of Magendie and cisterna magna.\n",
    "# \"\"\"\n",
    "\n",
    "chunk = \"\"\"\n",
    "Hypotheses Generated for Pathophysiology of Clinical Findings:\n",
    "- Loss of neurons, indicated by brain atrophy on MRI and functional assessments, contribute to the functional impairment seen in NGLY1-CDDG. \n",
    "- Biochemical abnormalities such as lower CSF metabolites (BH4, 5-HIAA, HVA) may be secondary to brain atrophy.\n",
    "- An imbalance between glial cells and neurons, possibly due to loss of neurons, significantly contributes to the severity of the phenotype. This imbalance is indicated by lower NAA levels and increased choline, myo-inositol and creatine levels. \n",
    "- Impaired sweat response, largely affecting distal responses, may be due to a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. \n",
    "- Decreased CSF protein and albumin concentrations, potentially representing a novel diagnostic marker, could be related to the disorder's etiology. \n",
    "- NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation due to the role of N-glycanase 1 in cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein. \n",
    "- A potential pathophysiologic contributor to NGLY1-CDDG is dysfunction in endoplasmic reticulum-associated degradation pathway (ERAD), as misfolded glycoproteins are processed through ERAD and then retrotranslocated into the cytoplasm, with N-glycanase, implicated in NGLY1-CDDG, being the first step in further degradation of these molecules. However, this hypothesis is tempered by preliminary experiments that found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts.\n",
    "\"\"\"\n",
    "\n",
    "print(chunk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[83, 1609, 5963, 374, 2294, 0]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:ngly1_gpt.llm:Prompt:\n",
      "Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- tissue\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- protein\n",
      "- metabolite\n",
      "- gene family \n",
      "- genotype\n",
      "- genetic variant\n",
      "- transcript variant\n",
      "- protein variant\n",
      "- macromolecular complex\n",
      "- molecular activity\n",
      "- organism\n",
      "- pathway\n",
      "- phenotype\n",
      "- symptom\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "\n",
      "Hypotheses Generated for Pathophysiology of Clinical Findings:\n",
      "- Loss of neurons, indicated by brain atrophy on MRI and functional assessments, contribute to the functional impairment seen in NGLY1-CDDG. \n",
      "- Biochemical abnormalities such as lower CSF metabolites (BH4, 5-HIAA, HVA) may be secondary to brain atrophy.\n",
      "- An imbalance between glial cells and neurons, possibly due to loss of neurons, significantly contributes to the severity of the phenotype. This imbalance is indicated by lower NAA levels and increased choline, myo-inositol and creatine levels. \n",
      "- Impaired sweat response, largely affecting distal responses, may be due to a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. \n",
      "- Decreased CSF protein and albumin concentrations, potentially representing a novel diagnostic marker, could be related to the disorder's etiology. \n",
      "- NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation due to the role of N-glycanase 1 in cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein. \n",
      "- A potential pathophysiologic contributor to NGLY1-CDDG is dysfunction in endoplasmic reticulum-associated degradation pathway (ERAD), as misfolded glycoproteins are processed through ERAD and then retrotranslocated into the cytoplasm, with N-glycanase, implicated in NGLY1-CDDG, being the first step in further degradation of these molecules. However, this hypothesis is tempered by preliminary experiments that found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts.\n",
      "\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for this data:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"NGLY1 deficiency\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n",
      "DEBUG:openai:message='Request to OpenAI API' method=post path=https://api.openai.com/v1/chat/completions\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:openai:api_version=None data='{\"model\": \"gpt-4\", \"messages\": [{\"role\": \"user\", \"content\": \"Text will be provided that contains information from a published, biomedical research article about NGLY1 deficiency. Extract subject-predicate-object relations from this text.\\\\n\\\\nTypes of subjects and objects to extract:\\\\n- biological process\\\\n- cell type\\\\n- cellular component\\\\n- tissue\\\\n- chemical substance\\\\n- clinical trial\\\\n- disease\\\\n- drug\\\\n- gene\\\\n- protein\\\\n- metabolite\\\\n- gene family \\\\n- genotype\\\\n- genetic variant\\\\n- transcript variant\\\\n- protein variant\\\\n- macromolecular complex\\\\n- molecular activity\\\\n- organism\\\\n- pathway\\\\n- phenotype\\\\n- symptom\\\\n\\\\nPredicates to extract:\\\\n- affects risk for\\\\n- associated with\\\\n- capable of\\\\n- caused by\\\\n- causes\\\\n- colocalizes with\\\\n- contributes to\\\\n- correlated with\\\\n- decreases abundance of\\\\n- decreases activity of\\\\n- derives from\\\\n- disrupts\\\\n- enables\\\\n- exact match\\\\n- expressed in\\\\n- expresses\\\\n- genetically interacts with\\\\n- has affected feature\\\\n- has attribute\\\\n- has gene product\\\\n- has genotype\\\\n- has metabolite\\\\n- has output\\\\n- has participant\\\\n- has phenotype\\\\n- has role\\\\n- in 1 to 1 orthology relationship with\\\\n- in orthology relationship with\\\\n- in paralogy relationship with\\\\n- in taxon\\\\n- in xenology relationship with\\\\n- increases abundance of\\\\n- increases activity of\\\\n- instance of\\\\n- interacts with\\\\n- involved in\\\\n- is allele of\\\\n- is marker for\\\\n- is model of\\\\n- is part of\\\\n- located in\\\\n- location of\\\\n- manifestation of\\\\n- molecularly interacts with\\\\n- negatively correlated with\\\\n- negatively regulates\\\\n- occurs in\\\\n- part of\\\\n- participates in\\\\n- pathogenic for condition\\\\n- positively correlated with\\\\n- positively regulates\\\\n- precedes\\\\n- prevents\\\\n- produced by\\\\n- produces\\\\n- regulates\\\\n- treats\\\\n\\\\nHere is the text to extract relations from: \\\\n--- BEGIN TEXT ---\\\\n\\\\nHypotheses Generated for Pathophysiology of Clinical Findings:\\\\n- Loss of neurons, indicated by brain atrophy on MRI and functional assessments, contribute to the functional impairment seen in NGLY1-CDDG. \\\\n- Biochemical abnormalities such as lower CSF metabolites (BH4, 5-HIAA, HVA) may be secondary to brain atrophy.\\\\n- An imbalance between glial cells and neurons, possibly due to loss of neurons, significantly contributes to the severity of the phenotype. This imbalance is indicated by lower NAA levels and increased choline, myo-inositol and creatine levels. \\\\n- Impaired sweat response, largely affecting distal responses, may be due to a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. \\\\n- Decreased CSF protein and albumin concentrations, potentially representing a novel diagnostic marker, could be related to the disorder\\'s etiology. \\\\n- NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation due to the role of N-glycanase 1 in cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein. \\\\n- A potential pathophysiologic contributor to NGLY1-CDDG is dysfunction in endoplasmic reticulum-associated degradation pathway (ERAD), as misfolded glycoproteins are processed through ERAD and then retrotranslocated into the cytoplasm, with N-glycanase, implicated in NGLY1-CDDG, being the first step in further degradation of these molecules. However, this hypothesis is tempered by preliminary experiments that found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts.\\\\n\\\\n--- END TEXT ---\\\\n\\\\nReport each relation record on a separate line in CSV format using a pipe (i.e. \\\\\"|\\\\\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\\\\n\\\\nRequirements for this data:\\\\n\\\\n- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\\\\n- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \\\\\"pathway\\\\\" or \\\\\"disease\\\\\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\\\\n- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \\\\\"causes\\\\\" or \\\\\"participates in\\\\\"); do not use predicates outside of this list.\\\\n- Assume that \\\\\"NGLY1 deficiency\\\\\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\\\\n- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\\\\n- Include the CSV headers in the response\\\\n\\\\nHere is an example response with illustrative data that demonstrates the desired result format:\\\\n--- BEGIN EXAMPLE ---\\\\nsubject|subject_entity|predicate|object|object_entity\\\\nHutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\\\\n--- END EXAMPLE ---\\\\n\\\\nReport only the relation records without explanation or other text of any kind.\\\\n\\\\nExtracted relation records:\"}]}' message='Post details'\n",
      "DEBUG:urllib3.util.retry:Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.openai.com:443\n",
      "DEBUG:urllib3.connectionpool:https://api.openai.com:443 \"POST /v1/chat/completions HTTP/1.1\" 200 None\n",
      "DEBUG:openai:message='OpenAI API response' path=https://api.openai.com/v1/chat/completions processing_ms=39146 request_id=ace8643beb9c1a5f5e7c17ab0e64358c response_code=200\n",
      "INFO:ngly1_gpt.llm:Response:\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Loss of neurons|biological process|contributes to|NGLY1-CDDG|disease\n",
      "Lower CSF metabolites|phenotype|associated with|brain atrophy|phenotype\n",
      "Imbalance between glial cells and neurons|phenotype|contributes to|severity of the phenotype in NGLY1-CDDG|disease\n",
      "Impaired sweat response|phenotype|associated with|small fiber neuropathy|disease\n",
      "Decreased CSF protein and albumin concentrations|phenotype|associated with|NGLY1-CDDG|disease\n",
      "NGLY1-CDDG|disease|enables|hypoglycosylation|biological process\n",
      "NGLY1-CDDG|disease|associated with|dysfunction in endoplasmic reticulum-associated degradation pathway|pathway\n",
      "Misfolded glycoproteins|protein|participates in|endoplasmic reticulum-associated degradation pathway|pathway\n",
      "N-glycanase 1|protein|has role in|cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein|biological process\n",
      "N-glycanase|protein|involved in|degradation of misfolded glycoproteins|molecular activity\n",
      "NGLY1-CDDG patient fibroblasts|cell type|expresses|standard ERAD marker|protein\n",
      "NGLY1-CDDG|disease|associated with|lower NAA levels and increased choline, myo-inositol and creatine levels|phenotype\n",
      "----------------------------------------------------------------------------------------------------\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Loss of neurons|biological process|contributes to|NGLY1-CDDG|disease\n",
      "Lower CSF metabolites|phenotype|associated with|brain atrophy|phenotype\n",
      "Imbalance between glial cells and neurons|phenotype|contributes to|severity of the phenotype in NGLY1-CDDG|disease\n",
      "Impaired sweat response|phenotype|associated with|small fiber neuropathy|disease\n",
      "Decreased CSF protein and albumin concentrations|phenotype|associated with|NGLY1-CDDG|disease\n",
      "NGLY1-CDDG|disease|enables|hypoglycosylation|biological process\n",
      "NGLY1-CDDG|disease|associated with|dysfunction in endoplasmic reticulum-associated degradation pathway|pathway\n",
      "Misfolded glycoproteins|protein|participates in|endoplasmic reticulum-associated degradation pathway|pathway\n",
      "N-glycanase 1|protein|has role in|cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein|biological process\n",
      "N-glycanase|protein|involved in|degradation of misfolded glycoproteins|molecular activity\n",
      "NGLY1-CDDG patient fibroblasts|cell type|expresses|standard ERAD marker|protein\n",
      "NGLY1-CDDG|disease|associated with|lower NAA levels and increased choline, myo-inositol and creatine levels|phenotype\n"
     ]
    }
   ],
   "source": [
    "result = llm.extract_relations(chunk, disease=\"NGLY1 deficiency\", model=\"gpt-4\")\n",
    "print('-'*100)\n",
    "print(result)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----- \n",
    "\n",
    "Mechanism Hypothesis Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "text = \"\"\"\n",
    "Our detailed assessment of the phenotypic features has also led to several hypotheses regarding underlying pathophysiology. The strong correlation between brain atrophy on MRI and functional assessments suggests that loss of neurons contributes to the functional impairment. The atrophy also correlated with CSF metabolites (BH4, 5-HIAA, HVA), which are known to be lower when there is damage to neurotransmitter producing neurons22–24. This suggests that these biochemical abnormalities may be secondary to brain atrophy.\n",
    "\n",
    "Results from MRS showed that as functional impairment worsened and age increased, NAA decreased, while choline, myo-inositol and creatine increased. Additionally, creatine and myoinositol were inversely correlated, and NAA directly correlated, with neurotransmitter levels (5-HIAA, 5-HVA, 3-OMD, and neopterin, Supplementary Figure S3). NAA may be found exclusively in neurons and declines as neurons become unhealthy or die25,26. Elevations in choline could be consistent with a relative abundance of glial cells secondary to neuronal loss26. Myo-inositol is a glia-specific marker that is elevated by gliosis or inflammation27,28; in our cohort we do not have any evidence of inflammation. Creatine levels are relatively higher in glial cells compared to neurons29. Taken together, the MRS findings suggest that there is a relative abundance of glial cells compared to neurons in the brain of NGLY1-CDDG individuals, possibly due to loss of neurons, and that the degree of this imbalance contributes significantly to the severity of the phenotype.\n",
    "\n",
    "The impaired sweat response, largely affecting distal responses, is consistent with a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. Albumin CSF/serum quotient, or QAlb, is a marker of blood-CSF barrier dysfunction, and its value is inversely correlated to the CSF turnover rate30. There was no significant change in the QAlb ratio (data not shown), arguing against increased turnover as a possible etiology. Regardless of the mechanism, we propose that decreased CSF protein and albumin concentrations represent a novel diagnostic marker for this disorder.\n",
    "\n",
    "Individuals with NGLY1-CDDG and those with N-linked glycosylation disorders share the phenotypic features of low cholesterol, hepatopathy, peripheral neuropathy, retinal and optic nerve abnormalities, seizures, developmental delay with socialization as a relative strength, and delayed bone age18. NGLY1-CDDG also overlaps with O-linked glycosylation disorders with respect to hypolacrima19. There may be a pathogenic relationship; N-glycanase 1 catalyzes the cleavage of the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein9, so NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation. Indeed, the individuals with NGLY1-CDDG exhibited subtle abnormalities in transferrin and ApoC-III glycosylation20.\n",
    "\n",
    "Another hypothesis for the etiology of the NGLY1-CDDG phenotype is based on the fact that misfolded glycoproteins are processed through the endoplasmic reticulum associated degradation pathway (ERAD) and then retrotranslocated into the cytoplasm. There, N-glycanase is the first step in further degradation of these molecules, making ERAD dysfunction a possible pathophysiologic contributor especially given evidence of ER-stress in mouse embryonic fibroblasts31. However, preliminary experiments found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts (H.H. Freeze, personal communication, April 2016).\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:ngly1_gpt.llm:Prompt:\n",
      "Text will be provided that contains information from a published, biomedical research article about \"NGLY1 deficiency\". Extract a summary of hypothesized causal, pathophysiological mechanisms that describe what biological processes and entities might be modulated by a therapeutic to treat the disease.  Summarize all hypotheses as a bullet-point list.  If there is only one hypothesis, summarize it as a single bullet point.  If there are no such hypotheses, respond with nothing at all.  Do not include any hypotheses that do not imply a clear, potential chemical or physical intervention.\n",
      "\n",
      "Here is the text to extract these potential mechanisms from:\n",
      "--- BEGIN TEXT ---\n",
      "\n",
      "Our detailed assessment of the phenotypic features has also led to several hypotheses regarding underlying pathophysiology. The strong correlation between brain atrophy on MRI and functional assessments suggests that loss of neurons contributes to the functional impairment. The atrophy also correlated with CSF metabolites (BH4, 5-HIAA, HVA), which are known to be lower when there is damage to neurotransmitter producing neurons22–24. This suggests that these biochemical abnormalities may be secondary to brain atrophy.\n",
      "\n",
      "Results from MRS showed that as functional impairment worsened and age increased, NAA decreased, while choline, myo-inositol and creatine increased. Additionally, creatine and myoinositol were inversely correlated, and NAA directly correlated, with neurotransmitter levels (5-HIAA, 5-HVA, 3-OMD, and neopterin, Supplementary Figure S3). NAA may be found exclusively in neurons and declines as neurons become unhealthy or die25,26. Elevations in choline could be consistent with a relative abundance of glial cells secondary to neuronal loss26. Myo-inositol is a glia-specific marker that is elevated by gliosis or inflammation27,28; in our cohort we do not have any evidence of inflammation. Creatine levels are relatively higher in glial cells compared to neurons29. Taken together, the MRS findings suggest that there is a relative abundance of glial cells compared to neurons in the brain of NGLY1-CDDG individuals, possibly due to loss of neurons, and that the degree of this imbalance contributes significantly to the severity of the phenotype.\n",
      "\n",
      "The impaired sweat response, largely affecting distal responses, is consistent with a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. Albumin CSF/serum quotient, or QAlb, is a marker of blood-CSF barrier dysfunction, and its value is inversely correlated to the CSF turnover rate30. There was no significant change in the QAlb ratio (data not shown), arguing against increased turnover as a possible etiology. Regardless of the mechanism, we propose that decreased CSF protein and albumin concentrations represent a novel diagnostic marker for this disorder.\n",
      "\n",
      "Individuals with NGLY1-CDDG and those with N-linked glycosylation disorders share the phenotypic features of low cholesterol, hepatopathy, peripheral neuropathy, retinal and optic nerve abnormalities, seizures, developmental delay with socialization as a relative strength, and delayed bone age18. NGLY1-CDDG also overlaps with O-linked glycosylation disorders with respect to hypolacrima19. There may be a pathogenic relationship; N-glycanase 1 catalyzes the cleavage of the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein9, so NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation. Indeed, the individuals with NGLY1-CDDG exhibited subtle abnormalities in transferrin and ApoC-III glycosylation20.\n",
      "\n",
      "Another hypothesis for the etiology of the NGLY1-CDDG phenotype is based on the fact that misfolded glycoproteins are processed through the endoplasmic reticulum associated degradation pathway (ERAD) and then retrotranslocated into the cytoplasm. There, N-glycanase is the first step in further degradation of these molecules, making ERAD dysfunction a possible pathophysiologic contributor especially given evidence of ER-stress in mouse embryonic fibroblasts31. However, preliminary experiments found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts (H.H. Freeze, personal communication, April 2016).\n",
      "\n",
      "--- END TEXT ---\n",
      "\n",
      "List of hypotheses:\n",
      "DEBUG:openai:message='Request to OpenAI API' method=post path=https://api.openai.com/v1/chat/completions\n",
      "DEBUG:openai:api_version=None data='{\"model\": \"gpt-4\", \"messages\": [{\"role\": \"user\", \"content\": \"Text will be provided that contains information from a published, biomedical research article about \\\\\"NGLY1 deficiency\\\\\". Extract a summary of hypothesized causal, pathophysiological mechanisms that describe what biological processes and entities might be modulated by a therapeutic to treat the disease.  Summarize all hypotheses as a bullet-point list.  If there is only one hypothesis, summarize it as a single bullet point.  If there are no such hypotheses, respond with nothing at all.  Do not include any hypotheses that do not imply a clear, potential chemical or physical intervention.\\\\n\\\\nHere is the text to extract these potential mechanisms from:\\\\n--- BEGIN TEXT ---\\\\n\\\\nOur detailed assessment of the phenotypic features has also led to several hypotheses regarding underlying pathophysiology. The strong correlation between brain atrophy on MRI and functional assessments suggests that loss of neurons contributes to the functional impairment. The atrophy also correlated with CSF metabolites (BH4, 5-HIAA, HVA), which are known to be lower when there is damage to neurotransmitter producing neurons22\\\\u201324. This suggests that these biochemical abnormalities may be secondary to brain atrophy.\\\\n\\\\nResults from MRS showed that as functional impairment worsened and age increased, NAA decreased, while choline, myo-inositol and creatine increased. Additionally, creatine and myoinositol were inversely correlated, and NAA directly correlated, with neurotransmitter levels (5-HIAA, 5-HVA, 3-OMD, and neopterin, Supplementary Figure S3). NAA may be found exclusively in neurons and declines as neurons become unhealthy or die25,26. Elevations in choline could be consistent with a relative abundance of glial cells secondary to neuronal loss26. Myo-inositol is a glia-specific marker that is elevated by gliosis or inflammation27,28; in our cohort we do not have any evidence of inflammation. Creatine levels are relatively higher in glial cells compared to neurons29. Taken together, the MRS findings suggest that there is a relative abundance of glial cells compared to neurons in the brain of NGLY1-CDDG individuals, possibly due to loss of neurons, and that the degree of this imbalance contributes significantly to the severity of the phenotype.\\\\n\\\\nThe impaired sweat response, largely affecting distal responses, is consistent with a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. Albumin CSF/serum quotient, or QAlb, is a marker of blood-CSF barrier dysfunction, and its value is inversely correlated to the CSF turnover rate30. There was no significant change in the QAlb ratio (data not shown), arguing against increased turnover as a possible etiology. Regardless of the mechanism, we propose that decreased CSF protein and albumin concentrations represent a novel diagnostic marker for this disorder.\\\\n\\\\nIndividuals with NGLY1-CDDG and those with N-linked glycosylation disorders share the phenotypic features of low cholesterol, hepatopathy, peripheral neuropathy, retinal and optic nerve abnormalities, seizures, developmental delay with socialization as a relative strength, and delayed bone age18. NGLY1-CDDG also overlaps with O-linked glycosylation disorders with respect to hypolacrima19. There may be a pathogenic relationship; N-glycanase 1 catalyzes the cleavage of the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein9, so NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation. Indeed, the individuals with NGLY1-CDDG exhibited subtle abnormalities in transferrin and ApoC-III glycosylation20.\\\\n\\\\nAnother hypothesis for the etiology of the NGLY1-CDDG phenotype is based on the fact that misfolded glycoproteins are processed through the endoplasmic reticulum associated degradation pathway (ERAD) and then retrotranslocated into the cytoplasm. There, N-glycanase is the first step in further degradation of these molecules, making ERAD dysfunction a possible pathophysiologic contributor especially given evidence of ER-stress in mouse embryonic fibroblasts31. However, preliminary experiments found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts (H.H. Freeze, personal communication, April 2016).\\\\n\\\\n--- END TEXT ---\\\\n\\\\nList of hypotheses:\"}]}' message='Post details'\n",
      "DEBUG:urllib3.util.retry:Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.openai.com:443\n",
      "DEBUG:urllib3.connectionpool:https://api.openai.com:443 \"POST /v1/chat/completions HTTP/1.1\" 200 None\n",
      "DEBUG:openai:message='OpenAI API response' path=https://api.openai.com/v1/chat/completions processing_ms=25557 request_id=76f7cb5af357111fb1de184fe918e424 response_code=200\n",
      "DEBUG:ngly1_gpt.llm:Response:\n",
      "- Loss of neurons, indicated by brain atrophy on MRI and functional assessments, contribute to the functional impairment seen in NGLY1-CDDG. \n",
      "- Biochemical abnormalities such as lower CSF metabolites (BH4, 5-HIAA, HVA) may be secondary to brain atrophy.\n",
      "- An imbalance between glial cells and neurons, possibly due to loss of neurons, significantly contributes to the severity of the phenotype. This imbalance is indicated by lower NAA levels and increased choline, myo-inositol and creatine levels. \n",
      "- Impaired sweat response, largely affecting distal responses, may be due to a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. \n",
      "- Decreased CSF protein and albumin concentrations, potentially representing a novel diagnostic marker, could be related to the disorder's etiology. \n",
      "- NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation due to the role of N-glycanase 1 in cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein. \n",
      "- A potential pathophysiologic contributor to NGLY1-CDDG is dysfunction in endoplasmic reticulum-associated degradation pathway (ERAD), as misfolded glycoproteins are processed through ERAD and then retrotranslocated into the cytoplasm, with N-glycanase, implicated in NGLY1-CDDG, being the first step in further degradation of these molecules. However, this hypothesis is tempered by preliminary experiments that found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts.\n",
      "----------------------------------------------------------------------------------------------------\n",
      "- Loss of neurons, indicated by brain atrophy on MRI and functional assessments, contribute to the functional impairment seen in NGLY1-CDDG. \n",
      "- Biochemical abnormalities such as lower CSF metabolites (BH4, 5-HIAA, HVA) may be secondary to brain atrophy.\n",
      "- An imbalance between glial cells and neurons, possibly due to loss of neurons, significantly contributes to the severity of the phenotype. This imbalance is indicated by lower NAA levels and increased choline, myo-inositol and creatine levels. \n",
      "- Impaired sweat response, largely affecting distal responses, may be due to a small fiber neuropathy rather than a central etiology or generalized cholinergic dysfunction. \n",
      "- Decreased CSF protein and albumin concentrations, potentially representing a novel diagnostic marker, could be related to the disorder's etiology. \n",
      "- NGLY1-CDDG could impair glycan recycling and lead to hypoglycosylation due to the role of N-glycanase 1 in cleaving the amide bond between the proximal N-acetylglucosamine residue of glycans and the asparagine residue of the protein. \n",
      "- A potential pathophysiologic contributor to NGLY1-CDDG is dysfunction in endoplasmic reticulum-associated degradation pathway (ERAD), as misfolded glycoproteins are processed through ERAD and then retrotranslocated into the cytoplasm, with N-glycanase, implicated in NGLY1-CDDG, being the first step in further degradation of these molecules. However, this hypothesis is tempered by preliminary experiments that found no impairment or enhancement of standard ERAD marker expression under normal conditions in NGLY1-CDDG patient fibroblasts.\n"
     ]
    }
   ],
   "source": [
    "result = llm.extract_mechanism_hypotheses(text, disease=utils.NGLY1_DEFICIENCY, model=\"gpt-4\")\n",
    "print('-'*100)\n",
    "print(result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
