{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### [POC 2]  NGLY1 deficiency relation extraction\n",
    "\n",
    "This analysis builds on POC 1 to process much more data (2 full-text papers w/ ~65k chars total vs ~3.5k chars in POC1) and with much more control over the relations identified. Outline:\n",
    "\n",
    "1. Define a very detailed [prompt](https://vscode.dev/github/eric-czech/ngly1-gpt/blob/main/ngly1_gpt/resources/prompts/relation_extraction_1.txt) defining entities and relations between them to identify\n",
    "2. Run this prompt on chunks of each paper\n",
    "    - There are about 60 chunks total between the two papers\n",
    "    - This takes ~40 mins w/ GPT4\n",
    "    - The two papers used are in [data/extract](data/extract)\n",
    "3. Show examples of text inputs and relation extraction outputs\n",
    "4. Analyze frequencies of relations and entities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import sys\n",
    "import pandas as pd\n",
    "from ngly1_gpt import utils, llm, doc\n",
    "import logging\n",
    "logging.basicConfig(level=logging.INFO, stream=sys.stdout)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Extraction"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For reference, here is the original prompt used that defines the entities and relations for this analysis:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Text will be provided that contains information from a published, biomedical research article about {disease}. Extract subject-predicate-object relations from this text.\n",
      "\n",
      "Types of subjects and objects to extract:\n",
      "- assay\n",
      "- biological process\n",
      "- cell type\n",
      "- cellular component\n",
      "- chemical substance\n",
      "- clinical trial\n",
      "- disease\n",
      "- drug\n",
      "- gene\n",
      "- gene family \n",
      "- genetic variant\n",
      "- genotype\n",
      "- macromolecular complex\n",
      "- metabolite\n",
      "- molecular activity\n",
      "- organism\n",
      "- organization\n",
      "- pathway\n",
      "- phenotype\n",
      "- protein\n",
      "- protein variant\n",
      "- symptom\n",
      "- tissue\n",
      "- transcript variant\n",
      "- <other>\n",
      "\n",
      "Predicates to extract:\n",
      "- affects risk for\n",
      "- associated with\n",
      "- capable of\n",
      "- caused by\n",
      "- causes\n",
      "- colocalizes with\n",
      "- contributes to\n",
      "- correlated with\n",
      "- decreases abundance of\n",
      "- decreases activity of\n",
      "- derives from\n",
      "- disrupts\n",
      "- enables\n",
      "- exact match\n",
      "- expressed in\n",
      "- expresses\n",
      "- genetically interacts with\n",
      "- has affected feature\n",
      "- has attribute\n",
      "- has gene product\n",
      "- has genotype\n",
      "- has metabolite\n",
      "- has output\n",
      "- has participant\n",
      "- has phenotype\n",
      "- has role\n",
      "- in 1 to 1 orthology relationship with\n",
      "- in orthology relationship with\n",
      "- in paralogy relationship with\n",
      "- in taxon\n",
      "- in xenology relationship with\n",
      "- increases abundance of\n",
      "- increases activity of\n",
      "- instance of\n",
      "- interacts with\n",
      "- involved in\n",
      "- is allele of\n",
      "- is marker for\n",
      "- is model of\n",
      "- is part of\n",
      "- located in\n",
      "- location of\n",
      "- manifestation of\n",
      "- molecularly interacts with\n",
      "- negatively correlated with\n",
      "- negatively regulates\n",
      "- occurs in\n",
      "- part of\n",
      "- participates in\n",
      "- pathogenic for condition\n",
      "- positively correlated with\n",
      "- positively regulates\n",
      "- precedes\n",
      "- prevents\n",
      "- produced by\n",
      "- produces\n",
      "- regulates\n",
      "- treats\n",
      "- <other>\n",
      "\n",
      "Here is the text to extract relations from: \n",
      "--- BEGIN TEXT ---\n",
      "{text}\n",
      "--- END TEXT ---\n",
      "\n",
      "Report each relation record on a separate line in CSV format using a pipe (i.e. \"|\") delimiter with the column headers: `subject`, `subject_entity`, `predicate`, `object`, `object_entity`.\n",
      "\n",
      "Requirements for extracted relations:\n",
      "\n",
      "- All `subject` and `object` values must be INDIVIDUAL, specific, named concepts that represent an instance of one of the biomedical entities listed above.\n",
      "- The `subject_entity` and `object_entity` values must be exactly one of the biomedical entities listed above (e.g. \"pathway\" or \"disease\"), and they must describe the entity associated with the specific `subject` and `object`, repsectively.\n",
      "- All `predicate` values must be EXACTLY equivalent to one of the provided predicates (e.g. \"causes\" or \"participates in\"); do not use predicates outside of this list.\n",
      "- Assume that \"{disease}\" is associated with any entities discussed where context does not dictate otherwise, as is this is the primary subject of the article.\n",
      "- Extract as many relations as possible; do not ignore any of them that match the criteria provided.\n",
      "- Include the CSV headers in the response.\n",
      "- If there are no relations to extract in the text, return only the CSV headers.\n",
      "\n",
      "Here is an example response with illustrative data that demonstrates the desired result format:\n",
      "--- BEGIN EXAMPLE ---\n",
      "subject|subject_entity|predicate|object|object_entity\n",
      "Hutchinson-Gilford progeria syndrome|disease|has phenotype|osteolysis|phenotype\n",
      "--- END EXAMPLE ---\n",
      "\n",
      "Report only the relation records without explanation or other text of any kind.\n",
      "\n",
      "Extracted relation records:\n"
     ]
    }
   ],
   "source": [
    "!find ./ | grep relation_extraction_1.txt | xargs cat"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This was then used to extract data from the papers via:\n",
    "\n",
    "```bash\n",
    "PYTHONPATH=\"$(pwd)\" python ngly1_gpt/cli.py extract_relations --output-filename=relations.tsv 2>&1 | tee data/logs/extract_relations.log.txt\n",
    "```\n",
    "\n",
    "The logs for these extractions showing all prompts and results are in [data/logs](data/logs)."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Examples"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here are a few select examples (from the logs) showing how various input chunks are transformed into structured outputs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "log = \"data/logs/extract_relations_1.log.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt text:\n",
      "--- BEGIN TEXT ---\n",
      "Table 1\n",
      "Patient #\tAge (y) / Sex\tAllele #1a\tAllele #2a\tFunctional Scores\n",
      "Mutation (NM_018297.3)\tProtein\tMutation (NM_018297.3)\tProtein\tNijmegenb\tIQ or DQ\tVinelandc\n",
      "1\t3 / M\tc.953T>C\tp.L318P\tc.1169G>C\tp.R390P\t 14\tND\t62\n",
      "2\t4 / M\tc.1201A>T\tp.R401*\tc.1201A>T\tp.R401*\t33\t25\t52\n",
      "3\t4 / F\tc.1201A>T\tp. R401*\tc.1201A>T\tp.R401*\t34\t5\t40\n",
      "4\t5 / F\tc.931G>A\t p.E311K\tc.730T>C\tp.W244R\t33\tND\tND\n",
      "5\t6 / M\tc.1604G>A\tp.W535*\t c.1910delT\tp.L637*\t25\t8\t43\n",
      "6\t7 / M\tc.1891delC\tp.Q631S\t c.1201A>T\tp.R401*\t36\t5\t37\n",
      "7\t8 / F\tc.622C>T\tp.Q208*\tc.930C>T\tp.G310G (splice site)\t 10\t74\t98\n",
      "8\t10 / M\tc.622C>T\tp. Q208*\tc.930C>T\tp.G310G (splice site)\t 9\t81\t94\n",
      "9\t16 / M\tc.347C>G\tp.S116*\tc.881+5G>T\tIVS5+5G>T\t32\t2\t28\n",
      "10\t17 / F\tc.1201A>T\tp.R401*\tc.1201A>T\tp.R401*\t25\t16\t42\n",
      "11\t18 / F\tc.1201A>T\tp.R401*\tc.1201A>T\tp.R401*\t52\t2\t24\n",
      "12\t21 / F\tc.1370dupG\tp.R458Kfs*14\tc.1370dupG\tp.R458Kfs*14\tND\tND\t37\n",
      "Mean\t9\t-\t-\t-\t-\t28\t23\t51\n",
      "SEM\t2\t-\t-\t-\t-\t4\t10\t7\n",
      "Abbreviations: IQ, intellectual quotient; DQ, developmental quotient; ND, not determined; M, male; F, female.\n",
      " aThe following list specifies each mutation and its associated dbSNP ID and clinVar ID in the format (mutation, dbSNP ID, clinVar ID): (c1201A>T, rs201337954, 65632), (c.931G>A, rs201791209, none), (c.1891delC, rs587776982, 65631), (c.622C>T, rs200561967, none), and (c.1370dupG, rs587777265, 131954). If a particular mutation does not have both a dbSNP ID and a clinVar ID, it is not listed. Gray shading indicates alleles that fall within the transglutaminase catalytic domain.\n",
      " Nijmegen Pediatric CDG Rating Scale total score: Mild 0–14, Moderate 15–25, Severe ≥26.\n",
      " Vineland Adaptive Behavior Scales composite score; in a normal population the mean is 100 and standard deviation is 15.\n",
      "--- END TEXT ---\n",
      "\n",
      "Prompt response:\n",
      "subject                subject_entity   predicate        object            object_entity\n",
      "c.953T>C               genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.L318P                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.1169G>C              genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.R390P                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.1201A>T              genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.R401*                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.931G>A               genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.E311K                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.730T>C               genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.W244R                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.1604G>A              genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.W535*                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.1910delT             genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.L637*                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.1891delC             genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.Q631S                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.622C>T               genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.Q208*                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.930C>T               genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.G310G (splice site)  protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.347C>G               genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.S116*                protein variant  associated with  NGLY1 deficiency  disease\n",
      "c.881+5G>T             genetic variant  associated with  NGLY1 deficiency  disease\n",
      "IVS5+5G>T              genetic variant  associated with  NGLY1 deficiency  disease\n",
      "c.1370dupG             genetic variant  associated with  NGLY1 deficiency  disease\n",
      "p.R458Kfs*14           protein variant  associated with  NGLY1 deficiency  disease\n"
     ]
    }
   ],
   "source": [
    "!echo \"Prompt text:\"\n",
    "!cat $log | sed -n \"5495,5518p\"\n",
    "!echo \"Prompt response:\"\n",
    "!cat $log | sed -n \"5541,5567p\" | column -s \"|\" -t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt text:\n",
      "--- BEGIN TEXT ---\n",
      "CSF Laboratory Results\n",
      "Nine subjects underwent lumbar puncture. CSF total protein and albumin concentrations, as well as the CSF/serum albumin ratios, were low in nearly all individuals (Supplementary Table S3). There was no correlation between age and CSF protein or albumin levels. In the two oldest subjects, CSF 5-hydroxyindolacetic acid (5-HIAA) and homovanillic acid (HVA) were decreased, suggesting neuronal loss. Neopterin levels were normal but decreased with age, and CSF tetrahydrobiopterin (BH4) levels were below the lower limit of normal in all but one subject tested; however, there was no correlation with age. CSF 5-HIAA, HVA, and BH4 levels strongly and directly correlated with brain atrophy. CSF lactate and amino acid levels were essentially normal (Supplementary Table S3). CSF leucocyte counts (0–4; normal 0–5), glucose concentrations (54–72 mg/dL; normal 40–70), 3-O-methyldopa concentrations (12–28 nM; normal <150) and 5-methyltetrahydrofolate concentrations (45– 85 nmol/L; normal 40–150) were unremarkable (data not shown).\n",
      "--- END TEXT ---\n",
      "\n",
      "Prompt response:\n",
      "subject           subject_entity  predicate        object                                        object_entity\n",
      "NGLY1 deficiency  disease         associated with  CSF total protein and albumin concentrations  chemical substance\n",
      "NGLY1 deficiency  disease         associated with  CSF/serum albumin ratios                      chemical substance\n",
      "NGLY1 deficiency  disease         associated with  CSF 5-hydroxyindolacetic acid (5-HIAA)        chemical substance\n",
      "NGLY1 deficiency  disease         associated with  homovanillic acid (HVA)                       chemical substance\n",
      "NGLY1 deficiency  disease         associated with  neuronal loss                                 phenotype\n",
      "NGLY1 deficiency  disease         associated with  Neopterin levels                              chemical substance\n",
      "NGLY1 deficiency  disease         associated with  CSF tetrahydrobiopterin (BH4) levels          chemical substance\n",
      "NGLY1 deficiency  disease         associated with  brain atrophy                                 phenotype\n",
      "NGLY1 deficiency  disease         associated with  CSF lactate and amino acid levels             chemical substance\n",
      "NGLY1 deficiency  disease         associated with  CSF leucocyte counts                          cell type\n",
      "NGLY1 deficiency  disease         associated with  glucose concentrations                        chemical substance\n",
      "NGLY1 deficiency  disease         associated with  3-O-methyldopa concentrations                 chemical substance\n",
      "NGLY1 deficiency  disease         associated with  5-methyltetrahydrofolate concentrations       chemical substance\n"
     ]
    }
   ],
   "source": [
    "!echo \"Prompt text:\"\n",
    "!cat $log | sed -n \"6331,6335p\"\n",
    "!echo \"Prompt response:\"\n",
    "!cat $log | sed -n \"6358,6371p\" | column -s \"|\" -t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prompt text:\n",
      "--- BEGIN TEXT ---\n",
      "Figure 3: Brain MRI findings.\n",
      " White matter lesions were seen in two of eleven individuals. A shows multiple lesions in the periventricular white matter, some of which were confluent. B shows a single lesion in the periventricular white matter. Both A and B illustrate cerebral atrophy and were performed using T2-weighting. Sulci are slightly prominent in both A and B, and prominent ventricles are visible in A. C illustrates high position of the cerebellar tonsils, and large foramen of Magendie and cisterna magna.\n",
      "--- END TEXT ---\n",
      "\n",
      "Prompt response:\n",
      "subject               subject_entity      predicate        object                        object_entity\n",
      "Brain MRI findings    phenotype           has participant  White matter lesions          phenotype\n",
      "White matter lesions  phenotype           located in       two of eleven individuals     organism\n",
      "White matter lesions  phenotype           located in       periventricular white matter  cellular component\n",
      "Cerebral atrophy      phenotype           has participant  Brain MRI findings            phenotype\n",
      "Cerebral atrophy      phenotype           located in       A and B                       organism\n",
      "Sulci                 cellular component  has attribute    slightly prominent            phenotype\n",
      "Ventricles            cellular component  has attribute    prominent                     phenotype\n",
      "Cerebellar tonsils    cellular component  has attribute    high position                 phenotype\n",
      "Foramen of Magendie   cellular component  has attribute    large                         phenotype\n",
      "Cisterna magna        cellular component  has attribute    large                         phenotype\n"
     ]
    }
   ],
   "source": [
    "!echo \"Prompt text:\"\n",
    "!cat $log | sed -n \"8776,8780p\"\n",
    "!echo \"Prompt response:\"\n",
    "!cat $log | sed -n \"8803,8813p\" | column -s \"|\" -t"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>subject_entity</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "      <th>object_entity</th>\n",
       "      <th>doc_id</th>\n",
       "      <th>doc_filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>cause</td>\n",
       "      <td>Inherited Disorder of the Endoplasmic Reticulum-Associated Degradation</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has gene product</td>\n",
       "      <td>Mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERAD pathway</td>\n",
       "      <td>pathway</td>\n",
       "      <td>responsible for</td>\n",
       "      <td>translocation of misfolded proteins</td>\n",
       "      <td>biological process</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERAD pathway</td>\n",
       "      <td>pathway</td>\n",
       "      <td>has participant</td>\n",
       "      <td>proteasome</td>\n",
       "      <td>molecular activity</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>N-glycanase 1</td>\n",
       "      <td>protein</td>\n",
       "      <td>involved in</td>\n",
       "      <td>ERAD pathway</td>\n",
       "      <td>pathway</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>causes</td>\n",
       "      <td>conjunctival injection</td>\n",
       "      <td>symptom</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>917</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>causes</td>\n",
       "      <td>limbal neovascularization</td>\n",
       "      <td>symptom</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>918</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>causes</td>\n",
       "      <td>corneal scarring</td>\n",
       "      <td>symptom</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>919</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>causes</td>\n",
       "      <td>severe dry eyes</td>\n",
       "      <td>symptom</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>causes</td>\n",
       "      <td>lagophthalmous</td>\n",
       "      <td>symptom</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>921 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              subject   subject_entity         predicate  \\\n",
       "0           Mutations  genetic variant             cause   \n",
       "1               NGLY1             gene  has gene product   \n",
       "2        ERAD pathway          pathway   responsible for   \n",
       "3        ERAD pathway          pathway   has participant   \n",
       "4       N-glycanase 1          protein       involved in   \n",
       "..                ...              ...               ...   \n",
       "916  NGLY1 deficiency          disease            causes   \n",
       "917  NGLY1 deficiency          disease            causes   \n",
       "918  NGLY1 deficiency          disease            causes   \n",
       "919  NGLY1 deficiency          disease            causes   \n",
       "920  NGLY1 deficiency          disease            causes   \n",
       "\n",
       "                                                                     object  \\\n",
       "0    Inherited Disorder of the Endoplasmic Reticulum-Associated Degradation   \n",
       "1                                                                 Mutations   \n",
       "2                                       translocation of misfolded proteins   \n",
       "3                                                                proteasome   \n",
       "4                                                              ERAD pathway   \n",
       "..                                                                      ...   \n",
       "916                                                  conjunctival injection   \n",
       "917                                               limbal neovascularization   \n",
       "918                                                        corneal scarring   \n",
       "919                                                         severe dry eyes   \n",
       "920                                                          lagophthalmous   \n",
       "\n",
       "          object_entity      doc_id    doc_filename  \n",
       "0               disease  PMC4243708  PMC4243708.txt  \n",
       "1       genetic variant  PMC4243708  PMC4243708.txt  \n",
       "2    biological process  PMC4243708  PMC4243708.txt  \n",
       "3    molecular activity  PMC4243708  PMC4243708.txt  \n",
       "4               pathway  PMC4243708  PMC4243708.txt  \n",
       "..                  ...         ...             ...  \n",
       "916             symptom  PMC7477955  PMC7477955.txt  \n",
       "917             symptom  PMC7477955  PMC7477955.txt  \n",
       "918             symptom  PMC7477955  PMC7477955.txt  \n",
       "919             symptom  PMC7477955  PMC7477955.txt  \n",
       "920             symptom  PMC7477955  PMC7477955.txt  \n",
       "\n",
       "[921 rows x 7 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relations = pd.read_csv(utils.get_paths().output_data / \"relations_2.tsv\", sep=\"\\t\")\n",
    "relations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option(\"display.max_colwidth\", None, \"display.max_rows\", 400, \"display.max_columns\", None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_4a06a_row0_col0, #T_4a06a_row0_col1, #T_4a06a_row0_col3, #T_4a06a_row0_col6, #T_4a06a_row0_col9, #T_4a06a_row0_col12, #T_4a06a_row0_col13, #T_4a06a_row1_col22, #T_4a06a_row2_col2, #T_4a06a_row2_col5, #T_4a06a_row2_col8, #T_4a06a_row2_col15, #T_4a06a_row2_col19, #T_4a06a_row3_col11, #T_4a06a_row3_col16, #T_4a06a_row4_col10, #T_4a06a_row4_col22, #T_4a06a_row5_col4, #T_4a06a_row5_col10, #T_4a06a_row5_col14, #T_4a06a_row5_col17, #T_4a06a_row6_col6, #T_4a06a_row6_col18, #T_4a06a_row7_col20, #T_4a06a_row8_col21, #T_4a06a_row11_col7, #T_4a06a_row13_col19 {\n",
       "  background-color: #023858;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row0_col2 {\n",
       "  background-color: #05659f;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row0_col4 {\n",
       "  background-color: #5ea0ca;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row0_col5, #T_4a06a_row0_col8, #T_4a06a_row0_col10, #T_4a06a_row0_col11, #T_4a06a_row0_col14, #T_4a06a_row0_col15, #T_4a06a_row0_col16, #T_4a06a_row0_col18, #T_4a06a_row0_col19, #T_4a06a_row0_col20, #T_4a06a_row0_col21, #T_4a06a_row0_col22, #T_4a06a_row1_col3, #T_4a06a_row1_col4, #T_4a06a_row1_col5, #T_4a06a_row1_col6, #T_4a06a_row1_col7, #T_4a06a_row1_col8, #T_4a06a_row1_col9, #T_4a06a_row1_col10, #T_4a06a_row1_col11, #T_4a06a_row1_col12, #T_4a06a_row1_col13, #T_4a06a_row1_col14, #T_4a06a_row1_col15, #T_4a06a_row1_col16, #T_4a06a_row1_col17, #T_4a06a_row1_col18, #T_4a06a_row1_col19, #T_4a06a_row1_col20, #T_4a06a_row1_col21, #T_4a06a_row2_col4, #T_4a06a_row2_col9, #T_4a06a_row2_col12, #T_4a06a_row2_col13, #T_4a06a_row2_col14, #T_4a06a_row2_col16, #T_4a06a_row2_col17, #T_4a06a_row2_col18, #T_4a06a_row2_col20, #T_4a06a_row2_col21, #T_4a06a_row2_col22, #T_4a06a_row3_col14, #T_4a06a_row3_col15, #T_4a06a_row3_col17, #T_4a06a_row3_col18, #T_4a06a_row3_col19, #T_4a06a_row3_col20, #T_4a06a_row3_col21, #T_4a06a_row3_col22, #T_4a06a_row4_col3, #T_4a06a_row4_col5, #T_4a06a_row4_col7, #T_4a06a_row4_col9, #T_4a06a_row4_col12, #T_4a06a_row4_col13, #T_4a06a_row4_col16, #T_4a06a_row4_col17, #T_4a06a_row4_col18, #T_4a06a_row4_col19, #T_4a06a_row4_col20, #T_4a06a_row4_col21, #T_4a06a_row5_col0, #T_4a06a_row5_col1, #T_4a06a_row5_col2, #T_4a06a_row5_col3, #T_4a06a_row5_col5, #T_4a06a_row5_col6, #T_4a06a_row5_col9, #T_4a06a_row5_col13, #T_4a06a_row5_col15, #T_4a06a_row5_col16, #T_4a06a_row5_col18, #T_4a06a_row5_col19, #T_4a06a_row5_col20, #T_4a06a_row5_col21, #T_4a06a_row5_col22, #T_4a06a_row6_col3, #T_4a06a_row6_col4, #T_4a06a_row6_col8, #T_4a06a_row6_col9, #T_4a06a_row6_col10, #T_4a06a_row6_col11, #T_4a06a_row6_col12, #T_4a06a_row6_col13, #T_4a06a_row6_col14, #T_4a06a_row6_col15, #T_4a06a_row6_col16, #T_4a06a_row6_col17, #T_4a06a_row6_col19, #T_4a06a_row6_col20, #T_4a06a_row6_col21, #T_4a06a_row6_col22, #T_4a06a_row7_col0, #T_4a06a_row7_col1, #T_4a06a_row7_col4, #T_4a06a_row7_col6, #T_4a06a_row7_col7, #T_4a06a_row7_col8, #T_4a06a_row7_col9, #T_4a06a_row7_col10, #T_4a06a_row7_col11, #T_4a06a_row7_col12, #T_4a06a_row7_col13, #T_4a06a_row7_col14, #T_4a06a_row7_col15, #T_4a06a_row7_col16, #T_4a06a_row7_col17, #T_4a06a_row7_col18, #T_4a06a_row7_col19, #T_4a06a_row7_col21, #T_4a06a_row7_col22, #T_4a06a_row8_col3, #T_4a06a_row8_col4, #T_4a06a_row8_col5, #T_4a06a_row8_col7, #T_4a06a_row8_col9, #T_4a06a_row8_col10, #T_4a06a_row8_col13, #T_4a06a_row8_col14, #T_4a06a_row8_col15, #T_4a06a_row8_col16, #T_4a06a_row8_col18, #T_4a06a_row8_col19, #T_4a06a_row8_col20, #T_4a06a_row8_col22, #T_4a06a_row9_col1, #T_4a06a_row9_col5, #T_4a06a_row9_col6, #T_4a06a_row9_col7, #T_4a06a_row9_col9, #T_4a06a_row9_col10, #T_4a06a_row9_col12, #T_4a06a_row9_col13, #T_4a06a_row9_col14, #T_4a06a_row9_col15, #T_4a06a_row9_col16, #T_4a06a_row9_col17, #T_4a06a_row9_col18, #T_4a06a_row9_col19, #T_4a06a_row9_col20, #T_4a06a_row9_col21, #T_4a06a_row9_col22, #T_4a06a_row10_col1, #T_4a06a_row10_col2, #T_4a06a_row10_col3, #T_4a06a_row10_col4, #T_4a06a_row10_col6, #T_4a06a_row10_col7, #T_4a06a_row10_col8, #T_4a06a_row10_col10, #T_4a06a_row10_col11, #T_4a06a_row10_col13, #T_4a06a_row10_col14, #T_4a06a_row10_col15, #T_4a06a_row10_col16, #T_4a06a_row10_col17, #T_4a06a_row10_col18, #T_4a06a_row10_col19, #T_4a06a_row10_col20, #T_4a06a_row10_col21, #T_4a06a_row10_col22, #T_4a06a_row11_col0, #T_4a06a_row11_col1, #T_4a06a_row11_col2, #T_4a06a_row11_col3, #T_4a06a_row11_col6, #T_4a06a_row11_col8, #T_4a06a_row11_col9, #T_4a06a_row11_col11, #T_4a06a_row11_col12, #T_4a06a_row11_col13, #T_4a06a_row11_col14, #T_4a06a_row11_col15, #T_4a06a_row11_col17, #T_4a06a_row11_col18, #T_4a06a_row11_col19, #T_4a06a_row11_col20, #T_4a06a_row11_col21, #T_4a06a_row11_col22, #T_4a06a_row12_col3, #T_4a06a_row12_col5, #T_4a06a_row12_col6, #T_4a06a_row12_col9, #T_4a06a_row12_col10, #T_4a06a_row12_col11, #T_4a06a_row12_col12, #T_4a06a_row12_col13, #T_4a06a_row12_col14, #T_4a06a_row12_col15, #T_4a06a_row12_col16, #T_4a06a_row12_col17, #T_4a06a_row12_col18, #T_4a06a_row12_col19, #T_4a06a_row12_col20, #T_4a06a_row12_col21, #T_4a06a_row12_col22, #T_4a06a_row13_col0, #T_4a06a_row13_col1, #T_4a06a_row13_col3, #T_4a06a_row13_col4, #T_4a06a_row13_col5, #T_4a06a_row13_col6, #T_4a06a_row13_col7, #T_4a06a_row13_col8, #T_4a06a_row13_col9, #T_4a06a_row13_col10, #T_4a06a_row13_col11, #T_4a06a_row13_col12, #T_4a06a_row13_col13, #T_4a06a_row13_col15, #T_4a06a_row13_col16, #T_4a06a_row13_col17, #T_4a06a_row13_col18, #T_4a06a_row13_col20, #T_4a06a_row13_col21, #T_4a06a_row13_col22, #T_4a06a_row14_col0, #T_4a06a_row14_col1, #T_4a06a_row14_col6, #T_4a06a_row14_col7, #T_4a06a_row14_col8, #T_4a06a_row14_col9, #T_4a06a_row14_col10, #T_4a06a_row14_col11, #T_4a06a_row14_col12, #T_4a06a_row14_col13, #T_4a06a_row14_col14, #T_4a06a_row14_col15, #T_4a06a_row14_col16, #T_4a06a_row14_col17, #T_4a06a_row14_col18, #T_4a06a_row14_col19, #T_4a06a_row14_col20, #T_4a06a_row14_col22, #T_4a06a_row15_col0, #T_4a06a_row15_col1, #T_4a06a_row15_col2, #T_4a06a_row15_col3, #T_4a06a_row15_col5, #T_4a06a_row15_col6, #T_4a06a_row15_col8, #T_4a06a_row15_col9, #T_4a06a_row15_col12, #T_4a06a_row15_col13, #T_4a06a_row15_col15, #T_4a06a_row15_col16, #T_4a06a_row15_col17, #T_4a06a_row15_col18, #T_4a06a_row15_col19, #T_4a06a_row15_col20, #T_4a06a_row15_col21, #T_4a06a_row15_col22, #T_4a06a_row16_col0, #T_4a06a_row16_col2, #T_4a06a_row16_col3, #T_4a06a_row16_col4, #T_4a06a_row16_col5, #T_4a06a_row16_col6, #T_4a06a_row16_col7, #T_4a06a_row16_col8, #T_4a06a_row16_col9, #T_4a06a_row16_col10, #T_4a06a_row16_col11, #T_4a06a_row16_col12, #T_4a06a_row16_col13, #T_4a06a_row16_col14, #T_4a06a_row16_col15, #T_4a06a_row16_col16, #T_4a06a_row16_col17, #T_4a06a_row16_col18, #T_4a06a_row16_col19, #T_4a06a_row16_col20, #T_4a06a_row16_col21, #T_4a06a_row16_col22, #T_4a06a_row17_col0, #T_4a06a_row17_col1, #T_4a06a_row17_col2, #T_4a06a_row17_col3, #T_4a06a_row17_col4, #T_4a06a_row17_col5, #T_4a06a_row17_col7, #T_4a06a_row17_col8, #T_4a06a_row17_col9, #T_4a06a_row17_col11, #T_4a06a_row17_col12, #T_4a06a_row17_col13, #T_4a06a_row17_col14, #T_4a06a_row17_col15, #T_4a06a_row17_col16, #T_4a06a_row17_col17, #T_4a06a_row17_col18, #T_4a06a_row17_col19, #T_4a06a_row17_col20, #T_4a06a_row17_col21, #T_4a06a_row17_col22, #T_4a06a_row18_col0, #T_4a06a_row18_col1, #T_4a06a_row18_col2, #T_4a06a_row18_col3, #T_4a06a_row18_col4, #T_4a06a_row18_col5, #T_4a06a_row18_col7, #T_4a06a_row18_col8, #T_4a06a_row18_col9, #T_4a06a_row18_col10, #T_4a06a_row18_col11, #T_4a06a_row18_col12, #T_4a06a_row18_col13, #T_4a06a_row18_col14, #T_4a06a_row18_col15, #T_4a06a_row18_col16, #T_4a06a_row18_col17, #T_4a06a_row18_col18, #T_4a06a_row18_col19, #T_4a06a_row18_col20, #T_4a06a_row18_col21, #T_4a06a_row18_col22, #T_4a06a_row19_col0, #T_4a06a_row19_col2, #T_4a06a_row19_col3, #T_4a06a_row19_col4, #T_4a06a_row19_col5, #T_4a06a_row19_col6, #T_4a06a_row19_col7, #T_4a06a_row19_col8, #T_4a06a_row19_col9, #T_4a06a_row19_col10, #T_4a06a_row19_col11, #T_4a06a_row19_col12, #T_4a06a_row19_col13, #T_4a06a_row19_col14, #T_4a06a_row19_col15, #T_4a06a_row19_col16, #T_4a06a_row19_col17, #T_4a06a_row19_col18, #T_4a06a_row19_col19, #T_4a06a_row19_col20, #T_4a06a_row19_col21, #T_4a06a_row19_col22, #T_4a06a_row20_col0, #T_4a06a_row20_col1, #T_4a06a_row20_col2, #T_4a06a_row20_col3, #T_4a06a_row20_col4, #T_4a06a_row20_col5, #T_4a06a_row20_col6, #T_4a06a_row20_col7, #T_4a06a_row20_col8, #T_4a06a_row20_col9, #T_4a06a_row20_col10, #T_4a06a_row20_col11, #T_4a06a_row20_col12, #T_4a06a_row20_col13, #T_4a06a_row20_col14, #T_4a06a_row20_col15, #T_4a06a_row20_col16, #T_4a06a_row20_col17, #T_4a06a_row20_col18, #T_4a06a_row20_col19, #T_4a06a_row20_col20, #T_4a06a_row20_col21, #T_4a06a_row20_col22 {\n",
       "  background-color: #fff7fb;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row0_col7, #T_4a06a_row2_col7, #T_4a06a_row6_col7, #T_4a06a_row12_col7, #T_4a06a_row15_col7 {\n",
       "  background-color: #ece7f2;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row0_col17, #T_4a06a_row3_col5, #T_4a06a_row4_col11, #T_4a06a_row4_col14, #T_4a06a_row8_col11, #T_4a06a_row8_col17, #T_4a06a_row9_col11, #T_4a06a_row13_col14, #T_4a06a_row15_col11, #T_4a06a_row15_col14, #T_4a06a_row17_col6 {\n",
       "  background-color: #b4c4df;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row1_col0 {\n",
       "  background-color: #d2d2e7;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row1_col1, #T_4a06a_row16_col1 {\n",
       "  background-color: #ebe6f2;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row1_col2, #T_4a06a_row3_col8 {\n",
       "  background-color: #91b5d6;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row2_col0, #T_4a06a_row8_col2, #T_4a06a_row9_col2, #T_4a06a_row12_col1, #T_4a06a_row12_col2, #T_4a06a_row14_col2 {\n",
       "  background-color: #f8f1f8;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row2_col1 {\n",
       "  background-color: #69a5cc;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row2_col3, #T_4a06a_row3_col3, #T_4a06a_row6_col0, #T_4a06a_row9_col3 {\n",
       "  background-color: #fbf4f9;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row2_col6, #T_4a06a_row3_col7, #T_4a06a_row5_col7, #T_4a06a_row11_col10, #T_4a06a_row14_col21 {\n",
       "  background-color: #73a9cf;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row2_col10, #T_4a06a_row3_col10, #T_4a06a_row3_col12, #T_4a06a_row5_col12, #T_4a06a_row8_col12, #T_4a06a_row11_col16, #T_4a06a_row15_col10, #T_4a06a_row17_col10 {\n",
       "  background-color: #d0d1e6;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row2_col11, #T_4a06a_row3_col6, #T_4a06a_row3_col13, #T_4a06a_row4_col15, #T_4a06a_row5_col11 {\n",
       "  background-color: #2685bb;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row3_col0 {\n",
       "  background-color: #fbf3f9;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row3_col1 {\n",
       "  background-color: #03446a;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row3_col2, #T_4a06a_row4_col8, #T_4a06a_row8_col8, #T_4a06a_row9_col8, #T_4a06a_row12_col8, #T_4a06a_row13_col2 {\n",
       "  background-color: #e8e4f0;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row3_col4, #T_4a06a_row3_col9 {\n",
       "  background-color: #c9cee4;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row4_col0, #T_4a06a_row10_col0 {\n",
       "  background-color: #fdf5fa;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row4_col1, #T_4a06a_row8_col1 {\n",
       "  background-color: #afc1dd;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row4_col2 {\n",
       "  background-color: #67a4cc;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row4_col4 {\n",
       "  background-color: #a9bfdc;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row4_col6, #T_4a06a_row8_col6, #T_4a06a_row18_col6 {\n",
       "  background-color: #e3e0ee;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row5_col8 {\n",
       "  background-color: #c4cbe3;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row6_col1 {\n",
       "  background-color: #e1dfed;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row6_col2 {\n",
       "  background-color: #f1ebf4;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row6_col5 {\n",
       "  background-color: #d7d6e9;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row7_col2 {\n",
       "  background-color: #509ac6;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row7_col3 {\n",
       "  background-color: #d3d4e7;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row7_col5 {\n",
       "  background-color: #8bb2d4;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row8_col0 {\n",
       "  background-color: #fcf4fa;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row9_col0 {\n",
       "  background-color: #faf2f8;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row9_col4, #T_4a06a_row10_col9, #T_4a06a_row11_col4, #T_4a06a_row12_col4, #T_4a06a_row14_col4 {\n",
       "  background-color: #f1ebf5;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row10_col5 {\n",
       "  background-color: #589ec8;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row10_col12 {\n",
       "  background-color: #056faf;\n",
       "  color: #f1f1f1;\n",
       "}\n",
       "#T_4a06a_row11_col5, #T_4a06a_row14_col5 {\n",
       "  background-color: #eee9f3;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row12_col0 {\n",
       "  background-color: #fef6fa;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row14_col3 {\n",
       "  background-color: #f3edf5;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row15_col4 {\n",
       "  background-color: #e0dded;\n",
       "  color: #000000;\n",
       "}\n",
       "#T_4a06a_row19_col1 {\n",
       "  background-color: #f2ecf5;\n",
       "  color: #000000;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_4a06a\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"index_name level0\" >object_entity</th>\n",
       "      <th id=\"T_4a06a_level0_col0\" class=\"col_heading level0 col0\" >phenotype</th>\n",
       "      <th id=\"T_4a06a_level0_col1\" class=\"col_heading level0 col1\" >disease</th>\n",
       "      <th id=\"T_4a06a_level0_col2\" class=\"col_heading level0 col2\" >genetic variant</th>\n",
       "      <th id=\"T_4a06a_level0_col3\" class=\"col_heading level0 col3\" >assay</th>\n",
       "      <th id=\"T_4a06a_level0_col4\" class=\"col_heading level0 col4\" >biological process</th>\n",
       "      <th id=\"T_4a06a_level0_col5\" class=\"col_heading level0 col5\" >other</th>\n",
       "      <th id=\"T_4a06a_level0_col6\" class=\"col_heading level0 col6\" >organism</th>\n",
       "      <th id=\"T_4a06a_level0_col7\" class=\"col_heading level0 col7\" >cellular component</th>\n",
       "      <th id=\"T_4a06a_level0_col8\" class=\"col_heading level0 col8\" >gene</th>\n",
       "      <th id=\"T_4a06a_level0_col9\" class=\"col_heading level0 col9\" >chemical substance</th>\n",
       "      <th id=\"T_4a06a_level0_col10\" class=\"col_heading level0 col10\" >protein</th>\n",
       "      <th id=\"T_4a06a_level0_col11\" class=\"col_heading level0 col11\" >molecular activity</th>\n",
       "      <th id=\"T_4a06a_level0_col12\" class=\"col_heading level0 col12\" >cell type</th>\n",
       "      <th id=\"T_4a06a_level0_col13\" class=\"col_heading level0 col13\" >symptom</th>\n",
       "      <th id=\"T_4a06a_level0_col14\" class=\"col_heading level0 col14\" >macromolecular complex</th>\n",
       "      <th id=\"T_4a06a_level0_col15\" class=\"col_heading level0 col15\" >protein variant</th>\n",
       "      <th id=\"T_4a06a_level0_col16\" class=\"col_heading level0 col16\" >metabolite</th>\n",
       "      <th id=\"T_4a06a_level0_col17\" class=\"col_heading level0 col17\" >pathway</th>\n",
       "      <th id=\"T_4a06a_level0_col18\" class=\"col_heading level0 col18\" >organization</th>\n",
       "      <th id=\"T_4a06a_level0_col19\" class=\"col_heading level0 col19\" >transcript variant</th>\n",
       "      <th id=\"T_4a06a_level0_col20\" class=\"col_heading level0 col20\" >clinical trial</th>\n",
       "      <th id=\"T_4a06a_level0_col21\" class=\"col_heading level0 col21\" >tissue</th>\n",
       "      <th id=\"T_4a06a_level0_col22\" class=\"col_heading level0 col22\" >genotype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th class=\"index_name level0\" >subject_entity</th>\n",
       "      <th class=\"blank col0\" >&nbsp;</th>\n",
       "      <th class=\"blank col1\" >&nbsp;</th>\n",
       "      <th class=\"blank col2\" >&nbsp;</th>\n",
       "      <th class=\"blank col3\" >&nbsp;</th>\n",
       "      <th class=\"blank col4\" >&nbsp;</th>\n",
       "      <th class=\"blank col5\" >&nbsp;</th>\n",
       "      <th class=\"blank col6\" >&nbsp;</th>\n",
       "      <th class=\"blank col7\" >&nbsp;</th>\n",
       "      <th class=\"blank col8\" >&nbsp;</th>\n",
       "      <th class=\"blank col9\" >&nbsp;</th>\n",
       "      <th class=\"blank col10\" >&nbsp;</th>\n",
       "      <th class=\"blank col11\" >&nbsp;</th>\n",
       "      <th class=\"blank col12\" >&nbsp;</th>\n",
       "      <th class=\"blank col13\" >&nbsp;</th>\n",
       "      <th class=\"blank col14\" >&nbsp;</th>\n",
       "      <th class=\"blank col15\" >&nbsp;</th>\n",
       "      <th class=\"blank col16\" >&nbsp;</th>\n",
       "      <th class=\"blank col17\" >&nbsp;</th>\n",
       "      <th class=\"blank col18\" >&nbsp;</th>\n",
       "      <th class=\"blank col19\" >&nbsp;</th>\n",
       "      <th class=\"blank col20\" >&nbsp;</th>\n",
       "      <th class=\"blank col21\" >&nbsp;</th>\n",
       "      <th class=\"blank col22\" >&nbsp;</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row0\" class=\"row_heading level0 row0\" >disease</th>\n",
       "      <td id=\"T_4a06a_row0_col0\" class=\"data row0 col0\" >336</td>\n",
       "      <td id=\"T_4a06a_row0_col1\" class=\"data row0 col1\" >23</td>\n",
       "      <td id=\"T_4a06a_row0_col2\" class=\"data row0 col2\" >17</td>\n",
       "      <td id=\"T_4a06a_row0_col3\" class=\"data row0 col3\" >38</td>\n",
       "      <td id=\"T_4a06a_row0_col4\" class=\"data row0 col4\" >6</td>\n",
       "      <td id=\"T_4a06a_row0_col5\" class=\"data row0 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col6\" class=\"data row0 col6\" >6</td>\n",
       "      <td id=\"T_4a06a_row0_col7\" class=\"data row0 col7\" >1</td>\n",
       "      <td id=\"T_4a06a_row0_col8\" class=\"data row0 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col9\" class=\"data row0 col9\" >11</td>\n",
       "      <td id=\"T_4a06a_row0_col10\" class=\"data row0 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col11\" class=\"data row0 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col12\" class=\"data row0 col12\" >4</td>\n",
       "      <td id=\"T_4a06a_row0_col13\" class=\"data row0 col13\" >6</td>\n",
       "      <td id=\"T_4a06a_row0_col14\" class=\"data row0 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col15\" class=\"data row0 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col16\" class=\"data row0 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col17\" class=\"data row0 col17\" >1</td>\n",
       "      <td id=\"T_4a06a_row0_col18\" class=\"data row0 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col19\" class=\"data row0 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col20\" class=\"data row0 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col21\" class=\"data row0 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row0_col22\" class=\"data row0 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row1\" class=\"row_heading level0 row1\" >organism</th>\n",
       "      <td id=\"T_4a06a_row1_col0\" class=\"data row1 col0\" >82</td>\n",
       "      <td id=\"T_4a06a_row1_col1\" class=\"data row1 col1\" >3</td>\n",
       "      <td id=\"T_4a06a_row1_col2\" class=\"data row1 col2\" >9</td>\n",
       "      <td id=\"T_4a06a_row1_col3\" class=\"data row1 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col4\" class=\"data row1 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col5\" class=\"data row1 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col6\" class=\"data row1 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col7\" class=\"data row1 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col8\" class=\"data row1 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col9\" class=\"data row1 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col10\" class=\"data row1 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col11\" class=\"data row1 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col12\" class=\"data row1 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col13\" class=\"data row1 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col14\" class=\"data row1 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col15\" class=\"data row1 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col16\" class=\"data row1 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col17\" class=\"data row1 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col18\" class=\"data row1 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col19\" class=\"data row1 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col20\" class=\"data row1 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col21\" class=\"data row1 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row1_col22\" class=\"data row1 col22\" >1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row2\" class=\"row_heading level0 row2\" >genetic variant</th>\n",
       "      <td id=\"T_4a06a_row2_col0\" class=\"data row2 col0\" >15</td>\n",
       "      <td id=\"T_4a06a_row2_col1\" class=\"data row2 col1\" >12</td>\n",
       "      <td id=\"T_4a06a_row2_col2\" class=\"data row2 col2\" >21</td>\n",
       "      <td id=\"T_4a06a_row2_col3\" class=\"data row2 col3\" >1</td>\n",
       "      <td id=\"T_4a06a_row2_col4\" class=\"data row2 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col5\" class=\"data row2 col5\" >9</td>\n",
       "      <td id=\"T_4a06a_row2_col6\" class=\"data row2 col6\" >3</td>\n",
       "      <td id=\"T_4a06a_row2_col7\" class=\"data row2 col7\" >1</td>\n",
       "      <td id=\"T_4a06a_row2_col8\" class=\"data row2 col8\" >7</td>\n",
       "      <td id=\"T_4a06a_row2_col9\" class=\"data row2 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col10\" class=\"data row2 col10\" >1</td>\n",
       "      <td id=\"T_4a06a_row2_col11\" class=\"data row2 col11\" >2</td>\n",
       "      <td id=\"T_4a06a_row2_col12\" class=\"data row2 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col13\" class=\"data row2 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col14\" class=\"data row2 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col15\" class=\"data row2 col15\" >3</td>\n",
       "      <td id=\"T_4a06a_row2_col16\" class=\"data row2 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col17\" class=\"data row2 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col18\" class=\"data row2 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col19\" class=\"data row2 col19\" >2</td>\n",
       "      <td id=\"T_4a06a_row2_col20\" class=\"data row2 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col21\" class=\"data row2 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row2_col22\" class=\"data row2 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row3\" class=\"row_heading level0 row3\" >phenotype</th>\n",
       "      <td id=\"T_4a06a_row3_col0\" class=\"data row3 col0\" >10</td>\n",
       "      <td id=\"T_4a06a_row3_col1\" class=\"data row3 col1\" >22</td>\n",
       "      <td id=\"T_4a06a_row3_col2\" class=\"data row3 col2\" >3</td>\n",
       "      <td id=\"T_4a06a_row3_col3\" class=\"data row3 col3\" >1</td>\n",
       "      <td id=\"T_4a06a_row3_col4\" class=\"data row3 col4\" >3</td>\n",
       "      <td id=\"T_4a06a_row3_col5\" class=\"data row3 col5\" >3</td>\n",
       "      <td id=\"T_4a06a_row3_col6\" class=\"data row3 col6\" >4</td>\n",
       "      <td id=\"T_4a06a_row3_col7\" class=\"data row3 col7\" >4</td>\n",
       "      <td id=\"T_4a06a_row3_col8\" class=\"data row3 col8\" >3</td>\n",
       "      <td id=\"T_4a06a_row3_col9\" class=\"data row3 col9\" >3</td>\n",
       "      <td id=\"T_4a06a_row3_col10\" class=\"data row3 col10\" >1</td>\n",
       "      <td id=\"T_4a06a_row3_col11\" class=\"data row3 col11\" >3</td>\n",
       "      <td id=\"T_4a06a_row3_col12\" class=\"data row3 col12\" >1</td>\n",
       "      <td id=\"T_4a06a_row3_col13\" class=\"data row3 col13\" >4</td>\n",
       "      <td id=\"T_4a06a_row3_col14\" class=\"data row3 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row3_col15\" class=\"data row3 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row3_col16\" class=\"data row3 col16\" >4</td>\n",
       "      <td id=\"T_4a06a_row3_col17\" class=\"data row3 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row3_col18\" class=\"data row3 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row3_col19\" class=\"data row3 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row3_col20\" class=\"data row3 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row3_col21\" class=\"data row3 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row3_col22\" class=\"data row3 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row4\" class=\"row_heading level0 row4\" >gene</th>\n",
       "      <td id=\"T_4a06a_row4_col0\" class=\"data row4 col0\" >4</td>\n",
       "      <td id=\"T_4a06a_row4_col1\" class=\"data row4 col1\" >8</td>\n",
       "      <td id=\"T_4a06a_row4_col2\" class=\"data row4 col2\" >11</td>\n",
       "      <td id=\"T_4a06a_row4_col3\" class=\"data row4 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col4\" class=\"data row4 col4\" >4</td>\n",
       "      <td id=\"T_4a06a_row4_col5\" class=\"data row4 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col6\" class=\"data row4 col6\" >1</td>\n",
       "      <td id=\"T_4a06a_row4_col7\" class=\"data row4 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col8\" class=\"data row4 col8\" >1</td>\n",
       "      <td id=\"T_4a06a_row4_col9\" class=\"data row4 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col10\" class=\"data row4 col10\" >4</td>\n",
       "      <td id=\"T_4a06a_row4_col11\" class=\"data row4 col11\" >1</td>\n",
       "      <td id=\"T_4a06a_row4_col12\" class=\"data row4 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col13\" class=\"data row4 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col14\" class=\"data row4 col14\" >1</td>\n",
       "      <td id=\"T_4a06a_row4_col15\" class=\"data row4 col15\" >2</td>\n",
       "      <td id=\"T_4a06a_row4_col16\" class=\"data row4 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col17\" class=\"data row4 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col18\" class=\"data row4 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col19\" class=\"data row4 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col20\" class=\"data row4 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col21\" class=\"data row4 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row4_col22\" class=\"data row4 col22\" >1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row5\" class=\"row_heading level0 row5\" >protein</th>\n",
       "      <td id=\"T_4a06a_row5_col0\" class=\"data row5 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col1\" class=\"data row5 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col2\" class=\"data row5 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col3\" class=\"data row5 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col4\" class=\"data row5 col4\" >11</td>\n",
       "      <td id=\"T_4a06a_row5_col5\" class=\"data row5 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col6\" class=\"data row5 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col7\" class=\"data row5 col7\" >4</td>\n",
       "      <td id=\"T_4a06a_row5_col8\" class=\"data row5 col8\" >2</td>\n",
       "      <td id=\"T_4a06a_row5_col9\" class=\"data row5 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col10\" class=\"data row5 col10\" >4</td>\n",
       "      <td id=\"T_4a06a_row5_col11\" class=\"data row5 col11\" >2</td>\n",
       "      <td id=\"T_4a06a_row5_col12\" class=\"data row5 col12\" >1</td>\n",
       "      <td id=\"T_4a06a_row5_col13\" class=\"data row5 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col14\" class=\"data row5 col14\" >3</td>\n",
       "      <td id=\"T_4a06a_row5_col15\" class=\"data row5 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col16\" class=\"data row5 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col17\" class=\"data row5 col17\" >3</td>\n",
       "      <td id=\"T_4a06a_row5_col18\" class=\"data row5 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col19\" class=\"data row5 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col20\" class=\"data row5 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col21\" class=\"data row5 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row5_col22\" class=\"data row5 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row6\" class=\"row_heading level0 row6\" >assay</th>\n",
       "      <td id=\"T_4a06a_row6_col0\" class=\"data row6 col0\" >9</td>\n",
       "      <td id=\"T_4a06a_row6_col1\" class=\"data row6 col1\" >4</td>\n",
       "      <td id=\"T_4a06a_row6_col2\" class=\"data row6 col2\" >2</td>\n",
       "      <td id=\"T_4a06a_row6_col3\" class=\"data row6 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col4\" class=\"data row6 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col5\" class=\"data row6 col5\" >2</td>\n",
       "      <td id=\"T_4a06a_row6_col6\" class=\"data row6 col6\" >6</td>\n",
       "      <td id=\"T_4a06a_row6_col7\" class=\"data row6 col7\" >1</td>\n",
       "      <td id=\"T_4a06a_row6_col8\" class=\"data row6 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col9\" class=\"data row6 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col10\" class=\"data row6 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col11\" class=\"data row6 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col12\" class=\"data row6 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col13\" class=\"data row6 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col14\" class=\"data row6 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col15\" class=\"data row6 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col16\" class=\"data row6 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col17\" class=\"data row6 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col18\" class=\"data row6 col18\" >5</td>\n",
       "      <td id=\"T_4a06a_row6_col19\" class=\"data row6 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col20\" class=\"data row6 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col21\" class=\"data row6 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row6_col22\" class=\"data row6 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row7\" class=\"row_heading level0 row7\" >organization</th>\n",
       "      <td id=\"T_4a06a_row7_col0\" class=\"data row7 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col1\" class=\"data row7 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col2\" class=\"data row7 col2\" >12</td>\n",
       "      <td id=\"T_4a06a_row7_col3\" class=\"data row7 col3\" >9</td>\n",
       "      <td id=\"T_4a06a_row7_col4\" class=\"data row7 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col5\" class=\"data row7 col5\" >4</td>\n",
       "      <td id=\"T_4a06a_row7_col6\" class=\"data row7 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col7\" class=\"data row7 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col8\" class=\"data row7 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col9\" class=\"data row7 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col10\" class=\"data row7 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col11\" class=\"data row7 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col12\" class=\"data row7 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col13\" class=\"data row7 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col14\" class=\"data row7 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col15\" class=\"data row7 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col16\" class=\"data row7 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col17\" class=\"data row7 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col18\" class=\"data row7 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col19\" class=\"data row7 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col20\" class=\"data row7 col20\" >3</td>\n",
       "      <td id=\"T_4a06a_row7_col21\" class=\"data row7 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row7_col22\" class=\"data row7 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row8\" class=\"row_heading level0 row8\" >biological process</th>\n",
       "      <td id=\"T_4a06a_row8_col0\" class=\"data row8 col0\" >7</td>\n",
       "      <td id=\"T_4a06a_row8_col1\" class=\"data row8 col1\" >8</td>\n",
       "      <td id=\"T_4a06a_row8_col2\" class=\"data row8 col2\" >1</td>\n",
       "      <td id=\"T_4a06a_row8_col3\" class=\"data row8 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col4\" class=\"data row8 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col5\" class=\"data row8 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col6\" class=\"data row8 col6\" >1</td>\n",
       "      <td id=\"T_4a06a_row8_col7\" class=\"data row8 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col8\" class=\"data row8 col8\" >1</td>\n",
       "      <td id=\"T_4a06a_row8_col9\" class=\"data row8 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col10\" class=\"data row8 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col11\" class=\"data row8 col11\" >1</td>\n",
       "      <td id=\"T_4a06a_row8_col12\" class=\"data row8 col12\" >1</td>\n",
       "      <td id=\"T_4a06a_row8_col13\" class=\"data row8 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col14\" class=\"data row8 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col15\" class=\"data row8 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col16\" class=\"data row8 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col17\" class=\"data row8 col17\" >1</td>\n",
       "      <td id=\"T_4a06a_row8_col18\" class=\"data row8 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col19\" class=\"data row8 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col20\" class=\"data row8 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row8_col21\" class=\"data row8 col21\" >2</td>\n",
       "      <td id=\"T_4a06a_row8_col22\" class=\"data row8 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row9\" class=\"row_heading level0 row9\" >cellular component</th>\n",
       "      <td id=\"T_4a06a_row9_col0\" class=\"data row9 col0\" >12</td>\n",
       "      <td id=\"T_4a06a_row9_col1\" class=\"data row9 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col2\" class=\"data row9 col2\" >1</td>\n",
       "      <td id=\"T_4a06a_row9_col3\" class=\"data row9 col3\" >1</td>\n",
       "      <td id=\"T_4a06a_row9_col4\" class=\"data row9 col4\" >1</td>\n",
       "      <td id=\"T_4a06a_row9_col5\" class=\"data row9 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col6\" class=\"data row9 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col7\" class=\"data row9 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col8\" class=\"data row9 col8\" >1</td>\n",
       "      <td id=\"T_4a06a_row9_col9\" class=\"data row9 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col10\" class=\"data row9 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col11\" class=\"data row9 col11\" >1</td>\n",
       "      <td id=\"T_4a06a_row9_col12\" class=\"data row9 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col13\" class=\"data row9 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col14\" class=\"data row9 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col15\" class=\"data row9 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col16\" class=\"data row9 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col17\" class=\"data row9 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col18\" class=\"data row9 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col19\" class=\"data row9 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col20\" class=\"data row9 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col21\" class=\"data row9 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row9_col22\" class=\"data row9 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row10\" class=\"row_heading level0 row10\" >chemical substance</th>\n",
       "      <td id=\"T_4a06a_row10_col0\" class=\"data row10 col0\" >5</td>\n",
       "      <td id=\"T_4a06a_row10_col1\" class=\"data row10 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col2\" class=\"data row10 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col3\" class=\"data row10 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col4\" class=\"data row10 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col5\" class=\"data row10 col5\" >5</td>\n",
       "      <td id=\"T_4a06a_row10_col6\" class=\"data row10 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col7\" class=\"data row10 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col8\" class=\"data row10 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col9\" class=\"data row10 col9\" >1</td>\n",
       "      <td id=\"T_4a06a_row10_col10\" class=\"data row10 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col11\" class=\"data row10 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col12\" class=\"data row10 col12\" >3</td>\n",
       "      <td id=\"T_4a06a_row10_col13\" class=\"data row10 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col14\" class=\"data row10 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col15\" class=\"data row10 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col16\" class=\"data row10 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col17\" class=\"data row10 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col18\" class=\"data row10 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col19\" class=\"data row10 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col20\" class=\"data row10 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col21\" class=\"data row10 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row10_col22\" class=\"data row10 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row11\" class=\"row_heading level0 row11\" >metabolite</th>\n",
       "      <td id=\"T_4a06a_row11_col0\" class=\"data row11 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col1\" class=\"data row11 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col2\" class=\"data row11 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col3\" class=\"data row11 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col4\" class=\"data row11 col4\" >1</td>\n",
       "      <td id=\"T_4a06a_row11_col5\" class=\"data row11 col5\" >1</td>\n",
       "      <td id=\"T_4a06a_row11_col6\" class=\"data row11 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col7\" class=\"data row11 col7\" >8</td>\n",
       "      <td id=\"T_4a06a_row11_col8\" class=\"data row11 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col9\" class=\"data row11 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col10\" class=\"data row11 col10\" >2</td>\n",
       "      <td id=\"T_4a06a_row11_col11\" class=\"data row11 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col12\" class=\"data row11 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col13\" class=\"data row11 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col14\" class=\"data row11 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col15\" class=\"data row11 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col16\" class=\"data row11 col16\" >1</td>\n",
       "      <td id=\"T_4a06a_row11_col17\" class=\"data row11 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col18\" class=\"data row11 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col19\" class=\"data row11 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col20\" class=\"data row11 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col21\" class=\"data row11 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row11_col22\" class=\"data row11 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row12\" class=\"row_heading level0 row12\" >molecular activity</th>\n",
       "      <td id=\"T_4a06a_row12_col0\" class=\"data row12 col0\" >3</td>\n",
       "      <td id=\"T_4a06a_row12_col1\" class=\"data row12 col1\" >1</td>\n",
       "      <td id=\"T_4a06a_row12_col2\" class=\"data row12 col2\" >1</td>\n",
       "      <td id=\"T_4a06a_row12_col3\" class=\"data row12 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col4\" class=\"data row12 col4\" >1</td>\n",
       "      <td id=\"T_4a06a_row12_col5\" class=\"data row12 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col6\" class=\"data row12 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col7\" class=\"data row12 col7\" >1</td>\n",
       "      <td id=\"T_4a06a_row12_col8\" class=\"data row12 col8\" >1</td>\n",
       "      <td id=\"T_4a06a_row12_col9\" class=\"data row12 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col10\" class=\"data row12 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col11\" class=\"data row12 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col12\" class=\"data row12 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col13\" class=\"data row12 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col14\" class=\"data row12 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col15\" class=\"data row12 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col16\" class=\"data row12 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col17\" class=\"data row12 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col18\" class=\"data row12 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col19\" class=\"data row12 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col20\" class=\"data row12 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col21\" class=\"data row12 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row12_col22\" class=\"data row12 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row13\" class=\"row_heading level0 row13\" >protein variant</th>\n",
       "      <td id=\"T_4a06a_row13_col0\" class=\"data row13 col0\" >1</td>\n",
       "      <td id=\"T_4a06a_row13_col1\" class=\"data row13 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col2\" class=\"data row13 col2\" >3</td>\n",
       "      <td id=\"T_4a06a_row13_col3\" class=\"data row13 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col4\" class=\"data row13 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col5\" class=\"data row13 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col6\" class=\"data row13 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col7\" class=\"data row13 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col8\" class=\"data row13 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col9\" class=\"data row13 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col10\" class=\"data row13 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col11\" class=\"data row13 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col12\" class=\"data row13 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col13\" class=\"data row13 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col14\" class=\"data row13 col14\" >1</td>\n",
       "      <td id=\"T_4a06a_row13_col15\" class=\"data row13 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col16\" class=\"data row13 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col17\" class=\"data row13 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col18\" class=\"data row13 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col19\" class=\"data row13 col19\" >2</td>\n",
       "      <td id=\"T_4a06a_row13_col20\" class=\"data row13 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col21\" class=\"data row13 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row13_col22\" class=\"data row13 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row14\" class=\"row_heading level0 row14\" >macromolecular complex</th>\n",
       "      <td id=\"T_4a06a_row14_col0\" class=\"data row14 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col1\" class=\"data row14 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col2\" class=\"data row14 col2\" >1</td>\n",
       "      <td id=\"T_4a06a_row14_col3\" class=\"data row14 col3\" >3</td>\n",
       "      <td id=\"T_4a06a_row14_col4\" class=\"data row14 col4\" >1</td>\n",
       "      <td id=\"T_4a06a_row14_col5\" class=\"data row14 col5\" >1</td>\n",
       "      <td id=\"T_4a06a_row14_col6\" class=\"data row14 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col7\" class=\"data row14 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col8\" class=\"data row14 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col9\" class=\"data row14 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col10\" class=\"data row14 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col11\" class=\"data row14 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col12\" class=\"data row14 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col13\" class=\"data row14 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col14\" class=\"data row14 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col15\" class=\"data row14 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col16\" class=\"data row14 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col17\" class=\"data row14 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col18\" class=\"data row14 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col19\" class=\"data row14 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col20\" class=\"data row14 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row14_col21\" class=\"data row14 col21\" >1</td>\n",
       "      <td id=\"T_4a06a_row14_col22\" class=\"data row14 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row15\" class=\"row_heading level0 row15\" >pathway</th>\n",
       "      <td id=\"T_4a06a_row15_col0\" class=\"data row15 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col1\" class=\"data row15 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col2\" class=\"data row15 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col3\" class=\"data row15 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col4\" class=\"data row15 col4\" >2</td>\n",
       "      <td id=\"T_4a06a_row15_col5\" class=\"data row15 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col6\" class=\"data row15 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col7\" class=\"data row15 col7\" >1</td>\n",
       "      <td id=\"T_4a06a_row15_col8\" class=\"data row15 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col9\" class=\"data row15 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col10\" class=\"data row15 col10\" >1</td>\n",
       "      <td id=\"T_4a06a_row15_col11\" class=\"data row15 col11\" >1</td>\n",
       "      <td id=\"T_4a06a_row15_col12\" class=\"data row15 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col13\" class=\"data row15 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col14\" class=\"data row15 col14\" >1</td>\n",
       "      <td id=\"T_4a06a_row15_col15\" class=\"data row15 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col16\" class=\"data row15 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col17\" class=\"data row15 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col18\" class=\"data row15 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col19\" class=\"data row15 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col20\" class=\"data row15 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col21\" class=\"data row15 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row15_col22\" class=\"data row15 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row16\" class=\"row_heading level0 row16\" >clinical trial</th>\n",
       "      <td id=\"T_4a06a_row16_col0\" class=\"data row16 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col1\" class=\"data row16 col1\" >3</td>\n",
       "      <td id=\"T_4a06a_row16_col2\" class=\"data row16 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col3\" class=\"data row16 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col4\" class=\"data row16 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col5\" class=\"data row16 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col6\" class=\"data row16 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col7\" class=\"data row16 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col8\" class=\"data row16 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col9\" class=\"data row16 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col10\" class=\"data row16 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col11\" class=\"data row16 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col12\" class=\"data row16 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col13\" class=\"data row16 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col14\" class=\"data row16 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col15\" class=\"data row16 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col16\" class=\"data row16 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col17\" class=\"data row16 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col18\" class=\"data row16 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col19\" class=\"data row16 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col20\" class=\"data row16 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col21\" class=\"data row16 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row16_col22\" class=\"data row16 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row17\" class=\"row_heading level0 row17\" >cell type</th>\n",
       "      <td id=\"T_4a06a_row17_col0\" class=\"data row17 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col1\" class=\"data row17 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col2\" class=\"data row17 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col3\" class=\"data row17 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col4\" class=\"data row17 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col5\" class=\"data row17 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col6\" class=\"data row17 col6\" >2</td>\n",
       "      <td id=\"T_4a06a_row17_col7\" class=\"data row17 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col8\" class=\"data row17 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col9\" class=\"data row17 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col10\" class=\"data row17 col10\" >1</td>\n",
       "      <td id=\"T_4a06a_row17_col11\" class=\"data row17 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col12\" class=\"data row17 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col13\" class=\"data row17 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col14\" class=\"data row17 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col15\" class=\"data row17 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col16\" class=\"data row17 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col17\" class=\"data row17 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col18\" class=\"data row17 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col19\" class=\"data row17 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col20\" class=\"data row17 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col21\" class=\"data row17 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row17_col22\" class=\"data row17 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row18\" class=\"row_heading level0 row18\" >other</th>\n",
       "      <td id=\"T_4a06a_row18_col0\" class=\"data row18 col0\" >1</td>\n",
       "      <td id=\"T_4a06a_row18_col1\" class=\"data row18 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col2\" class=\"data row18 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col3\" class=\"data row18 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col4\" class=\"data row18 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col5\" class=\"data row18 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col6\" class=\"data row18 col6\" >1</td>\n",
       "      <td id=\"T_4a06a_row18_col7\" class=\"data row18 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col8\" class=\"data row18 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col9\" class=\"data row18 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col10\" class=\"data row18 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col11\" class=\"data row18 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col12\" class=\"data row18 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col13\" class=\"data row18 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col14\" class=\"data row18 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col15\" class=\"data row18 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col16\" class=\"data row18 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col17\" class=\"data row18 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col18\" class=\"data row18 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col19\" class=\"data row18 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col20\" class=\"data row18 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col21\" class=\"data row18 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row18_col22\" class=\"data row18 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row19\" class=\"row_heading level0 row19\" >treatment</th>\n",
       "      <td id=\"T_4a06a_row19_col0\" class=\"data row19 col0\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col1\" class=\"data row19 col1\" >2</td>\n",
       "      <td id=\"T_4a06a_row19_col2\" class=\"data row19 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col3\" class=\"data row19 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col4\" class=\"data row19 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col5\" class=\"data row19 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col6\" class=\"data row19 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col7\" class=\"data row19 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col8\" class=\"data row19 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col9\" class=\"data row19 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col10\" class=\"data row19 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col11\" class=\"data row19 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col12\" class=\"data row19 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col13\" class=\"data row19 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col14\" class=\"data row19 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col15\" class=\"data row19 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col16\" class=\"data row19 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col17\" class=\"data row19 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col18\" class=\"data row19 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col19\" class=\"data row19 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col20\" class=\"data row19 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col21\" class=\"data row19 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row19_col22\" class=\"data row19 col22\" >0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_4a06a_level0_row20\" class=\"row_heading level0 row20\" >drug</th>\n",
       "      <td id=\"T_4a06a_row20_col0\" class=\"data row20 col0\" >1</td>\n",
       "      <td id=\"T_4a06a_row20_col1\" class=\"data row20 col1\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col2\" class=\"data row20 col2\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col3\" class=\"data row20 col3\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col4\" class=\"data row20 col4\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col5\" class=\"data row20 col5\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col6\" class=\"data row20 col6\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col7\" class=\"data row20 col7\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col8\" class=\"data row20 col8\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col9\" class=\"data row20 col9\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col10\" class=\"data row20 col10\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col11\" class=\"data row20 col11\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col12\" class=\"data row20 col12\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col13\" class=\"data row20 col13\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col14\" class=\"data row20 col14\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col15\" class=\"data row20 col15\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col16\" class=\"data row20 col16\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col17\" class=\"data row20 col17\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col18\" class=\"data row20 col18\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col19\" class=\"data row20 col19\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col20\" class=\"data row20 col20\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col21\" class=\"data row20 col21\" >0</td>\n",
       "      <td id=\"T_4a06a_row20_col22\" class=\"data row20 col22\" >0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x1afb59480>"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    relations\n",
    "    .assign(subject_entity=lambda df: df[\"subject_entity\"].str.replace(\"[<>]\", \"\", regex=True))\n",
    "    .assign(object_entity=lambda df: df[\"object_entity\"].str.replace(\"[<>]\", \"\", regex=True))\n",
    "    [[\"subject_entity\", \"object_entity\"]]\n",
    "    .value_counts().unstack().fillna(0).astype(int)\n",
    "    .pipe(lambda df: df.loc[\n",
    "        df.sum(axis=1).sort_values(ascending=False).index,\n",
    "        df.sum(axis=0).sort_values(ascending=False).index\n",
    "    ])\n",
    "    .style\n",
    "    .background_gradient()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "predicate\n",
       "has phenotype                   329\n",
       "associated with                 158\n",
       "has participant                  61\n",
       "causes                           55\n",
       "part of                          33\n",
       "has genotype                     32\n",
       "caused by                        30\n",
       "located in                       22\n",
       "produces                         17\n",
       "correlated with                  15\n",
       "has attribute                    14\n",
       "increases abundance of           12\n",
       "decreases abundance of           11\n",
       "has output                        9\n",
       "location of                       8\n",
       "involved in                       8\n",
       "occurs in                         7\n",
       "performed                         6\n",
       "used in                           6\n",
       "produced by                       5\n",
       "not associated with               5\n",
       "has gene product                  5\n",
       "contributes to                    5\n",
       "demonstrates                      4\n",
       "derives from                      4\n",
       "participates in                   3\n",
       "treats                            3\n",
       "not correlated with               3\n",
       "is marker for                     2\n",
       "enables                           2\n",
       "manifestation of                  2\n",
       "negatively correlated with        2\n",
       "confirms                          2\n",
       "cause                             2\n",
       "not seen in                       2\n",
       "has role                          2\n",
       "is allele of                      2\n",
       "interacts with                    2\n",
       "catalyzes                         1\n",
       "degraded by                       1\n",
       "translocated to                   1\n",
       "detected by                       1\n",
       "does not produce                  1\n",
       "also known as                     1\n",
       "initiated                         1\n",
       "positively correlated with        1\n",
       "processed through                 1\n",
       "is part of                        1\n",
       "expresses                         1\n",
       "responsible for                   1\n",
       "includes                          1\n",
       "published                         1\n",
       "approved                          1\n",
       "implicated in                     1\n",
       "confirmed by                      1\n",
       "disrupts                          1\n",
       "form                              1\n",
       "undergo                           1\n",
       "mediates                          1\n",
       "postulated to play a role in      1\n",
       "performed in                      1\n",
       "identified                        1\n",
       "plays a role in                   1\n",
       "encoded by                        1\n",
       "identifies                        1\n",
       "instance of                       1\n",
       "encodes                           1\n",
       "expressed in                      1\n",
       "validated by                      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "relations.predicate.value_counts().head(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "relation\n",
       "has phenotype          225\n",
       "associated with         71\n",
       "caused by               18\n",
       "causes                  16\n",
       "has genotype            13\n",
       "not associated with      4\n",
       "has participant          3\n",
       "is allele of             1\n",
       "produces                 1\n",
       "part of                  1\n",
       "involved in              1\n",
       "instance of              1\n",
       "is marker for            1\n",
       "manifestation of         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entity</th>\n",
       "      <th>entity_type</th>\n",
       "      <th>relation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Alacrima/hypolacrima</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Anal stenosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Chalazions</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>Communication scores</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>Constipation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Corneal ulcerations/scarring</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>Daily Living Skills scores</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>Developmental delays in mastication</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Dysmorphic features</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>Dystonic movements of the tongue</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>EEG abnormalities</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>EEG abnormalities</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>EEG abnormalities</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Elevated AFP</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Elevated liver transaminases</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>Gastric pH alteration</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>Generalized weakness of the lips and tongue</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Global developmental delay</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Hypotonia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>IUGR</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Lactic acidosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>Lagophthalmous</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Liver fibrosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Liver storage or vacuolization</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Microcephaly</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Microcephaly</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>Motor deficits</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Movement disorder</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Neonatal jaundice</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Ocular apraxia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>Oral motor deficits</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Peripheral neuropathy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>Premature spillage</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>QTcB &gt;440 ms</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Scoliosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Seizures</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Seizures</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>Small hands/feet</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Strabismus</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>Twelve individuals</td>\n",
       "      <td>organism</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>Vineland Adaptive Behavior Composite scores</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>a movement disorder</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>abnormal abdominal ultrasound</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>abnormal accumulation of misfolded glycoproteins</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>abnormal brain imaging</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>abnormal nerve conduction</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>abnormal tear production</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>absent tears resulting in blepharitis and corneal ulceration</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>accumulation of a substance with staining properties similar to glycogen in hepatocyte cytoplasm</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>accumulation of an amorphous unidentified substance</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>acquired microcephaly</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>acquired microcephaly</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>acquired microcephaly</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>action tremor</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>adrenal cortex vacuolation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>alacrima</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>alacrima</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>altered resting energy expenditure</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>ambulation with a walker</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>athetoid arm and hand movements</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>athetoid movements</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>atrophy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>auditory neural pathway dysfunction</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>below average IQ</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>blonde or poorly pigmented retinal periphery</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>bradycardia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>brain imaging abnormalities</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>central sleep apnea</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>choreiform movements</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>choreoathetosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>choreoathetosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>chronic conjunctivitis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>chronic constipation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>clasped hands</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>clinical seizures</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>combined obstructive and central sleep apnea</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>complex hyperkinetic movement disorder</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>cone dystrophy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>congenital disorder of glycosylation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>constipation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>constipation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>contractures in both small and large joints</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>corneal clouding</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>corneal neovascularization</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>cortical vision impairment</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>coxa valga</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>cyanosis at birth</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>decreased fetal movement</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>decreased tear production</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>delayed bone age</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>delayed bone age</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>developmental delay</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>developmental delay</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>developmental delay</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>diminished deep tendon reflexes</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>diminished reflexes</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>dislocations or subluxations involving the hips and shoulder joints</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>disordered mastication</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>distal tapering of hands and feet</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>dry eyes</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>dysmetric movements</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>dysmorphic features</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>dystonic movements</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>elevated alanine transaminase</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>elevated aspartate aminotransferase</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>elevated liver transaminases</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>elevated liver transaminases</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>elevated liver transaminases</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>elevation of AFP and liver enzymes in infancy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>esotropia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>exotropia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>expansion of the cortexes in the proximal and middle phalanges in digits 2–4</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>extremity athetosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>females</td>\n",
       "      <td>organism</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>flexion deformity of the fingers</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>frequent periodic limb movements</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>gastro-esophageal reflux</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>global developmental delay</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>global developmental delay</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>global developmental delay with relatively preserved socialization</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>growth arrest lines</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>hallux valgus</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>head bobbing</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>hearing impairment</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>heart rates in the low 100s</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>hepatocellular carcinoma</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>hepatocyte cytoplasmic storage material or vacuolization</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>high antibody titers towards rubella and/or rubeola following vaccination</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>history of IUGR</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>hollowed cheeks</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>hyperkinetic movement disorders</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>hypo- or alacrima</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>hypo- or complete alacrima</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>348</th>\n",
       "      <td>hypo-lacrima</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>hypohidrosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>hypolacrima</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>hypotonia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>hypotonia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>hypotonia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>hypotonia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>hypotonia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>hypotonic facies</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>impaired Daily Living Skills scores</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>inability to close eyes during sleep</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>intellectual disabilities</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>intellectual disability</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>intention tremor</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>intrauterine growth retardation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>involuntary movements of neck, hands and arm</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>joint hypermobility</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>joint hypermobility</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>learned to sit alone</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>limited expressive speech</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>liver disease</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>liver disease</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>loss of mobility</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>low CSF total protein and albumin</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>low LDL</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>low femoral bone density</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>low total cholesterol</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>low triglycerides</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>low uE3</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>low weight</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>lower than predicted resting energy expenditure</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>males</td>\n",
       "      <td>organism</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>metaphyseal banding</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>microcephaly</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>mild jaundice</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>more mild phenotype</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>movement disorder</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>movement disorder</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>multifocal epileptiform activity</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>myoclonic movements</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>no words</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>nonverbal communication</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>normal QTcF</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>obstructive sleep apnea</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>ocular apraxia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>optic atrophy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>optic nerve pallor</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>out of range elevations in antibody titers towards rubella and/or rubeola</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>pannus formation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>peripheral neuropathy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>peripheral neuropathy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>persistent oral reflexes of suckling and suck/swallows</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>pharyngeal swallow response delays</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>phrase speech</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>pigmentary changes</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>poor growth after mid-childhood</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>possible liver fibrosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>profound intellectual disability</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>prominent apophyseal bone formation in the ileac crests and ischia</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>ptosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>reduced photopic responses</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>refractive errors</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>relative normalization in early childhood</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>retinal</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>retinal pigmentary changes/cone dystrophy</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>scarring</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>sclerosis of the phalanges or tarsal bones</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>sclerotic lesion of the distal femur</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>scoliosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>seizures</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>seizures</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>severe developmental delay</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>severe scoliosis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>shortened and widened metatarsals</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>single word speech</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>slow progress without regression</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>small feet</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>started to crawl at 2 years</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>strabismus</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>strabismus</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>strong Socialization scores</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>subclinical seizures</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>talipes equinovarus</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>thin facies</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>thinning of the fibulas</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>total foot length &lt; 3rd percentile</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>transient nature of the AFP and liver transaminase elevation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>transient transaminitis</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>ulnar bowing</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>unremarkable Echocardiograms</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>visible zygomatic arches</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>widened phalanges</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>has phenotype</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                               entity  \\\n",
       "49                                                                               Alacrima/hypolacrima   \n",
       "105                                                                                     Anal stenosis   \n",
       "51                                                                                         Chalazions   \n",
       "245                                                                              Communication scores   \n",
       "59                                                                                       Constipation   \n",
       "50                                                                       Corneal ulcerations/scarring   \n",
       "246                                                                        Daily Living Skills scores   \n",
       "291                                                               Developmental delays in mastication   \n",
       "60                                                                                Dysmorphic features   \n",
       "293                                                                  Dystonic movements of the tongue   \n",
       "34                                                                                  EEG abnormalities   \n",
       "46                                                                                  EEG abnormalities   \n",
       "132                                                                                 EEG abnormalities   \n",
       "56                                                                                       Elevated AFP   \n",
       "55                                                                       Elevated liver transaminases   \n",
       "303                                                                             Gastric pH alteration   \n",
       "292                                                       Generalized weakness of the lips and tongue   \n",
       "42                                                                         Global developmental delay   \n",
       "44                                                                                          Hypotonia   \n",
       "103                                                                                              IUGR   \n",
       "53                                                                                    Lactic acidosis   \n",
       "271                                                                                    Lagophthalmous   \n",
       "57                                                                                     Liver fibrosis   \n",
       "58                                                                     Liver storage or vacuolization   \n",
       "43                                                                                       Microcephaly   \n",
       "95                                                                                       Microcephaly   \n",
       "238                                                                                    Motor deficits   \n",
       "45                                                                                  Movement disorder   \n",
       "54                                                                                  Neonatal jaundice   \n",
       "48                                                                                     Ocular apraxia   \n",
       "288                                                                               Oral motor deficits   \n",
       "63                                                                              Peripheral neuropathy   \n",
       "289                                                                                Premature spillage   \n",
       "297                                                                                      QTcB >440 ms   \n",
       "61                                                                                          Scoliosis   \n",
       "47                                                                                           Seizures   \n",
       "111                                                                                          Seizures   \n",
       "62                                                                                   Small hands/feet   \n",
       "52                                                                                         Strabismus   \n",
       "207                                                                                Twelve individuals   \n",
       "247                                                       Vineland Adaptive Behavior Composite scores   \n",
       "184                                                                               a movement disorder   \n",
       "305                                                                     abnormal abdominal ultrasound   \n",
       "21                                                   abnormal accumulation of misfolded glycoproteins   \n",
       "33                                                                             abnormal brain imaging   \n",
       "40                                                                          abnormal nerve conduction   \n",
       "126                                                                          abnormal tear production   \n",
       "183                                      absent tears resulting in blepharitis and corneal ulceration   \n",
       "182  accumulation of a substance with staining properties similar to glycogen in hepatocyte cytoplasm   \n",
       "20                                                accumulation of an amorphous unidentified substance   \n",
       "113                                                                             acquired microcephaly   \n",
       "130                                                                             acquired microcephaly   \n",
       "227                                                                             acquired microcephaly   \n",
       "256                                                                                     action tremor   \n",
       "138                                                                        adrenal cortex vacuolation   \n",
       "13                                                                                           alacrima   \n",
       "74                                                                                           alacrima   \n",
       "307                                                                altered resting energy expenditure   \n",
       "78                                                                           ambulation with a walker   \n",
       "92                                                                    athetoid arm and hand movements   \n",
       "253                                                                                athetoid movements   \n",
       "279                                                                                           atrophy   \n",
       "346                                                               auditory neural pathway dysfunction   \n",
       "240                                                                                  below average IQ   \n",
       "282                                                      blonde or poorly pigmented retinal periphery   \n",
       "102                                                                                       bradycardia   \n",
       "133                                                                       brain imaging abnormalities   \n",
       "300                                                                               central sleep apnea   \n",
       "252                                                                              choreiform movements   \n",
       "14                                                                                    choreoathetosis   \n",
       "127                                                                                   choreoathetosis   \n",
       "115                                                                            chronic conjunctivitis   \n",
       "121                                                                              chronic constipation   \n",
       "93                                                                                      clasped hands   \n",
       "248                                                                                 clinical seizures   \n",
       "301                                                      combined obstructive and central sleep apnea   \n",
       "199                                                            complex hyperkinetic movement disorder   \n",
       "285                                                                                    cone dystrophy   \n",
       "19                                                               congenital disorder of glycosylation   \n",
       "135                                                                                      constipation   \n",
       "304                                                                                      constipation   \n",
       "330                                                       contractures in both small and large joints   \n",
       "116                                                                                  corneal clouding   \n",
       "275                                                                        corneal neovascularization   \n",
       "73                                                                         cortical vision impairment   \n",
       "331                                                                                        coxa valga   \n",
       "104                                                                                 cyanosis at birth   \n",
       "101                                                                          decreased fetal movement   \n",
       "196                                                                         decreased tear production   \n",
       "191                                                                                  delayed bone age   \n",
       "327                                                                                  delayed bone age   \n",
       "66                                                                                developmental delay   \n",
       "110                                                                               developmental delay   \n",
       "235                                                                               developmental delay   \n",
       "72                                                                    diminished deep tendon reflexes   \n",
       "37                                                                                diminished reflexes   \n",
       "335                               dislocations or subluxations involving the hips and shoulder joints   \n",
       "349                                                                            disordered mastication   \n",
       "71                                                                  distal tapering of hands and feet   \n",
       "75                                                                                           dry eyes   \n",
       "257                                                                               dysmetric movements   \n",
       "137                                                                               dysmorphic features   \n",
       "254                                                                                dystonic movements   \n",
       "309                                                                     elevated alanine transaminase   \n",
       "308                                                               elevated aspartate aminotransferase   \n",
       "35                                                                       elevated liver transaminases   \n",
       "67                                                                       elevated liver transaminases   \n",
       "94                                                                       elevated liver transaminases   \n",
       "180                                                     elevation of AFP and liver enzymes in infancy   \n",
       "274                                                                                         esotropia   \n",
       "273                                                                                         exotropia   \n",
       "343                      expansion of the cortexes in the proximal and middle phalanges in digits 2–4   \n",
       "109                                                                               extremity athetosis   \n",
       "234                                                                                           females   \n",
       "337                                                                  flexion deformity of the fingers   \n",
       "302                                                                  frequent periodic limb movements   \n",
       "120                                                                          gastro-esophageal reflux   \n",
       "29                                                                         global developmental delay   \n",
       "129                                                                        global developmental delay   \n",
       "200                                global developmental delay with relatively preserved socialization   \n",
       "333                                                                               growth arrest lines   \n",
       "338                                                                                     hallux valgus   \n",
       "108                                                                                      head bobbing   \n",
       "119                                                                                hearing impairment   \n",
       "296                                                                       heart rates in the low 100s   \n",
       "310                                                                          hepatocellular carcinoma   \n",
       "38                                           hepatocyte cytoplasmic storage material or vacuolization   \n",
       "195                         high antibody titers towards rubella and/or rubeola following vaccination   \n",
       "136                                                                                   history of IUGR   \n",
       "224                                                                                   hollowed cheeks   \n",
       "251                                                                   hyperkinetic movement disorders   \n",
       "32                                                                                  hypo- or alacrima   \n",
       "284                                                                        hypo- or complete alacrima   \n",
       "348                                                                                      hypo-lacrima   \n",
       "347                                                                                      hypohidrosis   \n",
       "117                                                                                       hypolacrima   \n",
       "31                                                                                          hypotonia   \n",
       "65                                                                                          hypotonia   \n",
       "91                                                                                          hypotonia   \n",
       "106                                                                                         hypotonia   \n",
       "131                                                                                         hypotonia   \n",
       "221                                                                                  hypotonic facies   \n",
       "239                                                               impaired Daily Living Skills scores   \n",
       "114                                                              inability to close eyes during sleep   \n",
       "76                                                                          intellectual disabilities   \n",
       "236                                                                           intellectual disability   \n",
       "68                                                                                   intention tremor   \n",
       "64                                                                    intrauterine growth retardation   \n",
       "69                                                       involuntary movements of neck, hands and arm   \n",
       "192                                                                               joint hypermobility   \n",
       "329                                                                               joint hypermobility   \n",
       "98                                                                               learned to sit alone   \n",
       "77                                                                          limited expressive speech   \n",
       "16                                                                                      liver disease   \n",
       "128                                                                                     liver disease   \n",
       "112                                                                                  loss of mobility   \n",
       "194                                                                 low CSF total protein and albumin   \n",
       "312                                                                                           low LDL   \n",
       "328                                                                          low femoral bone density   \n",
       "311                                                                             low total cholesterol   \n",
       "313                                                                                 low triglycerides   \n",
       "139                                                                                           low uE3   \n",
       "222                                                                                        low weight   \n",
       "193                                                   lower than predicted resting energy expenditure   \n",
       "233                                                                                             males   \n",
       "334                                                                               metaphyseal banding   \n",
       "36                                                                                       microcephaly   \n",
       "89                                                                                      mild jaundice   \n",
       "187                                                                               more mild phenotype   \n",
       "30                                                                                  movement disorder   \n",
       "107                                                                                 movement disorder   \n",
       "250                                                                  multifocal epileptiform activity   \n",
       "255                                                                               myoclonic movements   \n",
       "100                                                                                          no words   \n",
       "241                                                                           nonverbal communication   \n",
       "298                                                                                       normal QTcF   \n",
       "299                                                                           obstructive sleep apnea   \n",
       "70                                                                                     ocular apraxia   \n",
       "189                                                                                     optic atrophy   \n",
       "278                                                                                optic nerve pallor   \n",
       "325                         out of range elevations in antibody titers towards rubella and/or rubeola   \n",
       "276                                                                                  pannus formation   \n",
       "134                                                                             peripheral neuropathy   \n",
       "185                                                                             peripheral neuropathy   \n",
       "294                                            persistent oral reflexes of suckling and suck/swallows   \n",
       "290                                                                pharyngeal swallow response delays   \n",
       "243                                                                                     phrase speech   \n",
       "281                                                                                pigmentary changes   \n",
       "226                                                                   poor growth after mid-childhood   \n",
       "306                                                                           possible liver fibrosis   \n",
       "237                                                                  profound intellectual disability   \n",
       "341                                prominent apophyseal bone formation in the ileac crests and ischia   \n",
       "272                                                                                            ptosis   \n",
       "286                                                                        reduced photopic responses   \n",
       "283                                                                                 refractive errors   \n",
       "181                                                         relative normalization in early childhood   \n",
       "280                                                                                           retinal   \n",
       "190                                                         retinal pigmentary changes/cone dystrophy   \n",
       "277                                                                                          scarring   \n",
       "336                                                        sclerosis of the phalanges or tarsal bones   \n",
       "339                                                              sclerotic lesion of the distal femur   \n",
       "332                                                                                         scoliosis   \n",
       "15                                                                                           seizures   \n",
       "39                                                                                           seizures   \n",
       "96                                                                         severe developmental delay   \n",
       "122                                                                                  severe scoliosis   \n",
       "345                                                                 shortened and widened metatarsals   \n",
       "242                                                                                single word speech   \n",
       "97                                                                   slow progress without regression   \n",
       "198                                                                                        small feet   \n",
       "99                                                                        started to crawl at 2 years   \n",
       "90                                                                                         strabismus   \n",
       "118                                                                                        strabismus   \n",
       "244                                                                       strong Socialization scores   \n",
       "249                                                                              subclinical seizures   \n",
       "123                                                                               talipes equinovarus   \n",
       "223                                                                                       thin facies   \n",
       "340                                                                           thinning of the fibulas   \n",
       "228                                                                total foot length < 3rd percentile   \n",
       "186                                      transient nature of the AFP and liver transaminase elevation   \n",
       "197                                                                           transient transaminitis   \n",
       "342                                                                                      ulnar bowing   \n",
       "295                                                                      unremarkable Echocardiograms   \n",
       "225                                                                          visible zygomatic arches   \n",
       "344                                                                                 widened phalanges   \n",
       "\n",
       "    entity_type       relation  \n",
       "49    phenotype  has phenotype  \n",
       "105   phenotype  has phenotype  \n",
       "51    phenotype  has phenotype  \n",
       "245   phenotype  has phenotype  \n",
       "59    phenotype  has phenotype  \n",
       "50    phenotype  has phenotype  \n",
       "246   phenotype  has phenotype  \n",
       "291   phenotype  has phenotype  \n",
       "60    phenotype  has phenotype  \n",
       "293   phenotype  has phenotype  \n",
       "34    phenotype  has phenotype  \n",
       "46    phenotype  has phenotype  \n",
       "132   phenotype  has phenotype  \n",
       "56    phenotype  has phenotype  \n",
       "55    phenotype  has phenotype  \n",
       "303   phenotype  has phenotype  \n",
       "292   phenotype  has phenotype  \n",
       "42    phenotype  has phenotype  \n",
       "44    phenotype  has phenotype  \n",
       "103   phenotype  has phenotype  \n",
       "53    phenotype  has phenotype  \n",
       "271   phenotype  has phenotype  \n",
       "57    phenotype  has phenotype  \n",
       "58    phenotype  has phenotype  \n",
       "43    phenotype  has phenotype  \n",
       "95    phenotype  has phenotype  \n",
       "238   phenotype  has phenotype  \n",
       "45    phenotype  has phenotype  \n",
       "54    phenotype  has phenotype  \n",
       "48    phenotype  has phenotype  \n",
       "288   phenotype  has phenotype  \n",
       "63    phenotype  has phenotype  \n",
       "289   phenotype  has phenotype  \n",
       "297   phenotype  has phenotype  \n",
       "61    phenotype  has phenotype  \n",
       "47    phenotype  has phenotype  \n",
       "111   phenotype  has phenotype  \n",
       "62    phenotype  has phenotype  \n",
       "52    phenotype  has phenotype  \n",
       "207    organism  has phenotype  \n",
       "247   phenotype  has phenotype  \n",
       "184   phenotype  has phenotype  \n",
       "305   phenotype  has phenotype  \n",
       "21    phenotype  has phenotype  \n",
       "33    phenotype  has phenotype  \n",
       "40    phenotype  has phenotype  \n",
       "126   phenotype  has phenotype  \n",
       "183   phenotype  has phenotype  \n",
       "182   phenotype  has phenotype  \n",
       "20    phenotype  has phenotype  \n",
       "113   phenotype  has phenotype  \n",
       "130   phenotype  has phenotype  \n",
       "227   phenotype  has phenotype  \n",
       "256   phenotype  has phenotype  \n",
       "138   phenotype  has phenotype  \n",
       "13    phenotype  has phenotype  \n",
       "74    phenotype  has phenotype  \n",
       "307   phenotype  has phenotype  \n",
       "78    phenotype  has phenotype  \n",
       "92    phenotype  has phenotype  \n",
       "253   phenotype  has phenotype  \n",
       "279   phenotype  has phenotype  \n",
       "346   phenotype  has phenotype  \n",
       "240   phenotype  has phenotype  \n",
       "282   phenotype  has phenotype  \n",
       "102   phenotype  has phenotype  \n",
       "133   phenotype  has phenotype  \n",
       "300   phenotype  has phenotype  \n",
       "252   phenotype  has phenotype  \n",
       "14    phenotype  has phenotype  \n",
       "127   phenotype  has phenotype  \n",
       "115   phenotype  has phenotype  \n",
       "121   phenotype  has phenotype  \n",
       "93    phenotype  has phenotype  \n",
       "248   phenotype  has phenotype  \n",
       "301   phenotype  has phenotype  \n",
       "199   phenotype  has phenotype  \n",
       "285   phenotype  has phenotype  \n",
       "19    phenotype  has phenotype  \n",
       "135   phenotype  has phenotype  \n",
       "304   phenotype  has phenotype  \n",
       "330   phenotype  has phenotype  \n",
       "116   phenotype  has phenotype  \n",
       "275   phenotype  has phenotype  \n",
       "73    phenotype  has phenotype  \n",
       "331   phenotype  has phenotype  \n",
       "104   phenotype  has phenotype  \n",
       "101   phenotype  has phenotype  \n",
       "196   phenotype  has phenotype  \n",
       "191   phenotype  has phenotype  \n",
       "327   phenotype  has phenotype  \n",
       "66    phenotype  has phenotype  \n",
       "110   phenotype  has phenotype  \n",
       "235   phenotype  has phenotype  \n",
       "72    phenotype  has phenotype  \n",
       "37    phenotype  has phenotype  \n",
       "335   phenotype  has phenotype  \n",
       "349   phenotype  has phenotype  \n",
       "71    phenotype  has phenotype  \n",
       "75    phenotype  has phenotype  \n",
       "257   phenotype  has phenotype  \n",
       "137   phenotype  has phenotype  \n",
       "254   phenotype  has phenotype  \n",
       "309   phenotype  has phenotype  \n",
       "308   phenotype  has phenotype  \n",
       "35    phenotype  has phenotype  \n",
       "67    phenotype  has phenotype  \n",
       "94    phenotype  has phenotype  \n",
       "180   phenotype  has phenotype  \n",
       "274   phenotype  has phenotype  \n",
       "273   phenotype  has phenotype  \n",
       "343   phenotype  has phenotype  \n",
       "109   phenotype  has phenotype  \n",
       "234    organism  has phenotype  \n",
       "337   phenotype  has phenotype  \n",
       "302   phenotype  has phenotype  \n",
       "120   phenotype  has phenotype  \n",
       "29    phenotype  has phenotype  \n",
       "129   phenotype  has phenotype  \n",
       "200   phenotype  has phenotype  \n",
       "333   phenotype  has phenotype  \n",
       "338   phenotype  has phenotype  \n",
       "108   phenotype  has phenotype  \n",
       "119   phenotype  has phenotype  \n",
       "296   phenotype  has phenotype  \n",
       "310   phenotype  has phenotype  \n",
       "38    phenotype  has phenotype  \n",
       "195   phenotype  has phenotype  \n",
       "136   phenotype  has phenotype  \n",
       "224   phenotype  has phenotype  \n",
       "251   phenotype  has phenotype  \n",
       "32    phenotype  has phenotype  \n",
       "284   phenotype  has phenotype  \n",
       "348   phenotype  has phenotype  \n",
       "347   phenotype  has phenotype  \n",
       "117   phenotype  has phenotype  \n",
       "31    phenotype  has phenotype  \n",
       "65    phenotype  has phenotype  \n",
       "91    phenotype  has phenotype  \n",
       "106   phenotype  has phenotype  \n",
       "131   phenotype  has phenotype  \n",
       "221   phenotype  has phenotype  \n",
       "239   phenotype  has phenotype  \n",
       "114   phenotype  has phenotype  \n",
       "76    phenotype  has phenotype  \n",
       "236   phenotype  has phenotype  \n",
       "68    phenotype  has phenotype  \n",
       "64    phenotype  has phenotype  \n",
       "69    phenotype  has phenotype  \n",
       "192   phenotype  has phenotype  \n",
       "329   phenotype  has phenotype  \n",
       "98    phenotype  has phenotype  \n",
       "77    phenotype  has phenotype  \n",
       "16    phenotype  has phenotype  \n",
       "128   phenotype  has phenotype  \n",
       "112   phenotype  has phenotype  \n",
       "194   phenotype  has phenotype  \n",
       "312   phenotype  has phenotype  \n",
       "328   phenotype  has phenotype  \n",
       "311   phenotype  has phenotype  \n",
       "313   phenotype  has phenotype  \n",
       "139   phenotype  has phenotype  \n",
       "222   phenotype  has phenotype  \n",
       "193   phenotype  has phenotype  \n",
       "233    organism  has phenotype  \n",
       "334   phenotype  has phenotype  \n",
       "36    phenotype  has phenotype  \n",
       "89    phenotype  has phenotype  \n",
       "187   phenotype  has phenotype  \n",
       "30    phenotype  has phenotype  \n",
       "107   phenotype  has phenotype  \n",
       "250   phenotype  has phenotype  \n",
       "255   phenotype  has phenotype  \n",
       "100   phenotype  has phenotype  \n",
       "241   phenotype  has phenotype  \n",
       "298   phenotype  has phenotype  \n",
       "299   phenotype  has phenotype  \n",
       "70    phenotype  has phenotype  \n",
       "189   phenotype  has phenotype  \n",
       "278   phenotype  has phenotype  \n",
       "325   phenotype  has phenotype  \n",
       "276   phenotype  has phenotype  \n",
       "134   phenotype  has phenotype  \n",
       "185   phenotype  has phenotype  \n",
       "294   phenotype  has phenotype  \n",
       "290   phenotype  has phenotype  \n",
       "243   phenotype  has phenotype  \n",
       "281   phenotype  has phenotype  \n",
       "226   phenotype  has phenotype  \n",
       "306   phenotype  has phenotype  \n",
       "237   phenotype  has phenotype  \n",
       "341   phenotype  has phenotype  \n",
       "272   phenotype  has phenotype  \n",
       "286   phenotype  has phenotype  \n",
       "283   phenotype  has phenotype  \n",
       "181   phenotype  has phenotype  \n",
       "280   phenotype  has phenotype  \n",
       "190   phenotype  has phenotype  \n",
       "277   phenotype  has phenotype  \n",
       "336   phenotype  has phenotype  \n",
       "339   phenotype  has phenotype  \n",
       "332   phenotype  has phenotype  \n",
       "15    phenotype  has phenotype  \n",
       "39    phenotype  has phenotype  \n",
       "96    phenotype  has phenotype  \n",
       "122   phenotype  has phenotype  \n",
       "345   phenotype  has phenotype  \n",
       "242   phenotype  has phenotype  \n",
       "97    phenotype  has phenotype  \n",
       "198   phenotype  has phenotype  \n",
       "99    phenotype  has phenotype  \n",
       "90    phenotype  has phenotype  \n",
       "118   phenotype  has phenotype  \n",
       "244   phenotype  has phenotype  \n",
       "249   phenotype  has phenotype  \n",
       "123   phenotype  has phenotype  \n",
       "223   phenotype  has phenotype  \n",
       "340   phenotype  has phenotype  \n",
       "228   phenotype  has phenotype  \n",
       "186   phenotype  has phenotype  \n",
       "197   phenotype  has phenotype  \n",
       "342   phenotype  has phenotype  \n",
       "295   phenotype  has phenotype  \n",
       "225   phenotype  has phenotype  \n",
       "344   phenotype  has phenotype  "
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def apply(df, fn):\n",
    "    fn(df)\n",
    "    return df\n",
    "\n",
    "(\n",
    "    relations\n",
    "    .pipe(lambda df: df[\n",
    "        (df['subject'] == utils.NGLY1_DEFICIENCY) | \n",
    "        (df['object'] == utils.NGLY1_DEFICIENCY)\n",
    "    ])\n",
    "    .pipe(lambda df: pd.DataFrame([\n",
    "        (r['object'], r['object_entity'], r['predicate'])\n",
    "        if r['subject'] == utils.NGLY1_DEFICIENCY else \n",
    "        (r['subject'], r['subject_entity'], r['predicate'])\n",
    "        for _, r in df.iterrows()\n",
    "    ], columns=['entity', 'entity_type', 'relation']))\n",
    "    .pipe(apply, lambda df: display(df['relation'].value_counts()))\n",
    "    .pipe(lambda df: df[\n",
    "        df['relation'].str.contains('phenotype|disease|symptom', regex=True) \n",
    "    ])\n",
    "    # .pipe(lambda df: df[df['predicate'].str.contains('cause')])\n",
    "    .sort_values(['entity', 'relation'])\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>subject_entity</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object</th>\n",
       "      <th>object_entity</th>\n",
       "      <th>doc_id</th>\n",
       "      <th>doc_filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>cause</td>\n",
       "      <td>Inherited Disorder of the Endoplasmic Reticulum-Associated Degradation</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has gene product</td>\n",
       "      <td>Mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>c.1201A&gt;T (p.R401X)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>is allele of</td>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>c.1201A&gt;T (p.R401X)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>severe disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>caused by</td>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>impaired cytosolic degradation</td>\n",
       "      <td>molecular activity</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>abnormal accumulation of misfolded glycoproteins</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>accumulation of an amorphous unidentified substance</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>congenital disorder of glycosylation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>ERAD process</td>\n",
       "      <td>biological process</td>\n",
       "      <td>has participant</td>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>cytoplasm</td>\n",
       "      <td>cellular component</td>\n",
       "      <td>location of</td>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>impaired cytosolic degradation</td>\n",
       "      <td>molecular activity</td>\n",
       "      <td>caused by</td>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>abnormal accumulation of misfolded glycoproteins</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>caused by</td>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>accumulation of an amorphous unidentified substance</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>caused by</td>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>congenital disorder of glycosylation</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>caused by</td>\n",
       "      <td>mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>trisomy 18</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>positive second trimester serum screen</td>\n",
       "      <td>assay</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>c.1201A&gt;T (p.R401X) mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>c.1201A&gt;T (p.R401X) mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>has phenotype</td>\n",
       "      <td>mild clinical phenotype</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>NGLY1 gene</td>\n",
       "      <td>gene</td>\n",
       "      <td>produces</td>\n",
       "      <td>Q631S</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>NGLY1 gene</td>\n",
       "      <td>gene</td>\n",
       "      <td>produces</td>\n",
       "      <td>R401X</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>Q631S</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>transcript ENST00000280700</td>\n",
       "      <td>transcript variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>R401X</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>transcript ENST00000280700</td>\n",
       "      <td>transcript variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>Q631S</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>nonsense mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>R401X</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>nonsense mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>Patient 2</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1370dupG or p.R458fs</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>c.1370dupG or p.R458fs</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>frame shift in codon 458</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>c.1370dupG or p.R458fs</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>insertion of 13 incorrect residues</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>c.1370dupG or p.R458fs</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>stop codon in exon 9</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>c.1370dupG or p.R458fs</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>not seen in</td>\n",
       "      <td>3321 other subjects at Duke</td>\n",
       "      <td>organism</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>c.1370dupG or p.R458fs</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>not seen in</td>\n",
       "      <td>6503 subjects on the Exome Variant Server</td>\n",
       "      <td>organism</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>Patient 3</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>Mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>G&gt;A mutation at position 3:25761670 (hg19)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>p.R542X</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>3 base pair in-frame deletion TCC&gt;</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>position 3:25775416 (hg19)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>G&gt;T mutation at position 3:25777564</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>produces</td>\n",
       "      <td>heterozygous SMP</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>caused by</td>\n",
       "      <td>Mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>Patient 4</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>3:25775422 (hg19)</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>ENST00000280700</td>\n",
       "      <td>transcript variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>c.1201A&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>exon 8 of NGLY1</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>assay</td>\n",
       "      <td>produces</td>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>Duke University</td>\n",
       "      <td>organization</td>\n",
       "      <td>produces</td>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>CLIA-certified laboratory</td>\n",
       "      <td>organization</td>\n",
       "      <td>confirms</td>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>GeneDx</td>\n",
       "      <td>organization</td>\n",
       "      <td>confirms</td>\n",
       "      <td>p.R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>single nucleotide variant T&gt; A at position 3:25775422 (hg19)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>single nucleotide variant T&gt; A at position 3:25775422 (hg19)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>R401X</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>R401X</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>caused by</td>\n",
       "      <td>base pair substitution</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>Patients 5 and 6</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>single nucleotide variant T&gt; A at position 3:25775422 (hg19)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>mother of Patients 5 and 6</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>heterozygous call of single nucleotide variant T&gt; A at position 3:25775422 (hg19)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>NGLY1 variant</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>validated by</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>molecular activity</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>Patients 7 and 8</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>3:25775422 (hg19)</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>ENST00000280700</td>\n",
       "      <td>transcript variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>c.1201A&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>R401X</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>exon 8 of NGLY1</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>caused by</td>\n",
       "      <td>knockout mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>abnormal erlin2</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>impaired ERAD</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>abnormal erlin2</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>disrupts</td>\n",
       "      <td>erlin1/2 complex</td>\n",
       "      <td>macromolecular complex</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>Mutations in RNF170</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>cause</td>\n",
       "      <td>late- onset autosomal dominant sensory ataxia</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>Caenorhabditis elegans</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>loss-of-function mutations in png-1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>loss-of-function mutations in png-1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>peripheral nervous system defects</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>c.1201A&gt;T (p.R401X)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>homozygous mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>p.R401X</td>\n",
       "      <td>protein variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>homozygous mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>severe phenotype</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>c.1201A&gt;T (p.R401X)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>associated with</td>\n",
       "      <td>functional genetic variation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>functional genetic variation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>human population</td>\n",
       "      <td>organism</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>disease</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC4243708</td>\n",
       "      <td>PMC4243708.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>caused by</td>\n",
       "      <td>biallelic mutations in the NGLY1 gene</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>exome sequencing</td>\n",
       "      <td>assay</td>\n",
       "      <td>identifies</td>\n",
       "      <td>biallelic mutations in the NGLY1 gene</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>NGLY1 gene</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>biallelic mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>478</th>\n",
       "      <td>3-year-old male</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1891delC (p.Q631S NM_018297.3) variant</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479</th>\n",
       "      <td>3-year-old male</td>\n",
       "      <td>organism</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1201A&gt;T (p.R401* NM_018297.3) variant</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>biallelic mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>biallelic mutations in NGLY1</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>causes</td>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>c.1201A&gt;T (p.R401*)</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>13 different mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>13 different mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>545</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>five missense mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546</th>\n",
       "      <td>five missense mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>547</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>five nonsense mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548</th>\n",
       "      <td>five nonsense mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>549</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>two splice site mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550</th>\n",
       "      <td>two splice site mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>551</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>one frameshift mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>one frameshift mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>four mutations in the catalytic domain</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>554</th>\n",
       "      <td>four mutations in the catalytic domain</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1</td>\n",
       "      <td>gene</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.953T&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1169G&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1201A&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.931G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.730T&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1604G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1910delT</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1891delC</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>563</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.622C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.930C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.347C&gt;G</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.881+5G&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567</th>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>has genotype</td>\n",
       "      <td>c.1370dupG</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>c.1201A&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>rs201337954</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>569</th>\n",
       "      <td>c.931G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>rs201791209</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570</th>\n",
       "      <td>c.1891delC</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>rs587776982</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>c.622C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>rs200561967</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>c.1370dupG</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>rs587777265</td>\n",
       "      <td>&lt;other&gt;</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>c.1201A&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>c.930C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>c.622C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>NGLY1 deficiency</td>\n",
       "      <td>disease</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>c.1201A&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>correlated with</td>\n",
       "      <td>disease severity</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>586</th>\n",
       "      <td>c.930C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>correlated with</td>\n",
       "      <td>mild impairment</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>c.622C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>correlated with</td>\n",
       "      <td>mild impairment</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>c.1201A&gt;T mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>lower Factor II</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>727</th>\n",
       "      <td>c.1201A&gt;T mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>lower Factor IX</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>728</th>\n",
       "      <td>c.1201A&gt;T mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>lower Factor XI</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729</th>\n",
       "      <td>c.1201A&gt;T mutation</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>associated with</td>\n",
       "      <td>Antithrombin III</td>\n",
       "      <td>protein</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>872</th>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>located in</td>\n",
       "      <td>exons</td>\n",
       "      <td>cellular component</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873</th>\n",
       "      <td>c.1624C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>c.1205_1207delGAA</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>c.1533_1536delTCAA</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>876</th>\n",
       "      <td>c.730T&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>877</th>\n",
       "      <td>c.953T&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>878</th>\n",
       "      <td>c.931G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>c.930C&gt; T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>880</th>\n",
       "      <td>catalytic domain</td>\n",
       "      <td>macromolecular complex</td>\n",
       "      <td>has participant</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>NM_018297.3:c.931G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>NM_018297.3:c.1604G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>NM_018297.3:c.622C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>NM_018297.3:c.347C&gt;G</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>885</th>\n",
       "      <td>NM_018297.3:c.1169G&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>NM_018297.3:c.730T&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>NM_018297.3:c.1910delT</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>NM_018297.3:c.930C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>889</th>\n",
       "      <td>NM_018297.3:c.881+5G&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>part of</td>\n",
       "      <td>NGLY1 mutations</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>ClinVar: SCV000259179</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.931G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891</th>\n",
       "      <td>ClinVar: SCV000259180</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.1604G&gt;A</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>ClinVar: SCV000259181</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.622C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>893</th>\n",
       "      <td>ClinVar: SCV000259182</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.347C&gt;G</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>894</th>\n",
       "      <td>ClinVar: SCV000259183</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.1169G&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>895</th>\n",
       "      <td>ClinVar: SCV000259184</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.730T&gt;C</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>ClinVar: SCV000259185</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.1910delT</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>897</th>\n",
       "      <td>ClinVar: SCV000259186</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.930C&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>ClinVar: SCV000259187</td>\n",
       "      <td>organization</td>\n",
       "      <td>has output</td>\n",
       "      <td>NM_018297.3:c.881+5G&gt;T</td>\n",
       "      <td>genetic variant</td>\n",
       "      <td>PMC7477955</td>\n",
       "      <td>PMC7477955.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                          subject  \\\n",
       "0                                                       Mutations   \n",
       "1                                                           NGLY1   \n",
       "14                                            c.1201A>T (p.R401X)   \n",
       "18                                            c.1201A>T (p.R401X)   \n",
       "33                                               NGLY1 deficiency   \n",
       "43                                             mutations in NGLY1   \n",
       "44                                             mutations in NGLY1   \n",
       "45                                             mutations in NGLY1   \n",
       "46                                             mutations in NGLY1   \n",
       "47                                             mutations in NGLY1   \n",
       "50                                                   ERAD process   \n",
       "59                                                      cytoplasm   \n",
       "63                                 impaired cytosolic degradation   \n",
       "71               abnormal accumulation of misfolded glycoproteins   \n",
       "77            accumulation of an amorphous unidentified substance   \n",
       "83                           congenital disorder of glycosylation   \n",
       "234                                                    trisomy 18   \n",
       "288                                  c.1201A>T (p.R401X) mutation   \n",
       "289                                  c.1201A>T (p.R401X) mutation   \n",
       "291                                                    NGLY1 gene   \n",
       "292                                                    NGLY1 gene   \n",
       "293                                                         Q631S   \n",
       "294                                                         R401X   \n",
       "295                                                         Q631S   \n",
       "296                                                         R401X   \n",
       "297                                                     Patient 2   \n",
       "299                                        c.1370dupG or p.R458fs   \n",
       "300                                        c.1370dupG or p.R458fs   \n",
       "301                                        c.1370dupG or p.R458fs   \n",
       "302                                        c.1370dupG or p.R458fs   \n",
       "303                                        c.1370dupG or p.R458fs   \n",
       "304                                                     Patient 3   \n",
       "307                    G>A mutation at position 3:25761670 (hg19)   \n",
       "308                            3 base pair in-frame deletion TCC>   \n",
       "309                           G>T mutation at position 3:25777564   \n",
       "312                                              NGLY1 deficiency   \n",
       "313                                                     Patient 4   \n",
       "314                                                       p.R401X   \n",
       "315                                                       p.R401X   \n",
       "316                                                       p.R401X   \n",
       "317                                                       p.R401X   \n",
       "318                                             Sanger sequencing   \n",
       "319                                               Duke University   \n",
       "320                                     CLIA-certified laboratory   \n",
       "321                                                        GeneDx   \n",
       "322                                                         NGLY1   \n",
       "323  single nucleotide variant T> A at position 3:25775422 (hg19)   \n",
       "324                                                         R401X   \n",
       "325                                              Patients 5 and 6   \n",
       "326                                    mother of Patients 5 and 6   \n",
       "327                                                 NGLY1 variant   \n",
       "328                                              Patients 7 and 8   \n",
       "329                                                         R401X   \n",
       "330                                                         R401X   \n",
       "331                                                         R401X   \n",
       "332                                                         R401X   \n",
       "349                                              NGLY1 deficiency   \n",
       "398                                               abnormal erlin2   \n",
       "399                                               abnormal erlin2   \n",
       "408                                           Mutations in RNF170   \n",
       "424                                        Caenorhabditis elegans   \n",
       "425                           loss-of-function mutations in png-1   \n",
       "430                                           c.1201A>T (p.R401X)   \n",
       "431                                          homozygous mutations   \n",
       "432                                          homozygous mutations   \n",
       "433                                           c.1201A>T (p.R401X)   \n",
       "435                                                         NGLY1   \n",
       "436                                  functional genetic variation   \n",
       "437                                                     mutations   \n",
       "470                                              NGLY1 deficiency   \n",
       "472                                              exome sequencing   \n",
       "473                                                    NGLY1 gene   \n",
       "478                                               3-year-old male   \n",
       "479                                               3-year-old male   \n",
       "536                                                         NGLY1   \n",
       "537                                  biallelic mutations in NGLY1   \n",
       "540                                           c.1201A>T (p.R401*)   \n",
       "543                                                         NGLY1   \n",
       "544                                        13 different mutations   \n",
       "545                                                         NGLY1   \n",
       "546                                       five missense mutations   \n",
       "547                                                         NGLY1   \n",
       "548                                       five nonsense mutations   \n",
       "549                                                         NGLY1   \n",
       "550                                     two splice site mutations   \n",
       "551                                                         NGLY1   \n",
       "552                                       one frameshift mutation   \n",
       "553                                                         NGLY1   \n",
       "554                        four mutations in the catalytic domain   \n",
       "555                                              NGLY1 deficiency   \n",
       "556                                              NGLY1 deficiency   \n",
       "557                                              NGLY1 deficiency   \n",
       "558                                              NGLY1 deficiency   \n",
       "559                                              NGLY1 deficiency   \n",
       "560                                              NGLY1 deficiency   \n",
       "561                                              NGLY1 deficiency   \n",
       "562                                              NGLY1 deficiency   \n",
       "563                                              NGLY1 deficiency   \n",
       "564                                              NGLY1 deficiency   \n",
       "565                                              NGLY1 deficiency   \n",
       "566                                              NGLY1 deficiency   \n",
       "567                                              NGLY1 deficiency   \n",
       "568                                                     c.1201A>T   \n",
       "569                                                      c.931G>A   \n",
       "570                                                    c.1891delC   \n",
       "571                                                      c.622C>T   \n",
       "572                                                    c.1370dupG   \n",
       "582                                                     c.1201A>T   \n",
       "583                                                      c.930C>T   \n",
       "584                                                      c.622C>T   \n",
       "585                                                     c.1201A>T   \n",
       "586                                                      c.930C>T   \n",
       "587                                                      c.622C>T   \n",
       "726                                            c.1201A>T mutation   \n",
       "727                                            c.1201A>T mutation   \n",
       "728                                            c.1201A>T mutation   \n",
       "729                                            c.1201A>T mutation   \n",
       "872                                               NGLY1 mutations   \n",
       "873                                                     c.1624C>T   \n",
       "874                                             c.1205_1207delGAA   \n",
       "875                                            c.1533_1536delTCAA   \n",
       "876                                                      c.730T>C   \n",
       "877                                                      c.953T>C   \n",
       "878                                                      c.931G>A   \n",
       "879                                                     c.930C> T   \n",
       "880                                              catalytic domain   \n",
       "881                                          NM_018297.3:c.931G>A   \n",
       "882                                         NM_018297.3:c.1604G>A   \n",
       "883                                          NM_018297.3:c.622C>T   \n",
       "884                                          NM_018297.3:c.347C>G   \n",
       "885                                         NM_018297.3:c.1169G>C   \n",
       "886                                          NM_018297.3:c.730T>C   \n",
       "887                                        NM_018297.3:c.1910delT   \n",
       "888                                          NM_018297.3:c.930C>T   \n",
       "889                                        NM_018297.3:c.881+5G>T   \n",
       "890                                         ClinVar: SCV000259179   \n",
       "891                                         ClinVar: SCV000259180   \n",
       "892                                         ClinVar: SCV000259181   \n",
       "893                                         ClinVar: SCV000259182   \n",
       "894                                         ClinVar: SCV000259183   \n",
       "895                                         ClinVar: SCV000259184   \n",
       "896                                         ClinVar: SCV000259185   \n",
       "897                                         ClinVar: SCV000259186   \n",
       "898                                         ClinVar: SCV000259187   \n",
       "\n",
       "             subject_entity         predicate  \\\n",
       "0           genetic variant             cause   \n",
       "1                      gene  has gene product   \n",
       "14          genetic variant      is allele of   \n",
       "18          genetic variant   associated with   \n",
       "33                  disease         caused by   \n",
       "43          genetic variant            causes   \n",
       "44          genetic variant            causes   \n",
       "45          genetic variant            causes   \n",
       "46          genetic variant            causes   \n",
       "47          genetic variant            causes   \n",
       "50       biological process   has participant   \n",
       "59       cellular component       location of   \n",
       "63       molecular activity         caused by   \n",
       "71                phenotype         caused by   \n",
       "77                phenotype         caused by   \n",
       "83                phenotype         caused by   \n",
       "234         genetic variant   associated with   \n",
       "288         genetic variant   associated with   \n",
       "289         genetic variant     has phenotype   \n",
       "291                    gene          produces   \n",
       "292                    gene          produces   \n",
       "293         protein variant        located in   \n",
       "294         protein variant        located in   \n",
       "295         protein variant            causes   \n",
       "296         protein variant            causes   \n",
       "297                organism      has genotype   \n",
       "299         genetic variant            causes   \n",
       "300         genetic variant            causes   \n",
       "301         genetic variant            causes   \n",
       "302         genetic variant       not seen in   \n",
       "303         genetic variant       not seen in   \n",
       "304                organism      has genotype   \n",
       "307         genetic variant            causes   \n",
       "308         genetic variant        located in   \n",
       "309         genetic variant          produces   \n",
       "312                 disease         caused by   \n",
       "313                organism      has genotype   \n",
       "314         genetic variant        located in   \n",
       "315         genetic variant        located in   \n",
       "316         genetic variant        located in   \n",
       "317         genetic variant        located in   \n",
       "318                   assay          produces   \n",
       "319            organization          produces   \n",
       "320            organization          confirms   \n",
       "321            organization          confirms   \n",
       "322                    gene      has genotype   \n",
       "323         genetic variant            causes   \n",
       "324         protein variant         caused by   \n",
       "325                organism      has genotype   \n",
       "326                organism      has genotype   \n",
       "327         genetic variant      validated by   \n",
       "328                organism      has genotype   \n",
       "329         genetic variant        located in   \n",
       "330         genetic variant        located in   \n",
       "331         genetic variant        located in   \n",
       "332         genetic variant        located in   \n",
       "349                 disease         caused by   \n",
       "398         protein variant            causes   \n",
       "399         protein variant          disrupts   \n",
       "408         genetic variant             cause   \n",
       "424                organism      has genotype   \n",
       "425         genetic variant            causes   \n",
       "430         genetic variant   associated with   \n",
       "431         genetic variant   associated with   \n",
       "432         genetic variant            causes   \n",
       "433         genetic variant   associated with   \n",
       "435                    gene   associated with   \n",
       "436         genetic variant   associated with   \n",
       "437         genetic variant   associated with   \n",
       "470                 disease         caused by   \n",
       "472                   assay        identifies   \n",
       "473                    gene      has genotype   \n",
       "478                organism      has genotype   \n",
       "479                organism      has genotype   \n",
       "536                    gene      has genotype   \n",
       "537         genetic variant            causes   \n",
       "540         genetic variant           part of   \n",
       "543                    gene      has genotype   \n",
       "544         genetic variant           part of   \n",
       "545                    gene      has genotype   \n",
       "546         genetic variant           part of   \n",
       "547                    gene      has genotype   \n",
       "548         genetic variant           part of   \n",
       "549                    gene      has genotype   \n",
       "550         genetic variant           part of   \n",
       "551                    gene      has genotype   \n",
       "552         genetic variant           part of   \n",
       "553                    gene      has genotype   \n",
       "554         genetic variant           part of   \n",
       "555                 disease      has genotype   \n",
       "556                 disease      has genotype   \n",
       "557                 disease      has genotype   \n",
       "558                 disease      has genotype   \n",
       "559                 disease      has genotype   \n",
       "560                 disease      has genotype   \n",
       "561                 disease      has genotype   \n",
       "562                 disease      has genotype   \n",
       "563                 disease      has genotype   \n",
       "564                 disease      has genotype   \n",
       "565                 disease      has genotype   \n",
       "566                 disease      has genotype   \n",
       "567                 disease      has genotype   \n",
       "568         genetic variant   associated with   \n",
       "569         genetic variant   associated with   \n",
       "570         genetic variant   associated with   \n",
       "571         genetic variant   associated with   \n",
       "572         genetic variant   associated with   \n",
       "582         genetic variant   associated with   \n",
       "583         genetic variant   associated with   \n",
       "584         genetic variant   associated with   \n",
       "585         genetic variant   correlated with   \n",
       "586         genetic variant   correlated with   \n",
       "587         genetic variant   correlated with   \n",
       "726         genetic variant   associated with   \n",
       "727         genetic variant   associated with   \n",
       "728         genetic variant   associated with   \n",
       "729         genetic variant   associated with   \n",
       "872         genetic variant        located in   \n",
       "873         genetic variant           part of   \n",
       "874         genetic variant           part of   \n",
       "875         genetic variant           part of   \n",
       "876         genetic variant           part of   \n",
       "877         genetic variant           part of   \n",
       "878         genetic variant           part of   \n",
       "879         genetic variant           part of   \n",
       "880  macromolecular complex   has participant   \n",
       "881         genetic variant           part of   \n",
       "882         genetic variant           part of   \n",
       "883         genetic variant           part of   \n",
       "884         genetic variant           part of   \n",
       "885         genetic variant           part of   \n",
       "886         genetic variant           part of   \n",
       "887         genetic variant           part of   \n",
       "888         genetic variant           part of   \n",
       "889         genetic variant           part of   \n",
       "890            organization        has output   \n",
       "891            organization        has output   \n",
       "892            organization        has output   \n",
       "893            organization        has output   \n",
       "894            organization        has output   \n",
       "895            organization        has output   \n",
       "896            organization        has output   \n",
       "897            organization        has output   \n",
       "898            organization        has output   \n",
       "\n",
       "                                                                                object  \\\n",
       "0               Inherited Disorder of the Endoplasmic Reticulum-Associated Degradation   \n",
       "1                                                                            Mutations   \n",
       "14                                                                    NGLY1 deficiency   \n",
       "18                                                                      severe disease   \n",
       "33                                                                  mutations in NGLY1   \n",
       "43                                                                    NGLY1 deficiency   \n",
       "44                                                      impaired cytosolic degradation   \n",
       "45                                    abnormal accumulation of misfolded glycoproteins   \n",
       "46                                 accumulation of an amorphous unidentified substance   \n",
       "47                                                congenital disorder of glycosylation   \n",
       "50                                                                  mutations in NGLY1   \n",
       "59                                                                  mutations in NGLY1   \n",
       "63                                                                  mutations in NGLY1   \n",
       "71                                                                  mutations in NGLY1   \n",
       "77                                                                  mutations in NGLY1   \n",
       "83                                                                  mutations in NGLY1   \n",
       "234                                             positive second trimester serum screen   \n",
       "288                                                                   NGLY1 deficiency   \n",
       "289                                                            mild clinical phenotype   \n",
       "291                                                                              Q631S   \n",
       "292                                                                              R401X   \n",
       "293                                                         transcript ENST00000280700   \n",
       "294                                                         transcript ENST00000280700   \n",
       "295                                                                  nonsense mutation   \n",
       "296                                                                  nonsense mutation   \n",
       "297                                                             c.1370dupG or p.R458fs   \n",
       "299                                                           frame shift in codon 458   \n",
       "300                                                 insertion of 13 incorrect residues   \n",
       "301                                                               stop codon in exon 9   \n",
       "302                                                        3321 other subjects at Duke   \n",
       "303                                          6503 subjects on the Exome Variant Server   \n",
       "304                                                                 Mutations in NGLY1   \n",
       "307                                                                            p.R542X   \n",
       "308                                                         position 3:25775416 (hg19)   \n",
       "309                                                                   heterozygous SMP   \n",
       "312                                                                 Mutations in NGLY1   \n",
       "313                                                                            p.R401X   \n",
       "314                                                                  3:25775422 (hg19)   \n",
       "315                                                                    ENST00000280700   \n",
       "316                                                                          c.1201A>T   \n",
       "317                                                                    exon 8 of NGLY1   \n",
       "318                                                                            p.R401X   \n",
       "319                                                                            p.R401X   \n",
       "320                                                                            p.R401X   \n",
       "321                                                                            p.R401X   \n",
       "322                       single nucleotide variant T> A at position 3:25775422 (hg19)   \n",
       "323                                                                              R401X   \n",
       "324                                                             base pair substitution   \n",
       "325                       single nucleotide variant T> A at position 3:25775422 (hg19)   \n",
       "326  heterozygous call of single nucleotide variant T> A at position 3:25775422 (hg19)   \n",
       "327                                                                  Sanger sequencing   \n",
       "328                                                                              R401X   \n",
       "329                                                                  3:25775422 (hg19)   \n",
       "330                                                                    ENST00000280700   \n",
       "331                                                                          c.1201A>T   \n",
       "332                                                                    exon 8 of NGLY1   \n",
       "349                                                        knockout mutations in NGLY1   \n",
       "398                                                                      impaired ERAD   \n",
       "399                                                                   erlin1/2 complex   \n",
       "408                                      late- onset autosomal dominant sensory ataxia   \n",
       "424                                                loss-of-function mutations in png-1   \n",
       "425                                                  peripheral nervous system defects   \n",
       "430                                                                              NGLY1   \n",
       "431                                                                            p.R401X   \n",
       "432                                                                   severe phenotype   \n",
       "433                                                                            disease   \n",
       "435                                                       functional genetic variation   \n",
       "436                                                                   human population   \n",
       "437                                                                            disease   \n",
       "470                                              biallelic mutations in the NGLY1 gene   \n",
       "472                                              biallelic mutations in the NGLY1 gene   \n",
       "473                                                                biallelic mutations   \n",
       "478                                           c.1891delC (p.Q631S NM_018297.3) variant   \n",
       "479                                            c.1201A>T (p.R401* NM_018297.3) variant   \n",
       "536                                                                biallelic mutations   \n",
       "537                                                                   NGLY1 deficiency   \n",
       "540                                                                    NGLY1 mutations   \n",
       "543                                                             13 different mutations   \n",
       "544                                                                              NGLY1   \n",
       "545                                                            five missense mutations   \n",
       "546                                                                              NGLY1   \n",
       "547                                                            five nonsense mutations   \n",
       "548                                                                              NGLY1   \n",
       "549                                                          two splice site mutations   \n",
       "550                                                                              NGLY1   \n",
       "551                                                            one frameshift mutation   \n",
       "552                                                                              NGLY1   \n",
       "553                                             four mutations in the catalytic domain   \n",
       "554                                                                              NGLY1   \n",
       "555                                                                           c.953T>C   \n",
       "556                                                                          c.1169G>C   \n",
       "557                                                                          c.1201A>T   \n",
       "558                                                                           c.931G>A   \n",
       "559                                                                           c.730T>C   \n",
       "560                                                                          c.1604G>A   \n",
       "561                                                                         c.1910delT   \n",
       "562                                                                         c.1891delC   \n",
       "563                                                                           c.622C>T   \n",
       "564                                                                           c.930C>T   \n",
       "565                                                                           c.347C>G   \n",
       "566                                                                         c.881+5G>T   \n",
       "567                                                                         c.1370dupG   \n",
       "568                                                                        rs201337954   \n",
       "569                                                                        rs201791209   \n",
       "570                                                                        rs587776982   \n",
       "571                                                                        rs200561967   \n",
       "572                                                                        rs587777265   \n",
       "582                                                                   NGLY1 deficiency   \n",
       "583                                                                   NGLY1 deficiency   \n",
       "584                                                                   NGLY1 deficiency   \n",
       "585                                                                   disease severity   \n",
       "586                                                                    mild impairment   \n",
       "587                                                                    mild impairment   \n",
       "726                                                                    lower Factor II   \n",
       "727                                                                    lower Factor IX   \n",
       "728                                                                    lower Factor XI   \n",
       "729                                                                   Antithrombin III   \n",
       "872                                                                              exons   \n",
       "873                                                                    NGLY1 mutations   \n",
       "874                                                                    NGLY1 mutations   \n",
       "875                                                                    NGLY1 mutations   \n",
       "876                                                                    NGLY1 mutations   \n",
       "877                                                                    NGLY1 mutations   \n",
       "878                                                                    NGLY1 mutations   \n",
       "879                                                                    NGLY1 mutations   \n",
       "880                                                                    NGLY1 mutations   \n",
       "881                                                                    NGLY1 mutations   \n",
       "882                                                                    NGLY1 mutations   \n",
       "883                                                                    NGLY1 mutations   \n",
       "884                                                                    NGLY1 mutations   \n",
       "885                                                                    NGLY1 mutations   \n",
       "886                                                                    NGLY1 mutations   \n",
       "887                                                                    NGLY1 mutations   \n",
       "888                                                                    NGLY1 mutations   \n",
       "889                                                                    NGLY1 mutations   \n",
       "890                                                               NM_018297.3:c.931G>A   \n",
       "891                                                              NM_018297.3:c.1604G>A   \n",
       "892                                                               NM_018297.3:c.622C>T   \n",
       "893                                                               NM_018297.3:c.347C>G   \n",
       "894                                                              NM_018297.3:c.1169G>C   \n",
       "895                                                               NM_018297.3:c.730T>C   \n",
       "896                                                             NM_018297.3:c.1910delT   \n",
       "897                                                               NM_018297.3:c.930C>T   \n",
       "898                                                             NM_018297.3:c.881+5G>T   \n",
       "\n",
       "              object_entity      doc_id    doc_filename  \n",
       "0                   disease  PMC4243708  PMC4243708.txt  \n",
       "1           genetic variant  PMC4243708  PMC4243708.txt  \n",
       "14                  disease  PMC4243708  PMC4243708.txt  \n",
       "18                  disease  PMC4243708  PMC4243708.txt  \n",
       "33          genetic variant  PMC4243708  PMC4243708.txt  \n",
       "43                  disease  PMC4243708  PMC4243708.txt  \n",
       "44       molecular activity  PMC4243708  PMC4243708.txt  \n",
       "45                phenotype  PMC4243708  PMC4243708.txt  \n",
       "46                phenotype  PMC4243708  PMC4243708.txt  \n",
       "47                phenotype  PMC4243708  PMC4243708.txt  \n",
       "50          genetic variant  PMC4243708  PMC4243708.txt  \n",
       "59          genetic variant  PMC4243708  PMC4243708.txt  \n",
       "63          genetic variant  PMC4243708  PMC4243708.txt  \n",
       "71          genetic variant  PMC4243708  PMC4243708.txt  \n",
       "77          genetic variant  PMC4243708  PMC4243708.txt  \n",
       "83          genetic variant  PMC4243708  PMC4243708.txt  \n",
       "234                   assay  PMC4243708  PMC4243708.txt  \n",
       "288                 disease  PMC4243708  PMC4243708.txt  \n",
       "289               phenotype  PMC4243708  PMC4243708.txt  \n",
       "291         protein variant  PMC4243708  PMC4243708.txt  \n",
       "292         protein variant  PMC4243708  PMC4243708.txt  \n",
       "293      transcript variant  PMC4243708  PMC4243708.txt  \n",
       "294      transcript variant  PMC4243708  PMC4243708.txt  \n",
       "295         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "296         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "297         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "299               phenotype  PMC4243708  PMC4243708.txt  \n",
       "300               phenotype  PMC4243708  PMC4243708.txt  \n",
       "301               phenotype  PMC4243708  PMC4243708.txt  \n",
       "302                organism  PMC4243708  PMC4243708.txt  \n",
       "303                organism  PMC4243708  PMC4243708.txt  \n",
       "304         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "307         protein variant  PMC4243708  PMC4243708.txt  \n",
       "308         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "309         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "312         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "313         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "314                 <other>  PMC4243708  PMC4243708.txt  \n",
       "315      transcript variant  PMC4243708  PMC4243708.txt  \n",
       "316         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "317                 <other>  PMC4243708  PMC4243708.txt  \n",
       "318         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "319         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "320         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "321         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "322         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "323         protein variant  PMC4243708  PMC4243708.txt  \n",
       "324         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "325         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "326         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "327      molecular activity  PMC4243708  PMC4243708.txt  \n",
       "328         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "329                 <other>  PMC4243708  PMC4243708.txt  \n",
       "330      transcript variant  PMC4243708  PMC4243708.txt  \n",
       "331         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "332                 <other>  PMC4243708  PMC4243708.txt  \n",
       "349         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "398               phenotype  PMC4243708  PMC4243708.txt  \n",
       "399  macromolecular complex  PMC4243708  PMC4243708.txt  \n",
       "408                 disease  PMC4243708  PMC4243708.txt  \n",
       "424         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "425               phenotype  PMC4243708  PMC4243708.txt  \n",
       "430                    gene  PMC4243708  PMC4243708.txt  \n",
       "431         protein variant  PMC4243708  PMC4243708.txt  \n",
       "432               phenotype  PMC4243708  PMC4243708.txt  \n",
       "433                 disease  PMC4243708  PMC4243708.txt  \n",
       "435         genetic variant  PMC4243708  PMC4243708.txt  \n",
       "436                organism  PMC4243708  PMC4243708.txt  \n",
       "437                 disease  PMC4243708  PMC4243708.txt  \n",
       "470         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "472         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "473         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "478         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "479         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "536         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "537                 disease  PMC7477955  PMC7477955.txt  \n",
       "540         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "543         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "544                    gene  PMC7477955  PMC7477955.txt  \n",
       "545         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "546                    gene  PMC7477955  PMC7477955.txt  \n",
       "547         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "548                    gene  PMC7477955  PMC7477955.txt  \n",
       "549         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "550                    gene  PMC7477955  PMC7477955.txt  \n",
       "551         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "552                    gene  PMC7477955  PMC7477955.txt  \n",
       "553         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "554                    gene  PMC7477955  PMC7477955.txt  \n",
       "555         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "556         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "557         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "558         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "559         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "560         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "561         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "562         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "563         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "564         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "565         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "566         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "567         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "568                 <other>  PMC7477955  PMC7477955.txt  \n",
       "569                 <other>  PMC7477955  PMC7477955.txt  \n",
       "570                 <other>  PMC7477955  PMC7477955.txt  \n",
       "571                 <other>  PMC7477955  PMC7477955.txt  \n",
       "572                 <other>  PMC7477955  PMC7477955.txt  \n",
       "582                 disease  PMC7477955  PMC7477955.txt  \n",
       "583                 disease  PMC7477955  PMC7477955.txt  \n",
       "584                 disease  PMC7477955  PMC7477955.txt  \n",
       "585               phenotype  PMC7477955  PMC7477955.txt  \n",
       "586               phenotype  PMC7477955  PMC7477955.txt  \n",
       "587               phenotype  PMC7477955  PMC7477955.txt  \n",
       "726               phenotype  PMC7477955  PMC7477955.txt  \n",
       "727               phenotype  PMC7477955  PMC7477955.txt  \n",
       "728               phenotype  PMC7477955  PMC7477955.txt  \n",
       "729                 protein  PMC7477955  PMC7477955.txt  \n",
       "872      cellular component  PMC7477955  PMC7477955.txt  \n",
       "873         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "874         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "875         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "876         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "877         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "878         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "879         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "880         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "881         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "882         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "883         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "884         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "885         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "886         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "887         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "888         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "889         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "890         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "891         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "892         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "893         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "894         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "895         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "896         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "897         genetic variant  PMC7477955  PMC7477955.txt  \n",
       "898         genetic variant  PMC7477955  PMC7477955.txt  "
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    relations\n",
    "    .pipe(lambda df: df[\n",
    "        df['subject_entity'].str.contains('variant|') | \n",
    "        df['object_entity'].str.contains('variant')\n",
    "    ])\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
